// @pulumi/aws-native/connect/integrationAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Connect::IntegrationAssociation
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const integrationAssociation = new aws_native.connect.IntegrationAssociation("integrationAssociation", {
 *     instanceId: "arn:aws:connect:region-name:aws-account-id:instance/instance-id",
 *     integrationType: aws_native.connect.IntegrationAssociationIntegrationType.LexBot,
 *     integrationArn: "arn:aws:lex:region-name:aws-account-id:bot/bot-name",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const integrationAssociation = new aws_native.connect.IntegrationAssociation("integrationAssociation", {
 *     instanceId: "arn:aws:connect:region-name:aws-account-id:instance/instance-id",
 *     integrationType: aws_native.connect.IntegrationAssociationIntegrationType.LexBot,
 *     integrationArn: "arn:aws:lex:region-name:aws-account-id:bot-alias/bot-id/alias-id",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const integrationAssociation = new aws_native.connect.IntegrationAssociation("integrationAssociation", {
 *     instanceId: "arn:aws:connect:region-name:aws-account-id:instance/instance-id",
 *     integrationType: aws_native.connect.IntegrationAssociationIntegrationType.LambdaFunction,
 *     integrationArn: "arn:aws:lambda:region-name:aws-account-id:function:function-arn",
 * });
 *
 * ```
 */
class IntegrationAssociation extends pulumi.CustomResource {
    /**
     * Get an existing IntegrationAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IntegrationAssociation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IntegrationAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IntegrationAssociation.__pulumiType;
    }
    /**
     * Create a IntegrationAssociation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceId'");
            }
            if ((!args || args.integrationArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integrationArn'");
            }
            if ((!args || args.integrationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integrationType'");
            }
            resourceInputs["instanceId"] = args ? args.instanceId : undefined;
            resourceInputs["integrationArn"] = args ? args.integrationArn : undefined;
            resourceInputs["integrationType"] = args ? args.integrationType : undefined;
            resourceInputs["integrationAssociationId"] = undefined /*out*/;
        }
        else {
            resourceInputs["instanceId"] = undefined /*out*/;
            resourceInputs["integrationArn"] = undefined /*out*/;
            resourceInputs["integrationAssociationId"] = undefined /*out*/;
            resourceInputs["integrationType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["instanceId", "integrationArn", "integrationType"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(IntegrationAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IntegrationAssociation = IntegrationAssociation;
/** @internal */
IntegrationAssociation.__pulumiType = 'aws-native:connect:IntegrationAssociation';
