// @pulumi/aws-native/connect/routingProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Connect::RoutingProfile
 */
class RoutingProfile extends pulumi.CustomResource {
    /**
     * Get an existing RoutingProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RoutingProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RoutingProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RoutingProfile.__pulumiType;
    }
    /**
     * Create a RoutingProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.defaultOutboundQueueArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultOutboundQueueArn'");
            }
            if ((!args || args.description === undefined) && !opts.urn) {
                throw new Error("Missing required property 'description'");
            }
            if ((!args || args.instanceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceArn'");
            }
            if ((!args || args.mediaConcurrencies === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mediaConcurrencies'");
            }
            resourceInputs["agentAvailabilityTimer"] = args ? args.agentAvailabilityTimer : undefined;
            resourceInputs["defaultOutboundQueueArn"] = args ? args.defaultOutboundQueueArn : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["instanceArn"] = args ? args.instanceArn : undefined;
            resourceInputs["mediaConcurrencies"] = args ? args.mediaConcurrencies : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["queueConfigs"] = args ? args.queueConfigs : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["routingProfileArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["agentAvailabilityTimer"] = undefined /*out*/;
            resourceInputs["defaultOutboundQueueArn"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["instanceArn"] = undefined /*out*/;
            resourceInputs["mediaConcurrencies"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["queueConfigs"] = undefined /*out*/;
            resourceInputs["routingProfileArn"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RoutingProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RoutingProfile = RoutingProfile;
/** @internal */
RoutingProfile.__pulumiType = 'aws-native:connect:RoutingProfile';
