// @pulumi/aws-native/connect/securityProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Connect::SecurityProfile
 */
class SecurityProfile extends pulumi.CustomResource {
    /**
     * Get an existing SecurityProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SecurityProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecurityProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityProfile.__pulumiType;
    }
    /**
     * Create a SecurityProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceArn'");
            }
            resourceInputs["allowedAccessControlHierarchyGroupId"] = args ? args.allowedAccessControlHierarchyGroupId : undefined;
            resourceInputs["allowedAccessControlTags"] = args ? args.allowedAccessControlTags : undefined;
            resourceInputs["applications"] = args ? args.applications : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["hierarchyRestrictedResources"] = args ? args.hierarchyRestrictedResources : undefined;
            resourceInputs["instanceArn"] = args ? args.instanceArn : undefined;
            resourceInputs["permissions"] = args ? args.permissions : undefined;
            resourceInputs["securityProfileName"] = args ? args.securityProfileName : undefined;
            resourceInputs["tagRestrictedResources"] = args ? args.tagRestrictedResources : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["lastModifiedRegion"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["securityProfileArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowedAccessControlHierarchyGroupId"] = undefined /*out*/;
            resourceInputs["allowedAccessControlTags"] = undefined /*out*/;
            resourceInputs["applications"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["hierarchyRestrictedResources"] = undefined /*out*/;
            resourceInputs["instanceArn"] = undefined /*out*/;
            resourceInputs["lastModifiedRegion"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["permissions"] = undefined /*out*/;
            resourceInputs["securityProfileArn"] = undefined /*out*/;
            resourceInputs["securityProfileName"] = undefined /*out*/;
            resourceInputs["tagRestrictedResources"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["instanceArn", "securityProfileName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(SecurityProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecurityProfile = SecurityProfile;
/** @internal */
SecurityProfile.__pulumiType = 'aws-native:connect:SecurityProfile';
