// @pulumi/aws-native/guardduty/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreatIntelSet = exports.Member = exports.Master = exports.MalwareProtectionPlan = exports.IpSet = exports.getThreatIntelSetOutput = exports.getThreatIntelSet = exports.getMemberOutput = exports.getMember = exports.getMalwareProtectionPlanOutput = exports.getMalwareProtectionPlan = exports.getIpSetOutput = exports.getIpSet = exports.getFilterOutput = exports.getFilter = exports.getDetectorOutput = exports.getDetector = exports.Filter = exports.Detector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Detector = null;
utilities.lazyLoad(exports, ["Detector"], () => require("./detector"));
exports.Filter = null;
utilities.lazyLoad(exports, ["Filter"], () => require("./filter"));
exports.getDetector = null;
exports.getDetectorOutput = null;
utilities.lazyLoad(exports, ["getDetector", "getDetectorOutput"], () => require("./getDetector"));
exports.getFilter = null;
exports.getFilterOutput = null;
utilities.lazyLoad(exports, ["getFilter", "getFilterOutput"], () => require("./getFilter"));
exports.getIpSet = null;
exports.getIpSetOutput = null;
utilities.lazyLoad(exports, ["getIpSet", "getIpSetOutput"], () => require("./getIpSet"));
exports.getMalwareProtectionPlan = null;
exports.getMalwareProtectionPlanOutput = null;
utilities.lazyLoad(exports, ["getMalwareProtectionPlan", "getMalwareProtectionPlanOutput"], () => require("./getMalwareProtectionPlan"));
exports.getMember = null;
exports.getMemberOutput = null;
utilities.lazyLoad(exports, ["getMember", "getMemberOutput"], () => require("./getMember"));
exports.getThreatIntelSet = null;
exports.getThreatIntelSetOutput = null;
utilities.lazyLoad(exports, ["getThreatIntelSet", "getThreatIntelSetOutput"], () => require("./getThreatIntelSet"));
exports.IpSet = null;
utilities.lazyLoad(exports, ["IpSet"], () => require("./ipSet"));
exports.MalwareProtectionPlan = null;
utilities.lazyLoad(exports, ["MalwareProtectionPlan"], () => require("./malwareProtectionPlan"));
exports.Master = null;
utilities.lazyLoad(exports, ["Master"], () => require("./master"));
exports.Member = null;
utilities.lazyLoad(exports, ["Member"], () => require("./member"));
exports.ThreatIntelSet = null;
utilities.lazyLoad(exports, ["ThreatIntelSet"], () => require("./threatIntelSet"));
// Export enums:
__exportStar(require("../types/enums/guardduty"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:guardduty:Detector":
                return new exports.Detector(name, undefined, { urn });
            case "aws-native:guardduty:Filter":
                return new exports.Filter(name, undefined, { urn });
            case "aws-native:guardduty:IpSet":
                return new exports.IpSet(name, undefined, { urn });
            case "aws-native:guardduty:MalwareProtectionPlan":
                return new exports.MalwareProtectionPlan(name, undefined, { urn });
            case "aws-native:guardduty:Master":
                return new exports.Master(name, undefined, { urn });
            case "aws-native:guardduty:Member":
                return new exports.Member(name, undefined, { urn });
            case "aws-native:guardduty:ThreatIntelSet":
                return new exports.ThreatIntelSet(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "guardduty", _module);
