// @pulumi/aws-native/elasticache/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserGroup = exports.User = exports.SubnetGroup = exports.ServerlessCache = exports.ParameterGroup = exports.GlobalReplicationGroup = exports.getUserGroupOutput = exports.getUserGroup = exports.getUserOutput = exports.getUser = exports.getSubnetGroupOutput = exports.getSubnetGroup = exports.getServerlessCacheOutput = exports.getServerlessCache = exports.getParameterGroupOutput = exports.getParameterGroup = exports.getGlobalReplicationGroupOutput = exports.getGlobalReplicationGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGlobalReplicationGroup = null;
exports.getGlobalReplicationGroupOutput = null;
utilities.lazyLoad(exports, ["getGlobalReplicationGroup", "getGlobalReplicationGroupOutput"], () => require("./getGlobalReplicationGroup"));
exports.getParameterGroup = null;
exports.getParameterGroupOutput = null;
utilities.lazyLoad(exports, ["getParameterGroup", "getParameterGroupOutput"], () => require("./getParameterGroup"));
exports.getServerlessCache = null;
exports.getServerlessCacheOutput = null;
utilities.lazyLoad(exports, ["getServerlessCache", "getServerlessCacheOutput"], () => require("./getServerlessCache"));
exports.getSubnetGroup = null;
exports.getSubnetGroupOutput = null;
utilities.lazyLoad(exports, ["getSubnetGroup", "getSubnetGroupOutput"], () => require("./getSubnetGroup"));
exports.getUser = null;
exports.getUserOutput = null;
utilities.lazyLoad(exports, ["getUser", "getUserOutput"], () => require("./getUser"));
exports.getUserGroup = null;
exports.getUserGroupOutput = null;
utilities.lazyLoad(exports, ["getUserGroup", "getUserGroupOutput"], () => require("./getUserGroup"));
exports.GlobalReplicationGroup = null;
utilities.lazyLoad(exports, ["GlobalReplicationGroup"], () => require("./globalReplicationGroup"));
exports.ParameterGroup = null;
utilities.lazyLoad(exports, ["ParameterGroup"], () => require("./parameterGroup"));
exports.ServerlessCache = null;
utilities.lazyLoad(exports, ["ServerlessCache"], () => require("./serverlessCache"));
exports.SubnetGroup = null;
utilities.lazyLoad(exports, ["SubnetGroup"], () => require("./subnetGroup"));
exports.User = null;
utilities.lazyLoad(exports, ["User"], () => require("./user"));
exports.UserGroup = null;
utilities.lazyLoad(exports, ["UserGroup"], () => require("./userGroup"));
// Export enums:
__exportStar(require("../types/enums/elasticache"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:elasticache:GlobalReplicationGroup":
                return new exports.GlobalReplicationGroup(name, undefined, { urn });
            case "aws-native:elasticache:ParameterGroup":
                return new exports.ParameterGroup(name, undefined, { urn });
            case "aws-native:elasticache:ServerlessCache":
                return new exports.ServerlessCache(name, undefined, { urn });
            case "aws-native:elasticache:SubnetGroup":
                return new exports.SubnetGroup(name, undefined, { urn });
            case "aws-native:elasticache:User":
                return new exports.User(name, undefined, { urn });
            case "aws-native:elasticache:UserGroup":
                return new exports.UserGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "elasticache", _module);
