// @pulumi/aws-native/apigateway/basePathMappingV2.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasePathMappingV2 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ApiGateway::BasePathMappingV2
 */
class BasePathMappingV2 extends pulumi.CustomResource {
    /**
     * Get an existing BasePathMappingV2 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BasePathMappingV2(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BasePathMappingV2.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BasePathMappingV2.__pulumiType;
    }
    /**
     * Create a BasePathMappingV2 resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.domainNameArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainNameArn'");
            }
            if ((!args || args.restApiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'restApiId'");
            }
            resourceInputs["basePath"] = args ? args.basePath : undefined;
            resourceInputs["domainNameArn"] = args ? args.domainNameArn : undefined;
            resourceInputs["restApiId"] = args ? args.restApiId : undefined;
            resourceInputs["stage"] = args ? args.stage : undefined;
            resourceInputs["basePathMappingArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["basePath"] = undefined /*out*/;
            resourceInputs["basePathMappingArn"] = undefined /*out*/;
            resourceInputs["domainNameArn"] = undefined /*out*/;
            resourceInputs["restApiId"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["basePath", "domainNameArn"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(BasePathMappingV2.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BasePathMappingV2 = BasePathMappingV2;
/** @internal */
BasePathMappingV2.__pulumiType = 'aws-native:apigateway:BasePathMappingV2';
