// @pulumi/aws-native/apigateway/domainNameV2.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainNameV2 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ApiGateway::DomainNameV2.
 */
class DomainNameV2 extends pulumi.CustomResource {
    /**
     * Get an existing DomainNameV2 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DomainNameV2(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DomainNameV2.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainNameV2.__pulumiType;
    }
    /**
     * Create a DomainNameV2 resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["certificateArn"] = args ? args.certificateArn : undefined;
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["endpointConfiguration"] = args ? args.endpointConfiguration : undefined;
            resourceInputs["policy"] = args ? args.policy : undefined;
            resourceInputs["securityPolicy"] = args ? args.securityPolicy : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["domainNameArn"] = undefined /*out*/;
            resourceInputs["domainNameId"] = undefined /*out*/;
        }
        else {
            resourceInputs["certificateArn"] = undefined /*out*/;
            resourceInputs["domainName"] = undefined /*out*/;
            resourceInputs["domainNameArn"] = undefined /*out*/;
            resourceInputs["domainNameId"] = undefined /*out*/;
            resourceInputs["endpointConfiguration"] = undefined /*out*/;
            resourceInputs["policy"] = undefined /*out*/;
            resourceInputs["securityPolicy"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["domainName", "endpointConfiguration", "securityPolicy"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(DomainNameV2.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DomainNameV2 = DomainNameV2;
/** @internal */
DomainNameV2.__pulumiType = 'aws-native:apigateway:DomainNameV2';
