// @pulumi/aws-native/verifiedpermissions/identitySource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentitySource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::VerifiedPermissions::IdentitySource Resource Type
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const config = new pulumi.Config();
 * const policyStoreId = config.require("policyStoreId");
 * const userPoolArn = config.require("userPoolArn");
 * const clientIds = config.requireObject<Array<string>>("clientIds");
 * const principalEntityType = config.require("principalEntityType");
 * const identitySource = new aws_native.verifiedpermissions.IdentitySource("identitySource", {
 *     policyStoreId: policyStoreId,
 *     configuration: {
 *         cognitoUserPoolConfiguration: {
 *             userPoolArn: userPoolArn,
 *             clientIds: clientIds,
 *         },
 *     },
 *     principalEntityType: principalEntityType,
 * });
 * export const identitySourceId = identitySource.identitySourceId;
 *
 * ```
 */
class IdentitySource extends pulumi.CustomResource {
    /**
     * Get an existing IdentitySource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IdentitySource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IdentitySource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IdentitySource.__pulumiType;
    }
    /**
     * Create a IdentitySource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.configuration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configuration'");
            }
            if ((!args || args.policyStoreId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'policyStoreId'");
            }
            resourceInputs["configuration"] = args ? args.configuration : undefined;
            resourceInputs["policyStoreId"] = args ? args.policyStoreId : undefined;
            resourceInputs["principalEntityType"] = args ? args.principalEntityType : undefined;
            resourceInputs["details"] = undefined /*out*/;
            resourceInputs["identitySourceId"] = undefined /*out*/;
        }
        else {
            resourceInputs["configuration"] = undefined /*out*/;
            resourceInputs["details"] = undefined /*out*/;
            resourceInputs["identitySourceId"] = undefined /*out*/;
            resourceInputs["policyStoreId"] = undefined /*out*/;
            resourceInputs["principalEntityType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["policyStoreId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(IdentitySource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IdentitySource = IdentitySource;
/** @internal */
IdentitySource.__pulumiType = 'aws-native:verifiedpermissions:IdentitySource';
