// @pulumi/aws-native/verifiedpermissions/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyTemplate = exports.PolicyStore = exports.Policy = exports.IdentitySource = exports.getPolicyTemplateOutput = exports.getPolicyTemplate = exports.getPolicyStoreOutput = exports.getPolicyStore = exports.getPolicyOutput = exports.getPolicy = exports.getIdentitySourceOutput = exports.getIdentitySource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getIdentitySource = null;
exports.getIdentitySourceOutput = null;
utilities.lazyLoad(exports, ["getIdentitySource", "getIdentitySourceOutput"], () => require("./getIdentitySource"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getPolicyStore = null;
exports.getPolicyStoreOutput = null;
utilities.lazyLoad(exports, ["getPolicyStore", "getPolicyStoreOutput"], () => require("./getPolicyStore"));
exports.getPolicyTemplate = null;
exports.getPolicyTemplateOutput = null;
utilities.lazyLoad(exports, ["getPolicyTemplate", "getPolicyTemplateOutput"], () => require("./getPolicyTemplate"));
exports.IdentitySource = null;
utilities.lazyLoad(exports, ["IdentitySource"], () => require("./identitySource"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.PolicyStore = null;
utilities.lazyLoad(exports, ["PolicyStore"], () => require("./policyStore"));
exports.PolicyTemplate = null;
utilities.lazyLoad(exports, ["PolicyTemplate"], () => require("./policyTemplate"));
// Export enums:
__exportStar(require("../types/enums/verifiedpermissions"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:verifiedpermissions:IdentitySource":
                return new exports.IdentitySource(name, undefined, { urn });
            case "aws-native:verifiedpermissions:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "aws-native:verifiedpermissions:PolicyStore":
                return new exports.PolicyStore(name, undefined, { urn });
            case "aws-native:verifiedpermissions:PolicyTemplate":
                return new exports.PolicyTemplate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "verifiedpermissions", _module);
