// @pulumi/aws-native/verifiedpermissions/policyTemplate.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::VerifiedPermissions::PolicyTemplate Resource Type
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const config = new pulumi.Config();
 * const policyStoreId = config.require("policyStoreId");
 * const description = config.require("description");
 * const statement = config.require("statement");
 * const policyTemplate = new aws_native.verifiedpermissions.PolicyTemplate("policyTemplate", {
 *     policyStoreId: policyStoreId,
 *     description: description,
 *     statement: statement,
 * });
 * export const policyTemplateId = policyTemplate.policyTemplateId;
 *
 * ```
 */
class PolicyTemplate extends pulumi.CustomResource {
    /**
     * Get an existing PolicyTemplate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PolicyTemplate(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PolicyTemplate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PolicyTemplate.__pulumiType;
    }
    /**
     * Create a PolicyTemplate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.policyStoreId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'policyStoreId'");
            }
            if ((!args || args.statement === undefined) && !opts.urn) {
                throw new Error("Missing required property 'statement'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["policyStoreId"] = args ? args.policyStoreId : undefined;
            resourceInputs["statement"] = args ? args.statement : undefined;
            resourceInputs["policyTemplateId"] = undefined /*out*/;
        }
        else {
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["policyStoreId"] = undefined /*out*/;
            resourceInputs["policyTemplateId"] = undefined /*out*/;
            resourceInputs["statement"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["policyStoreId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(PolicyTemplate.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PolicyTemplate = PolicyTemplate;
/** @internal */
PolicyTemplate.__pulumiType = 'aws-native:verifiedpermissions:PolicyTemplate';
