// @pulumi/aws-native/omics/annotationStore.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnnotationStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Omics::AnnotationStore Resource Type
 */
class AnnotationStore extends pulumi.CustomResource {
    /**
     * Get an existing AnnotationStore resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AnnotationStore(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AnnotationStore.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnnotationStore.__pulumiType;
    }
    /**
     * Create a AnnotationStore resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.storeFormat === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storeFormat'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["reference"] = args ? args.reference : undefined;
            resourceInputs["sseConfig"] = args ? args.sseConfig : undefined;
            resourceInputs["storeFormat"] = args ? args.storeFormat : undefined;
            resourceInputs["storeOptions"] = args ? args.storeOptions : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["awsId"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["storeArn"] = undefined /*out*/;
            resourceInputs["storeSizeBytes"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        else {
            resourceInputs["awsId"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["reference"] = undefined /*out*/;
            resourceInputs["sseConfig"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusMessage"] = undefined /*out*/;
            resourceInputs["storeArn"] = undefined /*out*/;
            resourceInputs["storeFormat"] = undefined /*out*/;
            resourceInputs["storeOptions"] = undefined /*out*/;
            resourceInputs["storeSizeBytes"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["name", "reference", "sseConfig", "storeFormat", "storeOptions", "tags.*"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(AnnotationStore.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AnnotationStore = AnnotationStore;
/** @internal */
AnnotationStore.__pulumiType = 'aws-native:omics:AnnotationStore';
