// @pulumi/aws-native/omics/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workflow = exports.VariantStore = exports.SequenceStore = exports.RunGroup = exports.ReferenceStore = exports.getWorkflowOutput = exports.getWorkflow = exports.getVariantStoreOutput = exports.getVariantStore = exports.getSequenceStoreOutput = exports.getSequenceStore = exports.getRunGroupOutput = exports.getRunGroup = exports.getReferenceStoreOutput = exports.getReferenceStore = exports.getAnnotationStoreOutput = exports.getAnnotationStore = exports.AnnotationStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnnotationStore = null;
utilities.lazyLoad(exports, ["AnnotationStore"], () => require("./annotationStore"));
exports.getAnnotationStore = null;
exports.getAnnotationStoreOutput = null;
utilities.lazyLoad(exports, ["getAnnotationStore", "getAnnotationStoreOutput"], () => require("./getAnnotationStore"));
exports.getReferenceStore = null;
exports.getReferenceStoreOutput = null;
utilities.lazyLoad(exports, ["getReferenceStore", "getReferenceStoreOutput"], () => require("./getReferenceStore"));
exports.getRunGroup = null;
exports.getRunGroupOutput = null;
utilities.lazyLoad(exports, ["getRunGroup", "getRunGroupOutput"], () => require("./getRunGroup"));
exports.getSequenceStore = null;
exports.getSequenceStoreOutput = null;
utilities.lazyLoad(exports, ["getSequenceStore", "getSequenceStoreOutput"], () => require("./getSequenceStore"));
exports.getVariantStore = null;
exports.getVariantStoreOutput = null;
utilities.lazyLoad(exports, ["getVariantStore", "getVariantStoreOutput"], () => require("./getVariantStore"));
exports.getWorkflow = null;
exports.getWorkflowOutput = null;
utilities.lazyLoad(exports, ["getWorkflow", "getWorkflowOutput"], () => require("./getWorkflow"));
exports.ReferenceStore = null;
utilities.lazyLoad(exports, ["ReferenceStore"], () => require("./referenceStore"));
exports.RunGroup = null;
utilities.lazyLoad(exports, ["RunGroup"], () => require("./runGroup"));
exports.SequenceStore = null;
utilities.lazyLoad(exports, ["SequenceStore"], () => require("./sequenceStore"));
exports.VariantStore = null;
utilities.lazyLoad(exports, ["VariantStore"], () => require("./variantStore"));
exports.Workflow = null;
utilities.lazyLoad(exports, ["Workflow"], () => require("./workflow"));
// Export enums:
__exportStar(require("../types/enums/omics"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:omics:AnnotationStore":
                return new exports.AnnotationStore(name, undefined, { urn });
            case "aws-native:omics:ReferenceStore":
                return new exports.ReferenceStore(name, undefined, { urn });
            case "aws-native:omics:RunGroup":
                return new exports.RunGroup(name, undefined, { urn });
            case "aws-native:omics:SequenceStore":
                return new exports.SequenceStore(name, undefined, { urn });
            case "aws-native:omics:VariantStore":
                return new exports.VariantStore(name, undefined, { urn });
            case "aws-native:omics:Workflow":
                return new exports.Workflow(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "omics", _module);
