// @pulumi/aws-native/networkmanager/directConnectGatewayAttachment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DirectConnectGatewayAttachment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * AWS::NetworkManager::DirectConnectGatewayAttachment Resource Type
 */
class DirectConnectGatewayAttachment extends pulumi.CustomResource {
    /**
     * Get an existing DirectConnectGatewayAttachment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DirectConnectGatewayAttachment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DirectConnectGatewayAttachment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DirectConnectGatewayAttachment.__pulumiType;
    }
    /**
     * Create a DirectConnectGatewayAttachment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.coreNetworkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'coreNetworkId'");
            }
            if ((!args || args.directConnectGatewayArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'directConnectGatewayArn'");
            }
            if ((!args || args.edgeLocations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'edgeLocations'");
            }
            resourceInputs["coreNetworkId"] = args ? args.coreNetworkId : undefined;
            resourceInputs["directConnectGatewayArn"] = args ? args.directConnectGatewayArn : undefined;
            resourceInputs["edgeLocations"] = args ? args.edgeLocations : undefined;
            resourceInputs["proposedNetworkFunctionGroupChange"] = args ? args.proposedNetworkFunctionGroupChange : undefined;
            resourceInputs["proposedSegmentChange"] = args ? args.proposedSegmentChange : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["attachmentId"] = undefined /*out*/;
            resourceInputs["attachmentPolicyRuleNumber"] = undefined /*out*/;
            resourceInputs["attachmentType"] = undefined /*out*/;
            resourceInputs["coreNetworkArn"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["networkFunctionGroupName"] = undefined /*out*/;
            resourceInputs["ownerAccountId"] = undefined /*out*/;
            resourceInputs["resourceArn"] = undefined /*out*/;
            resourceInputs["segmentName"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        else {
            resourceInputs["attachmentId"] = undefined /*out*/;
            resourceInputs["attachmentPolicyRuleNumber"] = undefined /*out*/;
            resourceInputs["attachmentType"] = undefined /*out*/;
            resourceInputs["coreNetworkArn"] = undefined /*out*/;
            resourceInputs["coreNetworkId"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["directConnectGatewayArn"] = undefined /*out*/;
            resourceInputs["edgeLocations"] = undefined /*out*/;
            resourceInputs["networkFunctionGroupName"] = undefined /*out*/;
            resourceInputs["ownerAccountId"] = undefined /*out*/;
            resourceInputs["proposedNetworkFunctionGroupChange"] = undefined /*out*/;
            resourceInputs["proposedSegmentChange"] = undefined /*out*/;
            resourceInputs["resourceArn"] = undefined /*out*/;
            resourceInputs["segmentName"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["coreNetworkId", "directConnectGatewayArn"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(DirectConnectGatewayAttachment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DirectConnectGatewayAttachment = DirectConnectGatewayAttachment;
/** @internal */
DirectConnectGatewayAttachment.__pulumiType = 'aws-native:networkmanager:DirectConnectGatewayAttachment';
