// @pulumi/aws-native/types/enums/entityresolution/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaMappingSchemaAttributeType = exports.PolicyStatementStatementEffect = exports.MatchingWorkflowRuleBasedPropertiesMatchPurpose = exports.MatchingWorkflowRuleBasedPropertiesAttributeMatchingModel = exports.MatchingWorkflowResolutionType = exports.MatchingWorkflowIncrementalRunConfigIncrementalRunType = exports.IdNamespaceType = exports.IdNamespaceRuleDefinitionType = exports.IdNamespaceRecordMatchingModel = exports.IdNamespaceNamespaceRuleBasedPropertiesAttributeMatchingModel = exports.IdNamespaceIdMappingWorkflowPropertiesIdMappingType = exports.IdMappingWorkflowInputSourceType = exports.IdMappingWorkflowIdMappingTechniquesIdMappingType = exports.IdMappingWorkflowIdMappingRuleBasedPropertiesRuleDefinitionType = exports.IdMappingWorkflowIdMappingRuleBasedPropertiesRecordMatchingModel = exports.IdMappingWorkflowIdMappingRuleBasedPropertiesAttributeMatchingModel = void 0;
exports.IdMappingWorkflowIdMappingRuleBasedPropertiesAttributeMatchingModel = {
    OneToOne: "ONE_TO_ONE",
    ManyToMany: "MANY_TO_MANY",
};
exports.IdMappingWorkflowIdMappingRuleBasedPropertiesRecordMatchingModel = {
    OneSourceToOneTarget: "ONE_SOURCE_TO_ONE_TARGET",
    ManySourceToOneTarget: "MANY_SOURCE_TO_ONE_TARGET",
};
exports.IdMappingWorkflowIdMappingRuleBasedPropertiesRuleDefinitionType = {
    Source: "SOURCE",
    Target: "TARGET",
};
exports.IdMappingWorkflowIdMappingTechniquesIdMappingType = {
    Provider: "PROVIDER",
    RuleBased: "RULE_BASED",
};
exports.IdMappingWorkflowInputSourceType = {
    Source: "SOURCE",
    Target: "TARGET",
};
exports.IdNamespaceIdMappingWorkflowPropertiesIdMappingType = {
    Provider: "PROVIDER",
    RuleBased: "RULE_BASED",
};
exports.IdNamespaceNamespaceRuleBasedPropertiesAttributeMatchingModel = {
    OneToOne: "ONE_TO_ONE",
    ManyToMany: "MANY_TO_MANY",
};
exports.IdNamespaceRecordMatchingModel = {
    OneSourceToOneTarget: "ONE_SOURCE_TO_ONE_TARGET",
    ManySourceToOneTarget: "MANY_SOURCE_TO_ONE_TARGET",
};
exports.IdNamespaceRuleDefinitionType = {
    Source: "SOURCE",
    Target: "TARGET",
};
exports.IdNamespaceType = {
    Source: "SOURCE",
    Target: "TARGET",
};
exports.MatchingWorkflowIncrementalRunConfigIncrementalRunType = {
    Immediate: "IMMEDIATE",
};
exports.MatchingWorkflowResolutionType = {
    RuleMatching: "RULE_MATCHING",
    MlMatching: "ML_MATCHING",
    Provider: "PROVIDER",
};
exports.MatchingWorkflowRuleBasedPropertiesAttributeMatchingModel = {
    OneToOne: "ONE_TO_ONE",
    ManyToMany: "MANY_TO_MANY",
};
exports.MatchingWorkflowRuleBasedPropertiesMatchPurpose = {
    IdentifierGeneration: "IDENTIFIER_GENERATION",
    Indexing: "INDEXING",
};
exports.PolicyStatementStatementEffect = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.SchemaMappingSchemaAttributeType = {
    Name: "NAME",
    NameFirst: "NAME_FIRST",
    NameMiddle: "NAME_MIDDLE",
    NameLast: "NAME_LAST",
    Address: "ADDRESS",
    AddressStreet1: "ADDRESS_STREET1",
    AddressStreet2: "ADDRESS_STREET2",
    AddressStreet3: "ADDRESS_STREET3",
    AddressCity: "ADDRESS_CITY",
    AddressState: "ADDRESS_STATE",
    AddressCountry: "ADDRESS_COUNTRY",
    AddressPostalcode: "ADDRESS_POSTALCODE",
    Phone: "PHONE",
    PhoneNumber: "PHONE_NUMBER",
    PhoneCountrycode: "PHONE_COUNTRYCODE",
    EmailAddress: "EMAIL_ADDRESS",
    UniqueId: "UNIQUE_ID",
    Date: "DATE",
    String: "STRING",
    ProviderId: "PROVIDER_ID",
};
