// @pulumi/aws-native/types/enums/mediapackagev2/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OriginEndpointTsEncryptionMethod = exports.OriginEndpointScteFilter = exports.OriginEndpointPresetSpeke20Video = exports.OriginEndpointPresetSpeke20Audio = exports.OriginEndpointEndpointErrorCondition = exports.OriginEndpointDrmSystem = exports.OriginEndpointDashUtcTimingMode = exports.OriginEndpointDashSegmentTemplateFormat = exports.OriginEndpointDashPeriodTrigger = exports.OriginEndpointDashDrmSignaling = exports.OriginEndpointContainerType = exports.OriginEndpointCmafEncryptionMethod = exports.OriginEndpointAdMarkerHls = exports.OriginEndpointAdMarkerDash = exports.ChannelInputType = void 0;
exports.ChannelInputType = {
    Hls: "HLS",
    Cmaf: "CMAF",
};
exports.OriginEndpointAdMarkerDash = {
    Binary: "BINARY",
    Xml: "XML",
};
exports.OriginEndpointAdMarkerHls = {
    Daterange: "DATERANGE",
};
exports.OriginEndpointCmafEncryptionMethod = {
    Cenc: "CENC",
    Cbcs: "CBCS",
};
exports.OriginEndpointContainerType = {
    Ts: "TS",
    Cmaf: "CMAF",
};
exports.OriginEndpointDashDrmSignaling = {
    Individual: "INDIVIDUAL",
    Referenced: "REFERENCED",
};
exports.OriginEndpointDashPeriodTrigger = {
    Avails: "AVAILS",
    DrmKeyRotation: "DRM_KEY_ROTATION",
    SourceChanges: "SOURCE_CHANGES",
    SourceDisruptions: "SOURCE_DISRUPTIONS",
    None: "NONE",
};
exports.OriginEndpointDashSegmentTemplateFormat = {
    NumberWithTimeline: "NUMBER_WITH_TIMELINE",
};
exports.OriginEndpointDashUtcTimingMode = {
    HttpHead: "HTTP_HEAD",
    HttpIso: "HTTP_ISO",
    HttpXsdate: "HTTP_XSDATE",
    UtcDirect: "UTC_DIRECT",
};
exports.OriginEndpointDrmSystem = {
    ClearKeyAes128: "CLEAR_KEY_AES_128",
    Fairplay: "FAIRPLAY",
    Playready: "PLAYREADY",
    Widevine: "WIDEVINE",
    Irdeto: "IRDETO",
};
exports.OriginEndpointEndpointErrorCondition = {
    StaleManifest: "STALE_MANIFEST",
    IncompleteManifest: "INCOMPLETE_MANIFEST",
    MissingDrmKey: "MISSING_DRM_KEY",
    SlateInput: "SLATE_INPUT",
};
exports.OriginEndpointPresetSpeke20Audio = {
    PresetAudio1: "PRESET_AUDIO_1",
    PresetAudio2: "PRESET_AUDIO_2",
    PresetAudio3: "PRESET_AUDIO_3",
    Shared: "SHARED",
    Unencrypted: "UNENCRYPTED",
};
exports.OriginEndpointPresetSpeke20Video = {
    PresetVideo1: "PRESET_VIDEO_1",
    PresetVideo2: "PRESET_VIDEO_2",
    PresetVideo3: "PRESET_VIDEO_3",
    PresetVideo4: "PRESET_VIDEO_4",
    PresetVideo5: "PRESET_VIDEO_5",
    PresetVideo6: "PRESET_VIDEO_6",
    PresetVideo7: "PRESET_VIDEO_7",
    PresetVideo8: "PRESET_VIDEO_8",
    Shared: "SHARED",
    Unencrypted: "UNENCRYPTED",
};
exports.OriginEndpointScteFilter = {
    SpliceInsert: "SPLICE_INSERT",
    Break: "BREAK",
    ProviderAdvertisement: "PROVIDER_ADVERTISEMENT",
    DistributorAdvertisement: "DISTRIBUTOR_ADVERTISEMENT",
    ProviderPlacementOpportunity: "PROVIDER_PLACEMENT_OPPORTUNITY",
    DistributorPlacementOpportunity: "DISTRIBUTOR_PLACEMENT_OPPORTUNITY",
    ProviderOverlayPlacementOpportunity: "PROVIDER_OVERLAY_PLACEMENT_OPPORTUNITY",
    DistributorOverlayPlacementOpportunity: "DISTRIBUTOR_OVERLAY_PLACEMENT_OPPORTUNITY",
    Program: "PROGRAM",
};
exports.OriginEndpointTsEncryptionMethod = {
    Aes128: "AES_128",
    SampleAes: "SAMPLE_AES",
};
