// @pulumi/aws-native/internetmonitor/monitor.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Monitor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a monitor, which defines the monitoring boundaries for measurements that Internet Monitor publishes information about for an application
 */
class Monitor extends pulumi.CustomResource {
    /**
     * Get an existing Monitor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Monitor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Monitor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Monitor.__pulumiType;
    }
    /**
     * Create a Monitor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["healthEventsConfig"] = args ? args.healthEventsConfig : undefined;
            resourceInputs["includeLinkedAccounts"] = args ? args.includeLinkedAccounts : undefined;
            resourceInputs["internetMeasurementsLogDelivery"] = args ? args.internetMeasurementsLogDelivery : undefined;
            resourceInputs["linkedAccountId"] = args ? args.linkedAccountId : undefined;
            resourceInputs["maxCityNetworksToMonitor"] = args ? args.maxCityNetworksToMonitor : undefined;
            resourceInputs["monitorName"] = args ? args.monitorName : undefined;
            resourceInputs["resources"] = args ? args.resources : undefined;
            resourceInputs["resourcesToAdd"] = args ? args.resourcesToAdd : undefined;
            resourceInputs["resourcesToRemove"] = args ? args.resourcesToRemove : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["trafficPercentageToMonitor"] = args ? args.trafficPercentageToMonitor : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["modifiedAt"] = undefined /*out*/;
            resourceInputs["monitorArn"] = undefined /*out*/;
            resourceInputs["processingStatus"] = undefined /*out*/;
            resourceInputs["processingStatusInfo"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["healthEventsConfig"] = undefined /*out*/;
            resourceInputs["includeLinkedAccounts"] = undefined /*out*/;
            resourceInputs["internetMeasurementsLogDelivery"] = undefined /*out*/;
            resourceInputs["linkedAccountId"] = undefined /*out*/;
            resourceInputs["maxCityNetworksToMonitor"] = undefined /*out*/;
            resourceInputs["modifiedAt"] = undefined /*out*/;
            resourceInputs["monitorArn"] = undefined /*out*/;
            resourceInputs["monitorName"] = undefined /*out*/;
            resourceInputs["processingStatus"] = undefined /*out*/;
            resourceInputs["processingStatusInfo"] = undefined /*out*/;
            resourceInputs["resources"] = undefined /*out*/;
            resourceInputs["resourcesToAdd"] = undefined /*out*/;
            resourceInputs["resourcesToRemove"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["trafficPercentageToMonitor"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["monitorName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(Monitor.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Monitor = Monitor;
/** @internal */
Monitor.__pulumiType = 'aws-native:internetmonitor:Monitor';
