// @pulumi/aws-native/scheduler/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleGroup = exports.Schedule = exports.getScheduleGroupOutput = exports.getScheduleGroup = exports.getScheduleOutput = exports.getSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getSchedule = null;
exports.getScheduleOutput = null;
utilities.lazyLoad(exports, ["getSchedule", "getScheduleOutput"], () => require("./getSchedule"));
exports.getScheduleGroup = null;
exports.getScheduleGroupOutput = null;
utilities.lazyLoad(exports, ["getScheduleGroup", "getScheduleGroupOutput"], () => require("./getScheduleGroup"));
exports.Schedule = null;
utilities.lazyLoad(exports, ["Schedule"], () => require("./schedule"));
exports.ScheduleGroup = null;
utilities.lazyLoad(exports, ["ScheduleGroup"], () => require("./scheduleGroup"));
// Export enums:
__exportStar(require("../types/enums/scheduler"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:scheduler:Schedule":
                return new exports.Schedule(name, undefined, { urn });
            case "aws-native:scheduler:ScheduleGroup":
                return new exports.ScheduleGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "scheduler", _module);
