// @pulumi/aws-native/customerprofiles/calculatedAttributeDefinition.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalculatedAttributeDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A calculated attribute definition for Customer Profiles
 */
class CalculatedAttributeDefinition extends pulumi.CustomResource {
    /**
     * Get an existing CalculatedAttributeDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CalculatedAttributeDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CalculatedAttributeDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CalculatedAttributeDefinition.__pulumiType;
    }
    /**
     * Create a CalculatedAttributeDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.attributeDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'attributeDetails'");
            }
            if ((!args || args.calculatedAttributeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'calculatedAttributeName'");
            }
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            if ((!args || args.statistic === undefined) && !opts.urn) {
                throw new Error("Missing required property 'statistic'");
            }
            resourceInputs["attributeDetails"] = args ? args.attributeDetails : undefined;
            resourceInputs["calculatedAttributeName"] = args ? args.calculatedAttributeName : undefined;
            resourceInputs["conditions"] = args ? args.conditions : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["statistic"] = args ? args.statistic : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["lastUpdatedAt"] = undefined /*out*/;
        }
        else {
            resourceInputs["attributeDetails"] = undefined /*out*/;
            resourceInputs["calculatedAttributeName"] = undefined /*out*/;
            resourceInputs["conditions"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["domainName"] = undefined /*out*/;
            resourceInputs["lastUpdatedAt"] = undefined /*out*/;
            resourceInputs["statistic"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["calculatedAttributeName", "domainName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(CalculatedAttributeDefinition.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CalculatedAttributeDefinition = CalculatedAttributeDefinition;
/** @internal */
CalculatedAttributeDefinition.__pulumiType = 'aws-native:customerprofiles:CalculatedAttributeDefinition';
