// @pulumi/aws-native/customerprofiles/segmentDefinition.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SegmentDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A segment definition resource of Amazon Connect Customer Profiles
 */
class SegmentDefinition extends pulumi.CustomResource {
    /**
     * Get an existing SegmentDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SegmentDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SegmentDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SegmentDefinition.__pulumiType;
    }
    /**
     * Create a SegmentDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            if ((!args || args.segmentGroups === undefined) && !opts.urn) {
                throw new Error("Missing required property 'segmentGroups'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["segmentDefinitionName"] = args ? args.segmentDefinitionName : undefined;
            resourceInputs["segmentGroups"] = args ? args.segmentGroups : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["segmentDefinitionArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["domainName"] = undefined /*out*/;
            resourceInputs["segmentDefinitionArn"] = undefined /*out*/;
            resourceInputs["segmentDefinitionName"] = undefined /*out*/;
            resourceInputs["segmentGroups"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["displayName", "domainName", "segmentDefinitionName", "segmentGroups"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(SegmentDefinition.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SegmentDefinition = SegmentDefinition;
/** @internal */
SegmentDefinition.__pulumiType = 'aws-native:customerprofiles:SegmentDefinition';
