// @pulumi/aws-native/ce/anomalyMonitor.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnomalyMonitor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * AWS Cost Anomaly Detection leverages advanced Machine Learning technologies to identify anomalous spend and root causes, so you can quickly take action. You can use Cost Anomaly Detection by creating monitor.
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const anomalyServiceMonitor = new aws_native.ce.AnomalyMonitor("anomalyServiceMonitor", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Dimensional,
 *     monitorDimension: aws_native.ce.AnomalyMonitorMonitorDimension.Service,
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const anomalyServiceMonitor = new aws_native.ce.AnomalyMonitor("anomalyServiceMonitor", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Dimensional,
 *     monitorDimension: aws_native.ce.AnomalyMonitorMonitorDimension.Service,
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithTags = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithTags", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Tags\" : { \"Key\" : \"Tag Key\", \"Values\" : [ \"TagValue1\", \"TagValue2\" ] } }",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithTags = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithTags", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Tags\" : { \"Key\" : \"Tag Key\", \"Values\" : [ \"TagValue1\", \"TagValue2\" ] } }",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithCC = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithCC", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"CostCategories\" : { \"Key\" : \"CostCategoryKey\", \"Values\" : [ \"CostCategoryValue\" ] } }",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithCC = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithCC", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"CostCategories\" : { \"Key\" : \"CostCategoryKey\", \"Values\" : [ \"CostCategoryValue\" ] } }",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithLinkedAccount = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithLinkedAccount", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Dimensions\" : { \"Key\" : \"LINKED_ACCOUNT\", \"Values\" : [ \"123456789012\", \"123456789013\" ] } }",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithLinkedAccount = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithLinkedAccount", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Dimensions\" : { \"Key\" : \"LINKED_ACCOUNT\", \"Values\" : [ \"123456789012\", \"123456789013\" ] } }",
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithLinkedAccount = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithLinkedAccount", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Dimensions\" : { \"Key\" : \"LINKED_ACCOUNT\", \"Values\" : [ \"123456789012\", \"123456789013\" ] } }",
 * });
 * const anomalyServiceMonitor = new aws_native.ce.AnomalyMonitor("anomalyServiceMonitor", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Dimensional,
 *     monitorDimension: aws_native.ce.AnomalyMonitorMonitorDimension.Service,
 * });
 * const anomalySubscription = new aws_native.ce.AnomalySubscription("anomalySubscription", {
 *     subscriptionName: "SubscriptionName",
 *     threshold: 100,
 *     frequency: aws_native.ce.AnomalySubscriptionFrequency.Daily,
 *     monitorArnList: [
 *         customAnomalyMonitorWithLinkedAccount.id,
 *         anomalyServiceMonitor.id,
 *     ],
 *     subscribers: [{
 *         type: aws_native.ce.AnomalySubscriptionSubscriberType.Email,
 *         address: "abc@def.com",
 *     }],
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithLinkedAccount = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithLinkedAccount", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Dimensions\" : { \"Key\" : \"LINKED_ACCOUNT\", \"Values\" : [ \"123456789012\", \"123456789013\" ] } }",
 * });
 * const anomalyServiceMonitor = new aws_native.ce.AnomalyMonitor("anomalyServiceMonitor", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Dimensional,
 *     monitorDimension: aws_native.ce.AnomalyMonitorMonitorDimension.Service,
 * });
 * const anomalySubscription = new aws_native.ce.AnomalySubscription("anomalySubscription", {
 *     subscriptionName: "SubscriptionName",
 *     threshold: 100,
 *     frequency: aws_native.ce.AnomalySubscriptionFrequency.Daily,
 *     monitorArnList: [
 *         customAnomalyMonitorWithLinkedAccount.id,
 *         anomalyServiceMonitor.id,
 *     ],
 *     subscribers: [{
 *         type: aws_native.ce.AnomalySubscriptionSubscriberType.Email,
 *         address: "abc@def.com",
 *     }],
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithLinkedAccount = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithLinkedAccount", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Dimensions\" : { \"Key\" : \"LINKED_ACCOUNT\", \"Values\" : [ \"123456789012\", \"123456789013\" ] } }",
 * });
 * const anomalyServiceMonitor = new aws_native.ce.AnomalyMonitor("anomalyServiceMonitor", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Dimensional,
 *     monitorDimension: aws_native.ce.AnomalyMonitorMonitorDimension.Service,
 * });
 * const anomalySubscription = new aws_native.ce.AnomalySubscription("anomalySubscription", {
 *     subscriptionName: "SubscriptionName",
 *     threshold: 100,
 *     frequency: aws_native.ce.AnomalySubscriptionFrequency.Daily,
 *     monitorArnList: [
 *         customAnomalyMonitorWithLinkedAccount.id,
 *         anomalyServiceMonitor.id,
 *     ],
 *     subscribers: [{
 *         type: aws_native.ce.AnomalySubscriptionSubscriberType.Email,
 *         address: "abc@def.com",
 *     }],
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const customAnomalyMonitorWithLinkedAccount = new aws_native.ce.AnomalyMonitor("customAnomalyMonitorWithLinkedAccount", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Custom,
 *     monitorSpecification: " { \"Dimensions\" : { \"Key\" : \"LINKED_ACCOUNT\", \"Values\" : [ \"123456789012\", \"123456789013\" ] } }",
 * });
 * const anomalyServiceMonitor = new aws_native.ce.AnomalyMonitor("anomalyServiceMonitor", {
 *     monitorName: "MonitorName",
 *     monitorType: aws_native.ce.AnomalyMonitorMonitorType.Dimensional,
 *     monitorDimension: aws_native.ce.AnomalyMonitorMonitorDimension.Service,
 * });
 * const anomalySubscription = new aws_native.ce.AnomalySubscription("anomalySubscription", {
 *     subscriptionName: "SubscriptionName",
 *     threshold: 100,
 *     frequency: aws_native.ce.AnomalySubscriptionFrequency.Daily,
 *     monitorArnList: [
 *         customAnomalyMonitorWithLinkedAccount.id,
 *         anomalyServiceMonitor.id,
 *     ],
 *     subscribers: [{
 *         type: aws_native.ce.AnomalySubscriptionSubscriberType.Email,
 *         address: "abc@def.com",
 *     }],
 * });
 *
 * ```
 */
class AnomalyMonitor extends pulumi.CustomResource {
    /**
     * Get an existing AnomalyMonitor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AnomalyMonitor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AnomalyMonitor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnomalyMonitor.__pulumiType;
    }
    /**
     * Create a AnomalyMonitor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.monitorType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'monitorType'");
            }
            resourceInputs["monitorDimension"] = args ? args.monitorDimension : undefined;
            resourceInputs["monitorName"] = args ? args.monitorName : undefined;
            resourceInputs["monitorSpecification"] = args ? args.monitorSpecification : undefined;
            resourceInputs["monitorType"] = args ? args.monitorType : undefined;
            resourceInputs["resourceTags"] = args ? args.resourceTags : undefined;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["dimensionalValueCount"] = undefined /*out*/;
            resourceInputs["lastEvaluatedDate"] = undefined /*out*/;
            resourceInputs["lastUpdatedDate"] = undefined /*out*/;
            resourceInputs["monitorArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["dimensionalValueCount"] = undefined /*out*/;
            resourceInputs["lastEvaluatedDate"] = undefined /*out*/;
            resourceInputs["lastUpdatedDate"] = undefined /*out*/;
            resourceInputs["monitorArn"] = undefined /*out*/;
            resourceInputs["monitorDimension"] = undefined /*out*/;
            resourceInputs["monitorName"] = undefined /*out*/;
            resourceInputs["monitorSpecification"] = undefined /*out*/;
            resourceInputs["monitorType"] = undefined /*out*/;
            resourceInputs["resourceTags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["monitorDimension", "monitorSpecification", "monitorType", "resourceTags[*]"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(AnomalyMonitor.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AnomalyMonitor = AnomalyMonitor;
/** @internal */
AnomalyMonitor.__pulumiType = 'aws-native:ce:AnomalyMonitor';
