// @pulumi/aws-native/stepfunctions/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StateMachineVersion = exports.StateMachineAlias = exports.StateMachine = exports.getStateMachineVersionOutput = exports.getStateMachineVersion = exports.getStateMachineAliasOutput = exports.getStateMachineAlias = exports.getStateMachineOutput = exports.getStateMachine = exports.getActivityOutput = exports.getActivity = exports.Activity = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Activity = null;
utilities.lazyLoad(exports, ["Activity"], () => require("./activity"));
exports.getActivity = null;
exports.getActivityOutput = null;
utilities.lazyLoad(exports, ["getActivity", "getActivityOutput"], () => require("./getActivity"));
exports.getStateMachine = null;
exports.getStateMachineOutput = null;
utilities.lazyLoad(exports, ["getStateMachine", "getStateMachineOutput"], () => require("./getStateMachine"));
exports.getStateMachineAlias = null;
exports.getStateMachineAliasOutput = null;
utilities.lazyLoad(exports, ["getStateMachineAlias", "getStateMachineAliasOutput"], () => require("./getStateMachineAlias"));
exports.getStateMachineVersion = null;
exports.getStateMachineVersionOutput = null;
utilities.lazyLoad(exports, ["getStateMachineVersion", "getStateMachineVersionOutput"], () => require("./getStateMachineVersion"));
exports.StateMachine = null;
utilities.lazyLoad(exports, ["StateMachine"], () => require("./stateMachine"));
exports.StateMachineAlias = null;
utilities.lazyLoad(exports, ["StateMachineAlias"], () => require("./stateMachineAlias"));
exports.StateMachineVersion = null;
utilities.lazyLoad(exports, ["StateMachineVersion"], () => require("./stateMachineVersion"));
// Export enums:
__exportStar(require("../types/enums/stepfunctions"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:stepfunctions:Activity":
                return new exports.Activity(name, undefined, { urn });
            case "aws-native:stepfunctions:StateMachine":
                return new exports.StateMachine(name, undefined, { urn });
            case "aws-native:stepfunctions:StateMachineAlias":
                return new exports.StateMachineAlias(name, undefined, { urn });
            case "aws-native:stepfunctions:StateMachineVersion":
                return new exports.StateMachineVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "stepfunctions", _module);
