// @pulumi/aws-native/msk/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcConnection = exports.ServerlessCluster = exports.Replicator = exports.getVpcConnectionOutput = exports.getVpcConnection = exports.getServerlessClusterOutput = exports.getServerlessCluster = exports.getReplicatorOutput = exports.getReplicator = exports.getConfigurationOutput = exports.getConfiguration = exports.getClusterPolicyOutput = exports.getClusterPolicy = exports.getClusterOutput = exports.getCluster = exports.getBatchScramSecretOutput = exports.getBatchScramSecret = exports.Configuration = exports.ClusterPolicy = exports.Cluster = exports.BatchScramSecret = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BatchScramSecret = null;
utilities.lazyLoad(exports, ["BatchScramSecret"], () => require("./batchScramSecret"));
exports.Cluster = null;
utilities.lazyLoad(exports, ["Cluster"], () => require("./cluster"));
exports.ClusterPolicy = null;
utilities.lazyLoad(exports, ["ClusterPolicy"], () => require("./clusterPolicy"));
exports.Configuration = null;
utilities.lazyLoad(exports, ["Configuration"], () => require("./configuration"));
exports.getBatchScramSecret = null;
exports.getBatchScramSecretOutput = null;
utilities.lazyLoad(exports, ["getBatchScramSecret", "getBatchScramSecretOutput"], () => require("./getBatchScramSecret"));
exports.getCluster = null;
exports.getClusterOutput = null;
utilities.lazyLoad(exports, ["getCluster", "getClusterOutput"], () => require("./getCluster"));
exports.getClusterPolicy = null;
exports.getClusterPolicyOutput = null;
utilities.lazyLoad(exports, ["getClusterPolicy", "getClusterPolicyOutput"], () => require("./getClusterPolicy"));
exports.getConfiguration = null;
exports.getConfigurationOutput = null;
utilities.lazyLoad(exports, ["getConfiguration", "getConfigurationOutput"], () => require("./getConfiguration"));
exports.getReplicator = null;
exports.getReplicatorOutput = null;
utilities.lazyLoad(exports, ["getReplicator", "getReplicatorOutput"], () => require("./getReplicator"));
exports.getServerlessCluster = null;
exports.getServerlessClusterOutput = null;
utilities.lazyLoad(exports, ["getServerlessCluster", "getServerlessClusterOutput"], () => require("./getServerlessCluster"));
exports.getVpcConnection = null;
exports.getVpcConnectionOutput = null;
utilities.lazyLoad(exports, ["getVpcConnection", "getVpcConnectionOutput"], () => require("./getVpcConnection"));
exports.Replicator = null;
utilities.lazyLoad(exports, ["Replicator"], () => require("./replicator"));
exports.ServerlessCluster = null;
utilities.lazyLoad(exports, ["ServerlessCluster"], () => require("./serverlessCluster"));
exports.VpcConnection = null;
utilities.lazyLoad(exports, ["VpcConnection"], () => require("./vpcConnection"));
// Export enums:
__exportStar(require("../types/enums/msk"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:msk:BatchScramSecret":
                return new exports.BatchScramSecret(name, undefined, { urn });
            case "aws-native:msk:Cluster":
                return new exports.Cluster(name, undefined, { urn });
            case "aws-native:msk:ClusterPolicy":
                return new exports.ClusterPolicy(name, undefined, { urn });
            case "aws-native:msk:Configuration":
                return new exports.Configuration(name, undefined, { urn });
            case "aws-native:msk:Replicator":
                return new exports.Replicator(name, undefined, { urn });
            case "aws-native:msk:ServerlessCluster":
                return new exports.ServerlessCluster(name, undefined, { urn });
            case "aws-native:msk:VpcConnection":
                return new exports.VpcConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "msk", _module);
