// @pulumi/aws-native/types/enums/wisdom/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageTemplateChannelSubtype = exports.KnowledgeBaseWebCrawlerConfigurationScope = exports.KnowledgeBaseVectorIngestionConfigurationParsingConfigurationPropertiesParsingStrategy = exports.KnowledgeBaseVectorIngestionConfigurationChunkingConfigurationPropertiesChunkingStrategy = exports.KnowledgeBaseType = exports.AssistantType = exports.AssistantAssociationAssociationType = exports.AiPromptAiPromptType = exports.AiPromptAiPromptTemplateType = exports.AiPromptAiPromptApiFormat = exports.AiGuardrailGuardrailTopicType = exports.AiGuardrailGuardrailSensitiveInformationAction = exports.AiGuardrailGuardrailPiiEntityType = exports.AiGuardrailGuardrailManagedWordsType = exports.AiGuardrailGuardrailFilterStrength = exports.AiGuardrailGuardrailContextualGroundingFilterType = exports.AiGuardrailGuardrailContentFilterType = exports.AiAgentKnowledgeBaseSearchType = exports.AiAgentAiAgentType = exports.AiAgentAiAgentAssociationConfigurationType = void 0;
exports.AiAgentAiAgentAssociationConfigurationType = {
    KnowledgeBase: "KNOWLEDGE_BASE",
};
exports.AiAgentAiAgentType = {
    ManualSearch: "MANUAL_SEARCH",
    AnswerRecommendation: "ANSWER_RECOMMENDATION",
    SelfService: "SELF_SERVICE",
};
exports.AiAgentKnowledgeBaseSearchType = {
    Hybrid: "HYBRID",
    Semantic: "SEMANTIC",
};
exports.AiGuardrailGuardrailContentFilterType = {
    Sexual: "SEXUAL",
    Violence: "VIOLENCE",
    Hate: "HATE",
    Insults: "INSULTS",
    Misconduct: "MISCONDUCT",
    PromptAttack: "PROMPT_ATTACK",
};
exports.AiGuardrailGuardrailContextualGroundingFilterType = {
    Grounding: "GROUNDING",
    Relevance: "RELEVANCE",
};
exports.AiGuardrailGuardrailFilterStrength = {
    None: "NONE",
    Low: "LOW",
    Medium: "MEDIUM",
    High: "HIGH",
};
exports.AiGuardrailGuardrailManagedWordsType = {
    Profanity: "PROFANITY",
};
exports.AiGuardrailGuardrailPiiEntityType = {
    Address: "ADDRESS",
    Age: "AGE",
    AwsAccessKey: "AWS_ACCESS_KEY",
    AwsSecretKey: "AWS_SECRET_KEY",
    CaHealthNumber: "CA_HEALTH_NUMBER",
    CaSocialInsuranceNumber: "CA_SOCIAL_INSURANCE_NUMBER",
    CreditDebitCardCvv: "CREDIT_DEBIT_CARD_CVV",
    CreditDebitCardExpiry: "CREDIT_DEBIT_CARD_EXPIRY",
    CreditDebitCardNumber: "CREDIT_DEBIT_CARD_NUMBER",
    DriverId: "DRIVER_ID",
    Email: "EMAIL",
    InternationalBankAccountNumber: "INTERNATIONAL_BANK_ACCOUNT_NUMBER",
    IpAddress: "IP_ADDRESS",
    LicensePlate: "LICENSE_PLATE",
    MacAddress: "MAC_ADDRESS",
    Name: "NAME",
    Password: "PASSWORD",
    Phone: "PHONE",
    Pin: "PIN",
    SwiftCode: "SWIFT_CODE",
    UkNationalHealthServiceNumber: "UK_NATIONAL_HEALTH_SERVICE_NUMBER",
    UkNationalInsuranceNumber: "UK_NATIONAL_INSURANCE_NUMBER",
    UkUniqueTaxpayerReferenceNumber: "UK_UNIQUE_TAXPAYER_REFERENCE_NUMBER",
    Url: "URL",
    Username: "USERNAME",
    UsBankAccountNumber: "US_BANK_ACCOUNT_NUMBER",
    UsBankRoutingNumber: "US_BANK_ROUTING_NUMBER",
    UsIndividualTaxIdentificationNumber: "US_INDIVIDUAL_TAX_IDENTIFICATION_NUMBER",
    UsPassportNumber: "US_PASSPORT_NUMBER",
    UsSocialSecurityNumber: "US_SOCIAL_SECURITY_NUMBER",
    VehicleIdentificationNumber: "VEHICLE_IDENTIFICATION_NUMBER",
};
exports.AiGuardrailGuardrailSensitiveInformationAction = {
    Block: "BLOCK",
    Anonymize: "ANONYMIZE",
};
exports.AiGuardrailGuardrailTopicType = {
    Deny: "DENY",
};
exports.AiPromptAiPromptApiFormat = {
    AnthropicClaudeMessages: "ANTHROPIC_CLAUDE_MESSAGES",
    AnthropicClaudeTextCompletions: "ANTHROPIC_CLAUDE_TEXT_COMPLETIONS",
};
exports.AiPromptAiPromptTemplateType = {
    Text: "TEXT",
};
exports.AiPromptAiPromptType = {
    AnswerGeneration: "ANSWER_GENERATION",
    IntentLabelingGeneration: "INTENT_LABELING_GENERATION",
    QueryReformulation: "QUERY_REFORMULATION",
    SelfServicePreProcessing: "SELF_SERVICE_PRE_PROCESSING",
    SelfServiceAnswerGeneration: "SELF_SERVICE_ANSWER_GENERATION",
};
exports.AssistantAssociationAssociationType = {
    KnowledgeBase: "KNOWLEDGE_BASE",
};
exports.AssistantType = {
    Agent: "AGENT",
};
exports.KnowledgeBaseType = {
    External: "EXTERNAL",
    Custom: "CUSTOM",
    MessageTemplates: "MESSAGE_TEMPLATES",
    Managed: "MANAGED",
};
exports.KnowledgeBaseVectorIngestionConfigurationChunkingConfigurationPropertiesChunkingStrategy = {
    FixedSize: "FIXED_SIZE",
    None: "NONE",
    Hierarchical: "HIERARCHICAL",
    Semantic: "SEMANTIC",
};
exports.KnowledgeBaseVectorIngestionConfigurationParsingConfigurationPropertiesParsingStrategy = {
    BedrockFoundationModel: "BEDROCK_FOUNDATION_MODEL",
};
exports.KnowledgeBaseWebCrawlerConfigurationScope = {
    HostOnly: "HOST_ONLY",
    Subdomains: "SUBDOMAINS",
};
exports.MessageTemplateChannelSubtype = {
    Email: "EMAIL",
    Sms: "SMS",
};
