// @pulumi/aws-native/robomaker/fleet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fleet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * AWS::RoboMaker::Fleet resource creates an AWS RoboMaker fleet. Fleets contain robots and can receive deployments.
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const basicFleet = new aws_native.robomaker.Fleet("basicFleet", {
 *     name: "MyFleet",
 *     tags: {
 *         name: "BasicFleet",
 *         type: "CFN",
 *     },
 * });
 * export const fleet = "BasicFleet";
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const basicFleet = new aws_native.robomaker.Fleet("basicFleet", {
 *     name: "MyFleet",
 *     tags: {
 *         name: "BasicFleet",
 *         type: "CFN",
 *     },
 * });
 * export const fleet = basicFleet.id;
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const basicFleet = new aws_native.robomaker.Fleet("basicFleet", {name: "MyFleet"});
 * const basicRobot = new aws_native.robomaker.Robot("basicRobot", {
 *     name: "MyRobot",
 *     greengrassGroupId: "51229986-abdc-4ca6-94f8-04735a0c9f07",
 *     architecture: aws_native.robomaker.RobotArchitecture.Armhf,
 *     fleet: basicFleet.arn,
 *     tags: {
 *         name: "BasicRobot",
 *         type: "CFN",
 *     },
 * });
 * export const robot = "BasicRobot";
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const basicFleet = new aws_native.robomaker.Fleet("basicFleet", {name: "MyFleet"});
 * const basicRobot = new aws_native.robomaker.Robot("basicRobot", {
 *     name: "MyRobot",
 *     greengrassGroupId: "51229986-abdc-4ca6-94f8-04735a0c9f07",
 *     architecture: aws_native.robomaker.RobotArchitecture.Armhf,
 *     fleet: basicFleet.arn,
 *     tags: {
 *         name: "BasicRobot",
 *         type: "CFN",
 *     },
 * });
 * export const robot = basicRobot.id;
 *
 * ```
 */
class Fleet extends pulumi.CustomResource {
    /**
     * Get an existing Fleet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Fleet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Fleet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Fleet.__pulumiType;
    }
    /**
     * Create a Fleet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["name"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(Fleet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Fleet = Fleet;
/** @internal */
Fleet.__pulumiType = 'aws-native:robomaker:Fleet';
