// @pulumi/aws-native/appconfig/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostedConfigurationVersion = exports.getHostedConfigurationVersionOutput = exports.getHostedConfigurationVersion = exports.getExtensionAssociationOutput = exports.getExtensionAssociation = exports.getExtensionOutput = exports.getExtension = exports.getEnvironmentOutput = exports.getEnvironment = exports.getDeploymentStrategyOutput = exports.getDeploymentStrategy = exports.getConfigurationProfileOutput = exports.getConfigurationProfile = exports.getApplicationOutput = exports.getApplication = exports.ExtensionAssociation = exports.Extension = exports.Environment = exports.DeploymentStrategy = exports.ConfigurationProfile = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ConfigurationProfile = null;
utilities.lazyLoad(exports, ["ConfigurationProfile"], () => require("./configurationProfile"));
exports.DeploymentStrategy = null;
utilities.lazyLoad(exports, ["DeploymentStrategy"], () => require("./deploymentStrategy"));
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.Extension = null;
utilities.lazyLoad(exports, ["Extension"], () => require("./extension"));
exports.ExtensionAssociation = null;
utilities.lazyLoad(exports, ["ExtensionAssociation"], () => require("./extensionAssociation"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getConfigurationProfile = null;
exports.getConfigurationProfileOutput = null;
utilities.lazyLoad(exports, ["getConfigurationProfile", "getConfigurationProfileOutput"], () => require("./getConfigurationProfile"));
exports.getDeploymentStrategy = null;
exports.getDeploymentStrategyOutput = null;
utilities.lazyLoad(exports, ["getDeploymentStrategy", "getDeploymentStrategyOutput"], () => require("./getDeploymentStrategy"));
exports.getEnvironment = null;
exports.getEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getEnvironment", "getEnvironmentOutput"], () => require("./getEnvironment"));
exports.getExtension = null;
exports.getExtensionOutput = null;
utilities.lazyLoad(exports, ["getExtension", "getExtensionOutput"], () => require("./getExtension"));
exports.getExtensionAssociation = null;
exports.getExtensionAssociationOutput = null;
utilities.lazyLoad(exports, ["getExtensionAssociation", "getExtensionAssociationOutput"], () => require("./getExtensionAssociation"));
exports.getHostedConfigurationVersion = null;
exports.getHostedConfigurationVersionOutput = null;
utilities.lazyLoad(exports, ["getHostedConfigurationVersion", "getHostedConfigurationVersionOutput"], () => require("./getHostedConfigurationVersion"));
exports.HostedConfigurationVersion = null;
utilities.lazyLoad(exports, ["HostedConfigurationVersion"], () => require("./hostedConfigurationVersion"));
// Export enums:
__exportStar(require("../types/enums/appconfig"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:appconfig:Application":
                return new exports.Application(name, undefined, { urn });
            case "aws-native:appconfig:ConfigurationProfile":
                return new exports.ConfigurationProfile(name, undefined, { urn });
            case "aws-native:appconfig:DeploymentStrategy":
                return new exports.DeploymentStrategy(name, undefined, { urn });
            case "aws-native:appconfig:Environment":
                return new exports.Environment(name, undefined, { urn });
            case "aws-native:appconfig:Extension":
                return new exports.Extension(name, undefined, { urn });
            case "aws-native:appconfig:ExtensionAssociation":
                return new exports.ExtensionAssociation(name, undefined, { urn });
            case "aws-native:appconfig:HostedConfigurationVersion":
                return new exports.HostedConfigurationVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "appconfig", _module);
