// @pulumi/aws-native/xray/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SamplingRule = exports.ResourcePolicy = exports.Group = exports.getSamplingRuleOutput = exports.getSamplingRule = exports.getResourcePolicyOutput = exports.getResourcePolicy = exports.getGroupOutput = exports.getGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.getSamplingRule = null;
exports.getSamplingRuleOutput = null;
utilities.lazyLoad(exports, ["getSamplingRule", "getSamplingRuleOutput"], () => require("./getSamplingRule"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.SamplingRule = null;
utilities.lazyLoad(exports, ["SamplingRule"], () => require("./samplingRule"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:xray:Group":
                return new exports.Group(name, undefined, { urn });
            case "aws-native:xray:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws-native:xray:SamplingRule":
                return new exports.SamplingRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "xray", _module);
