// @pulumi/aws-native/types/enums/mediatailor/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VodSourceType = exports.SourceLocationAccessType = exports.PlaybackConfigurationDashConfigurationOriginManifestType = exports.PlaybackConfigurationAvailSuppressionMode = exports.PlaybackConfigurationAvailSuppressionFillPolicy = exports.LiveSourceType = exports.ChannelTier = exports.ChannelPlaybackMode = exports.ChannelLogType = exports.ChannelAdMarkupType = void 0;
exports.ChannelAdMarkupType = {
    Daterange: "DATERANGE",
    Scte35Enhanced: "SCTE35_ENHANCED",
};
exports.ChannelLogType = {
    AsRun: "AS_RUN",
};
exports.ChannelPlaybackMode = {
    Loop: "LOOP",
    Linear: "LINEAR",
};
exports.ChannelTier = {
    Basic: "BASIC",
    Standard: "STANDARD",
};
exports.LiveSourceType = {
    Dash: "DASH",
    Hls: "HLS",
};
exports.PlaybackConfigurationAvailSuppressionFillPolicy = {
    PartialAvail: "PARTIAL_AVAIL",
    FullAvailOnly: "FULL_AVAIL_ONLY",
};
exports.PlaybackConfigurationAvailSuppressionMode = {
    Off: "OFF",
    BehindLiveEdge: "BEHIND_LIVE_EDGE",
    AfterLiveEdge: "AFTER_LIVE_EDGE",
};
exports.PlaybackConfigurationDashConfigurationOriginManifestType = {
    SinglePeriod: "SINGLE_PERIOD",
    MultiPeriod: "MULTI_PERIOD",
};
exports.SourceLocationAccessType = {
    S3Sigv4: "S3_SIGV4",
    SecretsManagerAccessToken: "SECRETS_MANAGER_ACCESS_TOKEN",
    AutodetectSigv4: "AUTODETECT_SIGV4",
};
exports.VodSourceType = {
    Dash: "DASH",
    Hls: "HLS",
};
