// @pulumi/aws-native/types/enums/comprehend/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlywheelTaskConfigLanguageCode = exports.FlywheelModelType = exports.FlywheelDocumentClassificationConfigMode = exports.DocumentClassifierMode = exports.DocumentClassifierLanguageCode = exports.DocumentClassifierInputDataConfigDocumentType = exports.DocumentClassifierInputDataConfigDataFormat = exports.DocumentClassifierDocumentReaderConfigFeatureTypesItem = exports.DocumentClassifierDocumentReaderConfigDocumentReadMode = exports.DocumentClassifierDocumentReaderConfigDocumentReadAction = exports.DocumentClassifierAugmentedManifestsListItemSplit = void 0;
exports.DocumentClassifierAugmentedManifestsListItemSplit = {
    Train: "TRAIN",
    Test: "TEST",
};
exports.DocumentClassifierDocumentReaderConfigDocumentReadAction = {
    TextractDetectDocumentText: "TEXTRACT_DETECT_DOCUMENT_TEXT",
    TextractAnalyzeDocument: "TEXTRACT_ANALYZE_DOCUMENT",
};
exports.DocumentClassifierDocumentReaderConfigDocumentReadMode = {
    ServiceDefault: "SERVICE_DEFAULT",
    ForceDocumentReadAction: "FORCE_DOCUMENT_READ_ACTION",
};
exports.DocumentClassifierDocumentReaderConfigFeatureTypesItem = {
    Tables: "TABLES",
    Forms: "FORMS",
};
exports.DocumentClassifierInputDataConfigDataFormat = {
    ComprehendCsv: "COMPREHEND_CSV",
    AugmentedManifest: "AUGMENTED_MANIFEST",
};
exports.DocumentClassifierInputDataConfigDocumentType = {
    PlainTextDocument: "PLAIN_TEXT_DOCUMENT",
    SemiStructuredDocument: "SEMI_STRUCTURED_DOCUMENT",
};
exports.DocumentClassifierLanguageCode = {
    En: "en",
    Es: "es",
    Fr: "fr",
    It: "it",
    De: "de",
    Pt: "pt",
};
exports.DocumentClassifierMode = {
    MultiClass: "MULTI_CLASS",
    MultiLabel: "MULTI_LABEL",
};
exports.FlywheelDocumentClassificationConfigMode = {
    MultiClass: "MULTI_CLASS",
    MultiLabel: "MULTI_LABEL",
};
exports.FlywheelModelType = {
    DocumentClassifier: "DOCUMENT_CLASSIFIER",
    EntityRecognizer: "ENTITY_RECOGNIZER",
};
exports.FlywheelTaskConfigLanguageCode = {
    En: "en",
    Es: "es",
    Fr: "fr",
    It: "it",
    De: "de",
    Pt: "pt",
};
