// @pulumi/aws-native/types/enums/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ecs = exports.ecr = exports.ec2 = exports.dynamodb = exports.dms = exports.devopsguru = exports.devicefarm = exports.deadline = exports.datazone = exports.datasync = exports.databrew = exports.customerprofiles = exports.controltower = exports.connectcampaignsv2 = exports.connect = exports.comprehend = exports.cognito = exports.codestarnotifications = exports.codestarconnections = exports.codepipeline = exports.codegurureviewer = exports.codeguruprofiler = exports.codebuild = exports.codeartifact = exports.cloudtrail = exports.cloudfront = exports.cloudformation = exports.cleanroomsml = exports.cleanrooms = exports.chatbot = exports.ce = exports.cassandra = exports.budgets = exports.bedrock = exports.batch = exports.backup = exports.b2bi = exports.autoscaling = exports.auditmanager = exports.athena = exports.arczonalshift = exports.appsync = exports.apprunner = exports.applicationsignals = exports.applicationinsights = exports.appflow = exports.appconfig = exports.apigateway = exports.amplifyuibuilder = exports.amplify = void 0;
exports.m2 = exports.lookoutmetrics = exports.logs = exports.location = exports.lightsail = exports.lex = exports.launchwizard = exports.lambda = exports.lakeformation = exports.kms = exports.kinesisvideo = exports.kinesisfirehose = exports.kinesisanalyticsv2 = exports.kinesis = exports.kendra = exports.kafkaconnect = exports.ivschat = exports.ivs = exports.iotwireless = exports.iottwinmaker = exports.iotsitewise = exports.iotevents = exports.iotanalytics = exports.iot = exports.internetmonitor = exports.inspectorv2 = exports.imagebuilder = exports.healthlake = exports.healthimaging = exports.guardduty = exports.groundstation = exports.greengrassv2 = exports.grafana = exports.glue = exports.globalaccelerator = exports.gamelift = exports.fsx = exports.frauddetector = exports.forecast = exports.fms = exports.fis = exports.finspace = exports.evidently = exports.events = exports.entityresolution = exports.emrserverless = exports.emr = exports.elasticache = exports.eks = exports.efs = void 0;
exports.servicecatalog = exports.securitylake = exports.securityhub = exports.scheduler = exports.sagemaker = exports.s3tables = exports.s3outposts = exports.s3express = exports.s3 = exports.rum = exports.route53resolver = exports.route53recoverycontrol = exports.route53 = exports.rolesanywhere = exports.robomaker = exports.resourcegroups = exports.resourceexplorer2 = exports.resiliencehub = exports.refactorspaces = exports.redshiftserverless = exports.redshift = exports.rds = exports.rbin = exports.quicksight = exports.qbusiness = exports.proton = exports.pipes = exports.pinpoint = exports.personalize = exports.pcs = exports.pcaconnectorscep = exports.pcaconnectorad = exports.paymentcryptography = exports.panorama = exports.osis = exports.organizations = exports.opensearchservice = exports.opensearchserverless = exports.omics = exports.oam = exports.networkfirewall = exports.mwaa = exports.msk = exports.memorydb = exports.mediatailor = exports.mediapackagev2 = exports.mediapackage = exports.medialive = exports.mediaconnect = exports.macie = void 0;
exports.Region = exports.workspacesweb = exports.workspacesthinclient = exports.workspaces = exports.wisdom = exports.wafv2 = exports.vpclattice = exports.verifiedpermissions = exports.transfer = exports.timestream = exports.systemsmanagersap = exports.synthetics = exports.supportapp = exports.stepfunctions = exports.sso = exports.ssmquicksetup = exports.ssmincidents = exports.ssmcontacts = exports.ssm = exports.sns = exports.signer = exports.shield = exports.ses = exports.servicecatalogappregistry = void 0;
// Export sub-modules:
const amplify = require("./amplify");
exports.amplify = amplify;
const amplifyuibuilder = require("./amplifyuibuilder");
exports.amplifyuibuilder = amplifyuibuilder;
const apigateway = require("./apigateway");
exports.apigateway = apigateway;
const appconfig = require("./appconfig");
exports.appconfig = appconfig;
const appflow = require("./appflow");
exports.appflow = appflow;
const applicationinsights = require("./applicationinsights");
exports.applicationinsights = applicationinsights;
const applicationsignals = require("./applicationsignals");
exports.applicationsignals = applicationsignals;
const apprunner = require("./apprunner");
exports.apprunner = apprunner;
const appsync = require("./appsync");
exports.appsync = appsync;
const arczonalshift = require("./arczonalshift");
exports.arczonalshift = arczonalshift;
const athena = require("./athena");
exports.athena = athena;
const auditmanager = require("./auditmanager");
exports.auditmanager = auditmanager;
const autoscaling = require("./autoscaling");
exports.autoscaling = autoscaling;
const b2bi = require("./b2bi");
exports.b2bi = b2bi;
const backup = require("./backup");
exports.backup = backup;
const batch = require("./batch");
exports.batch = batch;
const bedrock = require("./bedrock");
exports.bedrock = bedrock;
const budgets = require("./budgets");
exports.budgets = budgets;
const cassandra = require("./cassandra");
exports.cassandra = cassandra;
const ce = require("./ce");
exports.ce = ce;
const chatbot = require("./chatbot");
exports.chatbot = chatbot;
const cleanrooms = require("./cleanrooms");
exports.cleanrooms = cleanrooms;
const cleanroomsml = require("./cleanroomsml");
exports.cleanroomsml = cleanroomsml;
const cloudformation = require("./cloudformation");
exports.cloudformation = cloudformation;
const cloudfront = require("./cloudfront");
exports.cloudfront = cloudfront;
const cloudtrail = require("./cloudtrail");
exports.cloudtrail = cloudtrail;
const codeartifact = require("./codeartifact");
exports.codeartifact = codeartifact;
const codebuild = require("./codebuild");
exports.codebuild = codebuild;
const codeguruprofiler = require("./codeguruprofiler");
exports.codeguruprofiler = codeguruprofiler;
const codegurureviewer = require("./codegurureviewer");
exports.codegurureviewer = codegurureviewer;
const codepipeline = require("./codepipeline");
exports.codepipeline = codepipeline;
const codestarconnections = require("./codestarconnections");
exports.codestarconnections = codestarconnections;
const codestarnotifications = require("./codestarnotifications");
exports.codestarnotifications = codestarnotifications;
const cognito = require("./cognito");
exports.cognito = cognito;
const comprehend = require("./comprehend");
exports.comprehend = comprehend;
const connect = require("./connect");
exports.connect = connect;
const connectcampaignsv2 = require("./connectcampaignsv2");
exports.connectcampaignsv2 = connectcampaignsv2;
const controltower = require("./controltower");
exports.controltower = controltower;
const customerprofiles = require("./customerprofiles");
exports.customerprofiles = customerprofiles;
const databrew = require("./databrew");
exports.databrew = databrew;
const datasync = require("./datasync");
exports.datasync = datasync;
const datazone = require("./datazone");
exports.datazone = datazone;
const deadline = require("./deadline");
exports.deadline = deadline;
const devicefarm = require("./devicefarm");
exports.devicefarm = devicefarm;
const devopsguru = require("./devopsguru");
exports.devopsguru = devopsguru;
const dms = require("./dms");
exports.dms = dms;
const dynamodb = require("./dynamodb");
exports.dynamodb = dynamodb;
const ec2 = require("./ec2");
exports.ec2 = ec2;
const ecr = require("./ecr");
exports.ecr = ecr;
const ecs = require("./ecs");
exports.ecs = ecs;
const efs = require("./efs");
exports.efs = efs;
const eks = require("./eks");
exports.eks = eks;
const elasticache = require("./elasticache");
exports.elasticache = elasticache;
const emr = require("./emr");
exports.emr = emr;
const emrserverless = require("./emrserverless");
exports.emrserverless = emrserverless;
const entityresolution = require("./entityresolution");
exports.entityresolution = entityresolution;
const events = require("./events");
exports.events = events;
const evidently = require("./evidently");
exports.evidently = evidently;
const finspace = require("./finspace");
exports.finspace = finspace;
const fis = require("./fis");
exports.fis = fis;
const fms = require("./fms");
exports.fms = fms;
const forecast = require("./forecast");
exports.forecast = forecast;
const frauddetector = require("./frauddetector");
exports.frauddetector = frauddetector;
const fsx = require("./fsx");
exports.fsx = fsx;
const gamelift = require("./gamelift");
exports.gamelift = gamelift;
const globalaccelerator = require("./globalaccelerator");
exports.globalaccelerator = globalaccelerator;
const glue = require("./glue");
exports.glue = glue;
const grafana = require("./grafana");
exports.grafana = grafana;
const greengrassv2 = require("./greengrassv2");
exports.greengrassv2 = greengrassv2;
const groundstation = require("./groundstation");
exports.groundstation = groundstation;
const guardduty = require("./guardduty");
exports.guardduty = guardduty;
const healthimaging = require("./healthimaging");
exports.healthimaging = healthimaging;
const healthlake = require("./healthlake");
exports.healthlake = healthlake;
const imagebuilder = require("./imagebuilder");
exports.imagebuilder = imagebuilder;
const inspectorv2 = require("./inspectorv2");
exports.inspectorv2 = inspectorv2;
const internetmonitor = require("./internetmonitor");
exports.internetmonitor = internetmonitor;
const iot = require("./iot");
exports.iot = iot;
const iotanalytics = require("./iotanalytics");
exports.iotanalytics = iotanalytics;
const iotevents = require("./iotevents");
exports.iotevents = iotevents;
const iotsitewise = require("./iotsitewise");
exports.iotsitewise = iotsitewise;
const iottwinmaker = require("./iottwinmaker");
exports.iottwinmaker = iottwinmaker;
const iotwireless = require("./iotwireless");
exports.iotwireless = iotwireless;
const ivs = require("./ivs");
exports.ivs = ivs;
const ivschat = require("./ivschat");
exports.ivschat = ivschat;
const kafkaconnect = require("./kafkaconnect");
exports.kafkaconnect = kafkaconnect;
const kendra = require("./kendra");
exports.kendra = kendra;
const kinesis = require("./kinesis");
exports.kinesis = kinesis;
const kinesisanalyticsv2 = require("./kinesisanalyticsv2");
exports.kinesisanalyticsv2 = kinesisanalyticsv2;
const kinesisfirehose = require("./kinesisfirehose");
exports.kinesisfirehose = kinesisfirehose;
const kinesisvideo = require("./kinesisvideo");
exports.kinesisvideo = kinesisvideo;
const kms = require("./kms");
exports.kms = kms;
const lakeformation = require("./lakeformation");
exports.lakeformation = lakeformation;
const lambda = require("./lambda");
exports.lambda = lambda;
const launchwizard = require("./launchwizard");
exports.launchwizard = launchwizard;
const lex = require("./lex");
exports.lex = lex;
const lightsail = require("./lightsail");
exports.lightsail = lightsail;
const location = require("./location");
exports.location = location;
const logs = require("./logs");
exports.logs = logs;
const lookoutmetrics = require("./lookoutmetrics");
exports.lookoutmetrics = lookoutmetrics;
const m2 = require("./m2");
exports.m2 = m2;
const macie = require("./macie");
exports.macie = macie;
const mediaconnect = require("./mediaconnect");
exports.mediaconnect = mediaconnect;
const medialive = require("./medialive");
exports.medialive = medialive;
const mediapackage = require("./mediapackage");
exports.mediapackage = mediapackage;
const mediapackagev2 = require("./mediapackagev2");
exports.mediapackagev2 = mediapackagev2;
const mediatailor = require("./mediatailor");
exports.mediatailor = mediatailor;
const memorydb = require("./memorydb");
exports.memorydb = memorydb;
const msk = require("./msk");
exports.msk = msk;
const mwaa = require("./mwaa");
exports.mwaa = mwaa;
const networkfirewall = require("./networkfirewall");
exports.networkfirewall = networkfirewall;
const oam = require("./oam");
exports.oam = oam;
const omics = require("./omics");
exports.omics = omics;
const opensearchserverless = require("./opensearchserverless");
exports.opensearchserverless = opensearchserverless;
const opensearchservice = require("./opensearchservice");
exports.opensearchservice = opensearchservice;
const organizations = require("./organizations");
exports.organizations = organizations;
const osis = require("./osis");
exports.osis = osis;
const panorama = require("./panorama");
exports.panorama = panorama;
const paymentcryptography = require("./paymentcryptography");
exports.paymentcryptography = paymentcryptography;
const pcaconnectorad = require("./pcaconnectorad");
exports.pcaconnectorad = pcaconnectorad;
const pcaconnectorscep = require("./pcaconnectorscep");
exports.pcaconnectorscep = pcaconnectorscep;
const pcs = require("./pcs");
exports.pcs = pcs;
const personalize = require("./personalize");
exports.personalize = personalize;
const pinpoint = require("./pinpoint");
exports.pinpoint = pinpoint;
const pipes = require("./pipes");
exports.pipes = pipes;
const proton = require("./proton");
exports.proton = proton;
const qbusiness = require("./qbusiness");
exports.qbusiness = qbusiness;
const quicksight = require("./quicksight");
exports.quicksight = quicksight;
const rbin = require("./rbin");
exports.rbin = rbin;
const rds = require("./rds");
exports.rds = rds;
const redshift = require("./redshift");
exports.redshift = redshift;
const redshiftserverless = require("./redshiftserverless");
exports.redshiftserverless = redshiftserverless;
const refactorspaces = require("./refactorspaces");
exports.refactorspaces = refactorspaces;
const resiliencehub = require("./resiliencehub");
exports.resiliencehub = resiliencehub;
const resourceexplorer2 = require("./resourceexplorer2");
exports.resourceexplorer2 = resourceexplorer2;
const resourcegroups = require("./resourcegroups");
exports.resourcegroups = resourcegroups;
const robomaker = require("./robomaker");
exports.robomaker = robomaker;
const rolesanywhere = require("./rolesanywhere");
exports.rolesanywhere = rolesanywhere;
const route53 = require("./route53");
exports.route53 = route53;
const route53recoverycontrol = require("./route53recoverycontrol");
exports.route53recoverycontrol = route53recoverycontrol;
const route53resolver = require("./route53resolver");
exports.route53resolver = route53resolver;
const rum = require("./rum");
exports.rum = rum;
const s3 = require("./s3");
exports.s3 = s3;
const s3express = require("./s3express");
exports.s3express = s3express;
const s3outposts = require("./s3outposts");
exports.s3outposts = s3outposts;
const s3tables = require("./s3tables");
exports.s3tables = s3tables;
const sagemaker = require("./sagemaker");
exports.sagemaker = sagemaker;
const scheduler = require("./scheduler");
exports.scheduler = scheduler;
const securityhub = require("./securityhub");
exports.securityhub = securityhub;
const securitylake = require("./securitylake");
exports.securitylake = securitylake;
const servicecatalog = require("./servicecatalog");
exports.servicecatalog = servicecatalog;
const servicecatalogappregistry = require("./servicecatalogappregistry");
exports.servicecatalogappregistry = servicecatalogappregistry;
const ses = require("./ses");
exports.ses = ses;
const shield = require("./shield");
exports.shield = shield;
const signer = require("./signer");
exports.signer = signer;
const sns = require("./sns");
exports.sns = sns;
const ssm = require("./ssm");
exports.ssm = ssm;
const ssmcontacts = require("./ssmcontacts");
exports.ssmcontacts = ssmcontacts;
const ssmincidents = require("./ssmincidents");
exports.ssmincidents = ssmincidents;
const ssmquicksetup = require("./ssmquicksetup");
exports.ssmquicksetup = ssmquicksetup;
const sso = require("./sso");
exports.sso = sso;
const stepfunctions = require("./stepfunctions");
exports.stepfunctions = stepfunctions;
const supportapp = require("./supportapp");
exports.supportapp = supportapp;
const synthetics = require("./synthetics");
exports.synthetics = synthetics;
const systemsmanagersap = require("./systemsmanagersap");
exports.systemsmanagersap = systemsmanagersap;
const timestream = require("./timestream");
exports.timestream = timestream;
const transfer = require("./transfer");
exports.transfer = transfer;
const verifiedpermissions = require("./verifiedpermissions");
exports.verifiedpermissions = verifiedpermissions;
const vpclattice = require("./vpclattice");
exports.vpclattice = vpclattice;
const wafv2 = require("./wafv2");
exports.wafv2 = wafv2;
const wisdom = require("./wisdom");
exports.wisdom = wisdom;
const workspaces = require("./workspaces");
exports.workspaces = workspaces;
const workspacesthinclient = require("./workspacesthinclient");
exports.workspacesthinclient = workspacesthinclient;
const workspacesweb = require("./workspacesweb");
exports.workspacesweb = workspacesweb;
exports.Region = {
    /**
     * Africa (Cape Town)
     */
    AfSouth1: "af-south-1",
    /**
     * Asia Pacific (Hong Kong)
     */
    ApEast1: "ap-east-1",
    /**
     * Asia Pacific (Tokyo)
     */
    ApNortheast1: "ap-northeast-1",
    /**
     * Asia Pacific (Seoul)
     */
    ApNortheast2: "ap-northeast-2",
    /**
     * Asia Pacific (Osaka)
     */
    ApNortheast3: "ap-northeast-3",
    /**
     * Asia Pacific (Mumbai)
     */
    ApSouth1: "ap-south-1",
    /**
     * Asia Pacific (Hyderabad)
     */
    ApSouth2: "ap-south-2",
    /**
     * Asia Pacific (Singapore)
     */
    ApSoutheast1: "ap-southeast-1",
    /**
     * Asia Pacific (Sydney)
     */
    ApSoutheast2: "ap-southeast-2",
    /**
     * Asia Pacific (Jakarta)
     */
    ApSoutheast3: "ap-southeast-3",
    /**
     * Asia Pacific (Melbourne)
     */
    ApSoutheast4: "ap-southeast-4",
    /**
     * Asia Pacific (Malaysia)
     */
    ApSoutheast5: "ap-southeast-5",
    /**
     * Canada (Central)
     */
    CaCentral1: "ca-central-1",
    /**
     * Canada West (Calgary)
     */
    CaWest1: "ca-west-1",
    /**
     * China (Beijing)
     */
    CnNorth1: "cn-north-1",
    /**
     * China (Ningxia)
     */
    CnNorthwest1: "cn-northwest-1",
    /**
     * Europe (Frankfurt)
     */
    EuCentral1: "eu-central-1",
    /**
     * Europe (Zurich)
     */
    EuCentral2: "eu-central-2",
    /**
     * EU ISOE West
     */
    EuIsoeWest1: "eu-isoe-west-1",
    /**
     * Europe (Stockholm)
     */
    EuNorth1: "eu-north-1",
    /**
     * Europe (Milan)
     */
    EuSouth1: "eu-south-1",
    /**
     * Europe (Spain)
     */
    EuSouth2: "eu-south-2",
    /**
     * Europe (Ireland)
     */
    EuWest1: "eu-west-1",
    /**
     * Europe (London)
     */
    EuWest2: "eu-west-2",
    /**
     * Europe (Paris)
     */
    EuWest3: "eu-west-3",
    /**
     * Israel (Tel Aviv)
     */
    IlCentral1: "il-central-1",
    /**
     * Middle East (UAE)
     */
    MeCentral1: "me-central-1",
    /**
     * Middle East (Bahrain)
     */
    MeSouth1: "me-south-1",
    /**
     * South America (Sao Paulo)
     */
    SaEast1: "sa-east-1",
    /**
     * US East (N. Virginia)
     */
    UsEast1: "us-east-1",
    /**
     * US East (Ohio)
     */
    UsEast2: "us-east-2",
    /**
     * AWS GovCloud (US-East)
     */
    UsGovEast1: "us-gov-east-1",
    /**
     * AWS GovCloud (US-West)
     */
    UsGovWest1: "us-gov-west-1",
    /**
     * US ISO East
     */
    UsIsoEast1: "us-iso-east-1",
    /**
     * US ISO WEST
     */
    UsIsoWest1: "us-iso-west-1",
    /**
     * US ISOB East (Ohio)
     */
    UsIsobEast1: "us-isob-east-1",
    /**
     * US West (N. California)
     */
    UsWest1: "us-west-1",
    /**
     * US West (Oregon)
     */
    UsWest2: "us-west-2",
};
