// @pulumi/aws-native/types/enums/stepfunctions/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StateMachineType = exports.StateMachineLoggingConfigurationLevel = exports.StateMachineEncryptionConfigurationType = exports.StateMachineAliasDeploymentPreferenceType = exports.ActivityEncryptionConfigurationType = void 0;
exports.ActivityEncryptionConfigurationType = {
    CustomerManagedKmsKey: "CUSTOMER_MANAGED_KMS_KEY",
    AwsOwnedKey: "AWS_OWNED_KEY",
};
exports.StateMachineAliasDeploymentPreferenceType = {
    Linear: "LINEAR",
    AllAtOnce: "ALL_AT_ONCE",
    Canary: "CANARY",
};
exports.StateMachineEncryptionConfigurationType = {
    CustomerManagedKmsKey: "CUSTOMER_MANAGED_KMS_KEY",
    AwsOwnedKey: "AWS_OWNED_KEY",
};
exports.StateMachineLoggingConfigurationLevel = {
    All: "ALL",
    Error: "ERROR",
    Fatal: "FATAL",
    Off: "OFF",
};
exports.StateMachineType = {
    Standard: "STANDARD",
    Express: "EXPRESS",
};
