// @pulumi/aws-native/types/enums/securityhub/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityControlParameterConfigurationValueType = exports.PolicyAssociationTargetType = exports.PolicyAssociationAssociationType = exports.PolicyAssociationAssociationStatus = exports.OrganizationConfigurationStatus = exports.OrganizationConfigurationConfigurationType = exports.OrganizationConfigurationAutoEnableStandards = exports.InsightStringFilterComparison = exports.InsightMapFilterComparison = exports.InsightDateRangeUnit = exports.FindingAggregatorRegionLinkingMode = exports.DelegatedAdminStatus = exports.ConfigurationPolicyParameterConfigurationValueType = exports.AutomationRulesFindingFieldsUpdateVerificationState = exports.AutomationRulesActionType = exports.AutomationRuleWorkflowUpdateStatus = exports.AutomationRuleStringFilterComparison = exports.AutomationRuleSeverityUpdateLabel = exports.AutomationRuleRuleStatus = exports.AutomationRuleMapFilterComparison = exports.AutomationRuleDateRangeUnit = void 0;
exports.AutomationRuleDateRangeUnit = {
    Days: "DAYS",
};
exports.AutomationRuleMapFilterComparison = {
    Equals: "EQUALS",
    NotEquals: "NOT_EQUALS",
    Contains: "CONTAINS",
    NotContains: "NOT_CONTAINS",
};
exports.AutomationRuleRuleStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.AutomationRuleSeverityUpdateLabel = {
    Informational: "INFORMATIONAL",
    Low: "LOW",
    Medium: "MEDIUM",
    High: "HIGH",
    Critical: "CRITICAL",
};
exports.AutomationRuleStringFilterComparison = {
    Equals: "EQUALS",
    Prefix: "PREFIX",
    NotEquals: "NOT_EQUALS",
    PrefixNotEquals: "PREFIX_NOT_EQUALS",
    Contains: "CONTAINS",
    NotContains: "NOT_CONTAINS",
};
exports.AutomationRuleWorkflowUpdateStatus = {
    New: "NEW",
    Notified: "NOTIFIED",
    Resolved: "RESOLVED",
    Suppressed: "SUPPRESSED",
};
exports.AutomationRulesActionType = {
    FindingFieldsUpdate: "FINDING_FIELDS_UPDATE",
};
exports.AutomationRulesFindingFieldsUpdateVerificationState = {
    Unknown: "UNKNOWN",
    TruePositive: "TRUE_POSITIVE",
    FalsePositive: "FALSE_POSITIVE",
    BenignPositive: "BENIGN_POSITIVE",
};
exports.ConfigurationPolicyParameterConfigurationValueType = {
    Default: "DEFAULT",
    Custom: "CUSTOM",
};
exports.DelegatedAdminStatus = {
    Enabled: "ENABLED",
    DisableInProgress: "DISABLE_IN_PROGRESS",
};
exports.FindingAggregatorRegionLinkingMode = {
    AllRegions: "ALL_REGIONS",
    AllRegionsExceptSpecified: "ALL_REGIONS_EXCEPT_SPECIFIED",
    SpecifiedRegions: "SPECIFIED_REGIONS",
};
exports.InsightDateRangeUnit = {
    Days: "DAYS",
};
exports.InsightMapFilterComparison = {
    Equals: "EQUALS",
    NotEquals: "NOT_EQUALS",
};
exports.InsightStringFilterComparison = {
    Equals: "EQUALS",
    Prefix: "PREFIX",
    NotEquals: "NOT_EQUALS",
    PrefixNotEquals: "PREFIX_NOT_EQUALS",
};
exports.OrganizationConfigurationAutoEnableStandards = {
    Default: "DEFAULT",
    None: "NONE",
};
exports.OrganizationConfigurationConfigurationType = {
    Central: "CENTRAL",
    Local: "LOCAL",
};
exports.OrganizationConfigurationStatus = {
    Pending: "PENDING",
    Enabled: "ENABLED",
    Failed: "FAILED",
};
exports.PolicyAssociationAssociationStatus = {
    Success: "SUCCESS",
    Pending: "PENDING",
    Failed: "FAILED",
};
exports.PolicyAssociationAssociationType = {
    Applied: "APPLIED",
    Inherited: "INHERITED",
};
exports.PolicyAssociationTargetType = {
    Account: "ACCOUNT",
    OrganizationalUnit: "ORGANIZATIONAL_UNIT",
    Root: "ROOT",
};
exports.SecurityControlParameterConfigurationValueType = {
    Default: "DEFAULT",
    Custom: "CUSTOM",
};
