// @pulumi/aws-native/securitylake/awsLogSource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwsLogSource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SecurityLake::AwsLogSource
 */
class AwsLogSource extends pulumi.CustomResource {
    /**
     * Get an existing AwsLogSource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AwsLogSource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AwsLogSource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AwsLogSource.__pulumiType;
    }
    /**
     * Create a AwsLogSource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataLakeArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataLakeArn'");
            }
            if ((!args || args.sourceVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceVersion'");
            }
            resourceInputs["accounts"] = args ? args.accounts : undefined;
            resourceInputs["dataLakeArn"] = args ? args.dataLakeArn : undefined;
            resourceInputs["sourceName"] = args ? args.sourceName : undefined;
            resourceInputs["sourceVersion"] = args ? args.sourceVersion : undefined;
        }
        else {
            resourceInputs["accounts"] = undefined /*out*/;
            resourceInputs["dataLakeArn"] = undefined /*out*/;
            resourceInputs["sourceName"] = undefined /*out*/;
            resourceInputs["sourceVersion"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["dataLakeArn", "sourceName", "sourceVersion"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(AwsLogSource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AwsLogSource = AwsLogSource;
/** @internal */
AwsLogSource.__pulumiType = 'aws-native:securitylake:AwsLogSource';
