// @pulumi/aws-native/securitylake/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriberNotification = exports.Subscriber = exports.getSubscriberNotificationOutput = exports.getSubscriberNotification = exports.getSubscriberOutput = exports.getSubscriber = exports.getDataLakeOutput = exports.getDataLake = exports.getAwsLogSourceOutput = exports.getAwsLogSource = exports.DataLake = exports.AwsLogSource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AwsLogSource = null;
utilities.lazyLoad(exports, ["AwsLogSource"], () => require("./awsLogSource"));
exports.DataLake = null;
utilities.lazyLoad(exports, ["DataLake"], () => require("./dataLake"));
exports.getAwsLogSource = null;
exports.getAwsLogSourceOutput = null;
utilities.lazyLoad(exports, ["getAwsLogSource", "getAwsLogSourceOutput"], () => require("./getAwsLogSource"));
exports.getDataLake = null;
exports.getDataLakeOutput = null;
utilities.lazyLoad(exports, ["getDataLake", "getDataLakeOutput"], () => require("./getDataLake"));
exports.getSubscriber = null;
exports.getSubscriberOutput = null;
utilities.lazyLoad(exports, ["getSubscriber", "getSubscriberOutput"], () => require("./getSubscriber"));
exports.getSubscriberNotification = null;
exports.getSubscriberNotificationOutput = null;
utilities.lazyLoad(exports, ["getSubscriberNotification", "getSubscriberNotificationOutput"], () => require("./getSubscriberNotification"));
exports.Subscriber = null;
utilities.lazyLoad(exports, ["Subscriber"], () => require("./subscriber"));
exports.SubscriberNotification = null;
utilities.lazyLoad(exports, ["SubscriberNotification"], () => require("./subscriberNotification"));
// Export enums:
__exportStar(require("../types/enums/securitylake"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:securitylake:AwsLogSource":
                return new exports.AwsLogSource(name, undefined, { urn });
            case "aws-native:securitylake:DataLake":
                return new exports.DataLake(name, undefined, { urn });
            case "aws-native:securitylake:Subscriber":
                return new exports.Subscriber(name, undefined, { urn });
            case "aws-native:securitylake:SubscriberNotification":
                return new exports.SubscriberNotification(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "securitylake", _module);
