// @pulumi/aws-native/securitylake/subscriber.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscriber = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SecurityLake::Subscriber
 */
class Subscriber extends pulumi.CustomResource {
    /**
     * Get an existing Subscriber resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Subscriber(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Subscriber.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Subscriber.__pulumiType;
    }
    /**
     * Create a Subscriber resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accessTypes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accessTypes'");
            }
            if ((!args || args.dataLakeArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataLakeArn'");
            }
            if ((!args || args.sources === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sources'");
            }
            if ((!args || args.subscriberIdentity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subscriberIdentity'");
            }
            resourceInputs["accessTypes"] = args ? args.accessTypes : undefined;
            resourceInputs["dataLakeArn"] = args ? args.dataLakeArn : undefined;
            resourceInputs["sources"] = args ? args.sources : undefined;
            resourceInputs["subscriberDescription"] = args ? args.subscriberDescription : undefined;
            resourceInputs["subscriberIdentity"] = args ? args.subscriberIdentity : undefined;
            resourceInputs["subscriberName"] = args ? args.subscriberName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["resourceShareArn"] = undefined /*out*/;
            resourceInputs["resourceShareName"] = undefined /*out*/;
            resourceInputs["s3BucketArn"] = undefined /*out*/;
            resourceInputs["subscriberArn"] = undefined /*out*/;
            resourceInputs["subscriberRoleArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessTypes"] = undefined /*out*/;
            resourceInputs["dataLakeArn"] = undefined /*out*/;
            resourceInputs["resourceShareArn"] = undefined /*out*/;
            resourceInputs["resourceShareName"] = undefined /*out*/;
            resourceInputs["s3BucketArn"] = undefined /*out*/;
            resourceInputs["sources"] = undefined /*out*/;
            resourceInputs["subscriberArn"] = undefined /*out*/;
            resourceInputs["subscriberDescription"] = undefined /*out*/;
            resourceInputs["subscriberIdentity"] = undefined /*out*/;
            resourceInputs["subscriberName"] = undefined /*out*/;
            resourceInputs["subscriberRoleArn"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["dataLakeArn"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(Subscriber.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Subscriber = Subscriber;
/** @internal */
Subscriber.__pulumiType = 'aws-native:securitylake:Subscriber';
