// @pulumi/aws-native/types/enums/gamelift/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MatchmakingConfigurationFlexMatchMode = exports.MatchmakingConfigurationBackfillMode = exports.GameSessionQueuePriorityOrderItem = exports.GameServerGroupGameServerProtectionPolicy = exports.GameServerGroupDeleteOption = exports.GameServerGroupBalancingStrategy = exports.FleetType = exports.FleetScalingPolicyUpdateStatus = exports.FleetScalingPolicyStatus = exports.FleetScalingPolicyScalingAdjustmentType = exports.FleetScalingPolicyPolicyType = exports.FleetScalingPolicyMetricName = exports.FleetScalingPolicyComparisonOperator = exports.FleetNewGameSessionProtectionPolicy = exports.FleetIpPermissionProtocol = exports.FleetInstanceRoleCredentialsProvider = exports.FleetComputeType = exports.FleetCertificateConfigurationCertificateType = exports.FleetApplyCapacity = exports.ContainerGroupDefinitionStatus = exports.ContainerGroupDefinitionOperatingSystem = exports.ContainerGroupDefinitionContainerPortRangeProtocol = exports.ContainerGroupDefinitionContainerMountPointAccessLevel = exports.ContainerGroupDefinitionContainerGroupType = exports.ContainerGroupDefinitionContainerDependencyCondition = exports.ContainerFleetStoppedActionsItem = exports.ContainerFleetStatus = exports.ContainerFleetScalingPolicyScalingAdjustmentType = exports.ContainerFleetScalingPolicyPolicyType = exports.ContainerFleetScalingPolicyMetricName = exports.ContainerFleetScalingPolicyComparisonOperator = exports.ContainerFleetNewGameSessionProtectionPolicy = exports.ContainerFleetLogDestination = exports.ContainerFleetIpPermissionProtocol = exports.ContainerFleetDeploymentConfigurationProtectionStrategy = exports.ContainerFleetDeploymentConfigurationImpairmentStrategy = exports.ContainerFleetBillingType = exports.BuildOperatingSystem = exports.AliasRoutingStrategyType = void 0;
exports.AliasRoutingStrategyType = {
    Simple: "SIMPLE",
    Terminal: "TERMINAL",
};
exports.BuildOperatingSystem = {
    AmazonLinux: "AMAZON_LINUX",
    AmazonLinux2: "AMAZON_LINUX_2",
    AmazonLinux2023: "AMAZON_LINUX_2023",
    Windows2012: "WINDOWS_2012",
    Windows2016: "WINDOWS_2016",
};
exports.ContainerFleetBillingType = {
    OnDemand: "ON_DEMAND",
    Spot: "SPOT",
};
exports.ContainerFleetDeploymentConfigurationImpairmentStrategy = {
    Maintain: "MAINTAIN",
    Rollback: "ROLLBACK",
};
exports.ContainerFleetDeploymentConfigurationProtectionStrategy = {
    WithProtection: "WITH_PROTECTION",
    IgnoreProtection: "IGNORE_PROTECTION",
};
exports.ContainerFleetIpPermissionProtocol = {
    Tcp: "TCP",
    Udp: "UDP",
};
exports.ContainerFleetLogDestination = {
    None: "NONE",
    Cloudwatch: "CLOUDWATCH",
    S3: "S3",
};
exports.ContainerFleetNewGameSessionProtectionPolicy = {
    FullProtection: "FullProtection",
    NoProtection: "NoProtection",
};
exports.ContainerFleetScalingPolicyComparisonOperator = {
    GreaterThanOrEqualToThreshold: "GreaterThanOrEqualToThreshold",
    GreaterThanThreshold: "GreaterThanThreshold",
    LessThanThreshold: "LessThanThreshold",
    LessThanOrEqualToThreshold: "LessThanOrEqualToThreshold",
};
exports.ContainerFleetScalingPolicyMetricName = {
    ActivatingGameSessions: "ActivatingGameSessions",
    ActiveGameSessions: "ActiveGameSessions",
    ActiveInstances: "ActiveInstances",
    AvailableGameSessions: "AvailableGameSessions",
    AvailablePlayerSessions: "AvailablePlayerSessions",
    CurrentPlayerSessions: "CurrentPlayerSessions",
    IdleInstances: "IdleInstances",
    PercentAvailableGameSessions: "PercentAvailableGameSessions",
    PercentIdleInstances: "PercentIdleInstances",
    QueueDepth: "QueueDepth",
    WaitTime: "WaitTime",
    ConcurrentActivatableGameSessions: "ConcurrentActivatableGameSessions",
};
exports.ContainerFleetScalingPolicyPolicyType = {
    RuleBased: "RuleBased",
    TargetBased: "TargetBased",
};
exports.ContainerFleetScalingPolicyScalingAdjustmentType = {
    ChangeInCapacity: "ChangeInCapacity",
    ExactCapacity: "ExactCapacity",
    PercentChangeInCapacity: "PercentChangeInCapacity",
};
exports.ContainerFleetStatus = {
    Pending: "PENDING",
    Creating: "CREATING",
    Created: "CREATED",
    Activating: "ACTIVATING",
    Active: "ACTIVE",
    Updating: "UPDATING",
    Deleting: "DELETING",
};
exports.ContainerFleetStoppedActionsItem = {
    AutoScaling: "AUTO_SCALING",
};
exports.ContainerGroupDefinitionContainerDependencyCondition = {
    Start: "START",
    Complete: "COMPLETE",
    Success: "SUCCESS",
    Healthy: "HEALTHY",
};
exports.ContainerGroupDefinitionContainerGroupType = {
    GameServer: "GAME_SERVER",
    PerInstance: "PER_INSTANCE",
};
exports.ContainerGroupDefinitionContainerMountPointAccessLevel = {
    ReadOnly: "READ_ONLY",
    ReadAndWrite: "READ_AND_WRITE",
};
exports.ContainerGroupDefinitionContainerPortRangeProtocol = {
    Tcp: "TCP",
    Udp: "UDP",
};
exports.ContainerGroupDefinitionOperatingSystem = {
    AmazonLinux2023: "AMAZON_LINUX_2023",
};
exports.ContainerGroupDefinitionStatus = {
    Ready: "READY",
    Copying: "COPYING",
    Failed: "FAILED",
};
exports.FleetApplyCapacity = {
    OnUpdate: "ON_UPDATE",
    OnCreateAndUpdate: "ON_CREATE_AND_UPDATE",
};
exports.FleetCertificateConfigurationCertificateType = {
    Disabled: "DISABLED",
    Generated: "GENERATED",
};
exports.FleetComputeType = {
    Ec2: "EC2",
    Anywhere: "ANYWHERE",
};
exports.FleetInstanceRoleCredentialsProvider = {
    SharedCredentialFile: "SHARED_CREDENTIAL_FILE",
};
exports.FleetIpPermissionProtocol = {
    Tcp: "TCP",
    Udp: "UDP",
};
exports.FleetNewGameSessionProtectionPolicy = {
    FullProtection: "FullProtection",
    NoProtection: "NoProtection",
};
exports.FleetScalingPolicyComparisonOperator = {
    GreaterThanOrEqualToThreshold: "GreaterThanOrEqualToThreshold",
    GreaterThanThreshold: "GreaterThanThreshold",
    LessThanThreshold: "LessThanThreshold",
    LessThanOrEqualToThreshold: "LessThanOrEqualToThreshold",
};
exports.FleetScalingPolicyMetricName = {
    ActivatingGameSessions: "ActivatingGameSessions",
    ActiveGameSessions: "ActiveGameSessions",
    ActiveInstances: "ActiveInstances",
    AvailableGameSessions: "AvailableGameSessions",
    AvailablePlayerSessions: "AvailablePlayerSessions",
    CurrentPlayerSessions: "CurrentPlayerSessions",
    IdleInstances: "IdleInstances",
    PercentAvailableGameSessions: "PercentAvailableGameSessions",
    PercentIdleInstances: "PercentIdleInstances",
    QueueDepth: "QueueDepth",
    WaitTime: "WaitTime",
    ConcurrentActivatableGameSessions: "ConcurrentActivatableGameSessions",
};
exports.FleetScalingPolicyPolicyType = {
    RuleBased: "RuleBased",
    TargetBased: "TargetBased",
};
exports.FleetScalingPolicyScalingAdjustmentType = {
    ChangeInCapacity: "ChangeInCapacity",
    ExactCapacity: "ExactCapacity",
    PercentChangeInCapacity: "PercentChangeInCapacity",
};
exports.FleetScalingPolicyStatus = {
    Active: "ACTIVE",
    UpdateRequested: "UPDATE_REQUESTED",
    Updating: "UPDATING",
    DeleteRequested: "DELETE_REQUESTED",
    Deleting: "DELETING",
    Deleted: "DELETED",
    Error: "ERROR",
};
exports.FleetScalingPolicyUpdateStatus = {
    PendingUpdate: "PENDING_UPDATE",
};
exports.FleetType = {
    OnDemand: "ON_DEMAND",
    Spot: "SPOT",
};
exports.GameServerGroupBalancingStrategy = {
    SpotOnly: "SPOT_ONLY",
    SpotPreferred: "SPOT_PREFERRED",
    OnDemandOnly: "ON_DEMAND_ONLY",
};
exports.GameServerGroupDeleteOption = {
    SafeDelete: "SAFE_DELETE",
    ForceDelete: "FORCE_DELETE",
    Retain: "RETAIN",
};
exports.GameServerGroupGameServerProtectionPolicy = {
    NoProtection: "NO_PROTECTION",
    FullProtection: "FULL_PROTECTION",
};
exports.GameSessionQueuePriorityOrderItem = {
    Latency: "LATENCY",
    Cost: "COST",
    Destination: "DESTINATION",
    Location: "LOCATION",
};
exports.MatchmakingConfigurationBackfillMode = {
    Automatic: "AUTOMATIC",
    Manual: "MANUAL",
};
exports.MatchmakingConfigurationFlexMatchMode = {
    Standalone: "STANDALONE",
    WithQueue: "WITH_QUEUE",
};
