// @pulumi/aws-native/entityresolution/idMappingWorkflow.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdMappingWorkflow = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * IdMappingWorkflow defined in AWS Entity Resolution service
 */
class IdMappingWorkflow extends pulumi.CustomResource {
    /**
     * Get an existing IdMappingWorkflow resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IdMappingWorkflow(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IdMappingWorkflow.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IdMappingWorkflow.__pulumiType;
    }
    /**
     * Create a IdMappingWorkflow resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.idMappingTechniques === undefined) && !opts.urn) {
                throw new Error("Missing required property 'idMappingTechniques'");
            }
            if ((!args || args.inputSourceConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inputSourceConfig'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["idMappingTechniques"] = args ? args.idMappingTechniques : undefined;
            resourceInputs["inputSourceConfig"] = args ? args.inputSourceConfig : undefined;
            resourceInputs["outputSourceConfig"] = args ? args.outputSourceConfig : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["workflowName"] = args ? args.workflowName : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
            resourceInputs["workflowArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["idMappingTechniques"] = undefined /*out*/;
            resourceInputs["inputSourceConfig"] = undefined /*out*/;
            resourceInputs["outputSourceConfig"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
            resourceInputs["workflowArn"] = undefined /*out*/;
            resourceInputs["workflowName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["workflowName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(IdMappingWorkflow.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IdMappingWorkflow = IdMappingWorkflow;
/** @internal */
IdMappingWorkflow.__pulumiType = 'aws-native:entityresolution:IdMappingWorkflow';
