// @pulumi/aws-native/entityresolution/policyStatement.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyStatement = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Policy Statement defined in AWS Entity Resolution Service
 */
class PolicyStatement extends pulumi.CustomResource {
    /**
     * Get an existing PolicyStatement resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PolicyStatement(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PolicyStatement.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PolicyStatement.__pulumiType;
    }
    /**
     * Create a PolicyStatement resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.arn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'arn'");
            }
            if ((!args || args.statementId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'statementId'");
            }
            resourceInputs["action"] = args ? args.action : undefined;
            resourceInputs["arn"] = args ? args.arn : undefined;
            resourceInputs["condition"] = args ? args.condition : undefined;
            resourceInputs["effect"] = args ? args.effect : undefined;
            resourceInputs["principal"] = args ? args.principal : undefined;
            resourceInputs["statementId"] = args ? args.statementId : undefined;
        }
        else {
            resourceInputs["action"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["condition"] = undefined /*out*/;
            resourceInputs["effect"] = undefined /*out*/;
            resourceInputs["principal"] = undefined /*out*/;
            resourceInputs["statementId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["arn", "statementId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(PolicyStatement.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PolicyStatement = PolicyStatement;
/** @internal */
PolicyStatement.__pulumiType = 'aws-native:entityresolution:PolicyStatement';
