// @pulumi/aws-native/types/enums/ec2/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotFleetInstanceRequirementsRequestCpuManufacturersItem = exports.SpotFleetInstanceRequirementsRequestBurstablePerformance = exports.SpotFleetInstanceRequirementsRequestBareMetal = exports.SpotFleetInstanceRequirementsRequestAcceleratorTypesItem = exports.SpotFleetInstanceRequirementsRequestAcceleratorNamesItem = exports.SpotFleetInstanceRequirementsRequestAcceleratorManufacturersItem = exports.SpotFleetEbsBlockDeviceVolumeType = exports.SnapshotBlockPublicAccessState = exports.SecurityGroupVpcAssociationState = exports.PrefixListAddressFamily = exports.NetworkInsightsPathProtocol = exports.NetworkInsightsAnalysisStatus = exports.NetworkInsightsAccessScopeProtocol = exports.NetworkInsightsAccessScopeAnalysisStatus = exports.NetworkInsightsAccessScopeAnalysisFindingsFound = exports.LaunchTemplateCpuOptionsAmdSevSnp = exports.KeyPairKeyType = exports.KeyPairKeyFormat = exports.IpamTier = exports.IpamScopeType = exports.IpamPoolState = exports.IpamPoolPublicIpSource = exports.IpamPoolIpamScopeType = exports.IpamPoolAwsService = exports.InstancePrivateDnsNameOptionsHostnameType = exports.InstanceAffinity = exports.FlowLogTrafficType = exports.FlowLogResourceType = exports.FlowLogLogDestinationType = exports.FlowLogDestinationOptionsPropertiesFileFormat = exports.Ec2FleetType = exports.Ec2FleetTargetCapacitySpecificationRequestTargetCapacityUnitType = exports.Ec2FleetTargetCapacitySpecificationRequestDefaultTargetCapacityType = exports.Ec2FleetTagSpecificationResourceType = exports.Ec2FleetSpotOptionsRequestInstanceInterruptionBehavior = exports.Ec2FleetSpotOptionsRequestAllocationStrategy = exports.Ec2FleetInstanceRequirementsRequestLocalStorageTypesItem = exports.Ec2FleetInstanceRequirementsRequestLocalStorage = exports.Ec2FleetInstanceRequirementsRequestInstanceGenerationsItem = exports.Ec2FleetInstanceRequirementsRequestCpuManufacturersItem = exports.Ec2FleetInstanceRequirementsRequestBurstablePerformance = exports.Ec2FleetInstanceRequirementsRequestBareMetal = exports.Ec2FleetInstanceRequirementsRequestAcceleratorTypesItem = exports.Ec2FleetInstanceRequirementsRequestAcceleratorNamesItem = exports.Ec2FleetInstanceRequirementsRequestAcceleratorManufacturersItem = exports.Ec2FleetExcessCapacityTerminationPolicy = exports.Ec2FleetCapacityReservationOptionsRequestUsageStrategy = exports.Ec2FleetCapacityRebalanceReplacementStrategy = exports.CapacityReservationFleetTenancy = exports.CapacityReservationFleetInstanceMatchCriteria = void 0;
exports.VpnConnectionVpnTunnelOptionsSpecificationStartupAction = exports.VpnConnectionVpnTunnelOptionsSpecificationDpdTimeoutAction = exports.VpnConnectionPhase2IntegrityAlgorithmsRequestListValueValue = exports.VpnConnectionPhase2EncryptionAlgorithmsRequestListValueValue = exports.VpnConnectionPhase1IntegrityAlgorithmsRequestListValueValue = exports.VpnConnectionPhase1EncryptionAlgorithmsRequestListValueValue = exports.VpnConnectionIkeVersionsRequestListValueValue = exports.VpnConnectionCloudwatchLogOptionsSpecificationLogOutputFormat = exports.VpcEndpointType = exports.VpcEndpointIpAddressType = exports.VpcEndpointDnsOptionsSpecificationPrivateDnsOnlyForInboundResolverEndpoint = exports.VpcEndpointDnsOptionsSpecificationDnsRecordIpType = exports.VpcBlockPublicAccessOptionsInternetGatewayBlockMode = exports.VpcBlockPublicAccessExclusionInternetGatewayExclusionMode = exports.SpotFleetTagSpecificationResourceType = exports.SpotFleetSpotPlacementTenancy = exports.SpotFleetSpotCapacityRebalanceReplacementStrategy = exports.SpotFleetRequestConfigDataType = exports.SpotFleetRequestConfigDataTargetCapacityUnitType = exports.SpotFleetRequestConfigDataInstanceInterruptionBehavior = exports.SpotFleetRequestConfigDataExcessCapacityTerminationPolicy = exports.SpotFleetRequestConfigDataAllocationStrategy = exports.SpotFleetInstanceRequirementsRequestLocalStorageTypesItem = exports.SpotFleetInstanceRequirementsRequestLocalStorage = exports.SpotFleetInstanceRequirementsRequestInstanceGenerationsItem = void 0;
exports.CapacityReservationFleetInstanceMatchCriteria = {
    Open: "open",
};
exports.CapacityReservationFleetTenancy = {
    Default: "default",
};
exports.Ec2FleetCapacityRebalanceReplacementStrategy = {
    Launch: "launch",
    LaunchBeforeTerminate: "launch-before-terminate",
};
exports.Ec2FleetCapacityReservationOptionsRequestUsageStrategy = {
    UseCapacityReservationsFirst: "use-capacity-reservations-first",
};
exports.Ec2FleetExcessCapacityTerminationPolicy = {
    Termination: "termination",
    NoTermination: "no-termination",
};
exports.Ec2FleetInstanceRequirementsRequestAcceleratorManufacturersItem = {
    AmazonWebServices: "amazon-web-services",
    Amd: "amd",
    Habana: "habana",
    Nvidia: "nvidia",
    Xilinx: "xilinx",
};
exports.Ec2FleetInstanceRequirementsRequestAcceleratorNamesItem = {
    A10g: "a10g",
    A100: "a100",
    H100: "h100",
    Inferentia: "inferentia",
    K520: "k520",
    K80: "k80",
    M60: "m60",
    RadeonProV520: "radeon-pro-v520",
    T4: "t4",
    T4g: "t4g",
    Vu9p: "vu9p",
    V100: "v100",
};
exports.Ec2FleetInstanceRequirementsRequestAcceleratorTypesItem = {
    Gpu: "gpu",
    Fpga: "fpga",
    Inference: "inference",
};
exports.Ec2FleetInstanceRequirementsRequestBareMetal = {
    Included: "included",
    Required: "required",
    Excluded: "excluded",
};
exports.Ec2FleetInstanceRequirementsRequestBurstablePerformance = {
    Included: "included",
    Required: "required",
    Excluded: "excluded",
};
exports.Ec2FleetInstanceRequirementsRequestCpuManufacturersItem = {
    Intel: "intel",
    Amd: "amd",
    AmazonWebServices: "amazon-web-services",
    Apple: "apple",
};
exports.Ec2FleetInstanceRequirementsRequestInstanceGenerationsItem = {
    Current: "current",
    Previous: "previous",
};
exports.Ec2FleetInstanceRequirementsRequestLocalStorage = {
    Included: "included",
    Required: "required",
    Excluded: "excluded",
};
exports.Ec2FleetInstanceRequirementsRequestLocalStorageTypesItem = {
    Hdd: "hdd",
    Ssd: "ssd",
};
exports.Ec2FleetSpotOptionsRequestAllocationStrategy = {
    LowestPrice: "lowest-price",
    Diversified: "diversified",
    CapacityOptimized: "capacityOptimized",
    CapacityOptimizedPrioritized: "capacityOptimizedPrioritized",
    PriceCapacityOptimized: "priceCapacityOptimized",
};
exports.Ec2FleetSpotOptionsRequestInstanceInterruptionBehavior = {
    Hibernate: "hibernate",
    Stop: "stop",
    Terminate: "terminate",
};
exports.Ec2FleetTagSpecificationResourceType = {
    ClientVpnEndpoint: "client-vpn-endpoint",
    CustomerGateway: "customer-gateway",
    DedicatedHost: "dedicated-host",
    DhcpOptions: "dhcp-options",
    EgressOnlyInternetGateway: "egress-only-internet-gateway",
    ElasticGpu: "elastic-gpu",
    ElasticIp: "elastic-ip",
    ExportImageTask: "export-image-task",
    ExportInstanceTask: "export-instance-task",
    Fleet: "fleet",
    FpgaImage: "fpga-image",
    HostReservation: "host-reservation",
    Image: "image",
    ImportImageTask: "import-image-task",
    ImportSnapshotTask: "import-snapshot-task",
    Instance: "instance",
    InternetGateway: "internet-gateway",
    KeyPair: "key-pair",
    LaunchTemplate: "launch-template",
    LocalGatewayRouteTableVpcAssociation: "local-gateway-route-table-vpc-association",
    Natgateway: "natgateway",
    NetworkAcl: "network-acl",
    NetworkInsightsAnalysis: "network-insights-analysis",
    NetworkInsightsPath: "network-insights-path",
    NetworkInterface: "network-interface",
    PlacementGroup: "placement-group",
    ReservedInstances: "reserved-instances",
    RouteTable: "route-table",
    SecurityGroup: "security-group",
    Snapshot: "snapshot",
    SpotFleetRequest: "spot-fleet-request",
    SpotInstancesRequest: "spot-instances-request",
    Subnet: "subnet",
    TrafficMirrorFilter: "traffic-mirror-filter",
    TrafficMirrorSession: "traffic-mirror-session",
    TrafficMirrorTarget: "traffic-mirror-target",
    TransitGateway: "transit-gateway",
    TransitGatewayAttachment: "transit-gateway-attachment",
    TransitGatewayConnectPeer: "transit-gateway-connect-peer",
    TransitGatewayMulticastDomain: "transit-gateway-multicast-domain",
    TransitGatewayRouteTable: "transit-gateway-route-table",
    Volume: "volume",
    Vpc: "vpc",
    VpcFlowLog: "vpc-flow-log",
    VpcPeeringConnection: "vpc-peering-connection",
    VpnConnection: "vpn-connection",
    VpnGateway: "vpn-gateway",
};
exports.Ec2FleetTargetCapacitySpecificationRequestDefaultTargetCapacityType = {
    OnDemand: "on-demand",
    Spot: "spot",
};
exports.Ec2FleetTargetCapacitySpecificationRequestTargetCapacityUnitType = {
    Vcpu: "vcpu",
    MemoryMib: "memory-mib",
    Units: "units",
};
exports.Ec2FleetType = {
    Maintain: "maintain",
    Request: "request",
    Instant: "instant",
};
exports.FlowLogDestinationOptionsPropertiesFileFormat = {
    PlainText: "plain-text",
    Parquet: "parquet",
};
exports.FlowLogLogDestinationType = {
    CloudWatchLogs: "cloud-watch-logs",
    S3: "s3",
    KinesisDataFirehose: "kinesis-data-firehose",
};
exports.FlowLogResourceType = {
    NetworkInterface: "NetworkInterface",
    Subnet: "Subnet",
    Vpc: "VPC",
    TransitGateway: "TransitGateway",
    TransitGatewayAttachment: "TransitGatewayAttachment",
};
exports.FlowLogTrafficType = {
    Accept: "ACCEPT",
    All: "ALL",
    Reject: "REJECT",
};
exports.InstanceAffinity = {
    Default: "default",
    Host: "host",
};
exports.InstancePrivateDnsNameOptionsHostnameType = {
    IpName: "ip-name",
    ResourceName: "resource-name",
};
exports.IpamPoolAwsService = {
    Ec2: "ec2",
};
exports.IpamPoolIpamScopeType = {
    Public: "public",
    Private: "private",
};
exports.IpamPoolPublicIpSource = {
    Byoip: "byoip",
    Amazon: "amazon",
};
exports.IpamPoolState = {
    CreateInProgress: "create-in-progress",
    CreateComplete: "create-complete",
    ModifyInProgress: "modify-in-progress",
    ModifyComplete: "modify-complete",
    DeleteInProgress: "delete-in-progress",
    DeleteComplete: "delete-complete",
};
exports.IpamScopeType = {
    Public: "public",
    Private: "private",
};
exports.IpamTier = {
    Free: "free",
    Advanced: "advanced",
};
exports.KeyPairKeyFormat = {
    Pem: "pem",
    Ppk: "ppk",
};
exports.KeyPairKeyType = {
    Rsa: "rsa",
    Ed25519: "ed25519",
};
exports.LaunchTemplateCpuOptionsAmdSevSnp = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.NetworkInsightsAccessScopeAnalysisFindingsFound = {
    True: "true",
    False: "false",
    Unknown: "unknown",
};
exports.NetworkInsightsAccessScopeAnalysisStatus = {
    Running: "running",
    Failed: "failed",
    Succeeded: "succeeded",
};
exports.NetworkInsightsAccessScopeProtocol = {
    Tcp: "tcp",
    Udp: "udp",
};
exports.NetworkInsightsAnalysisStatus = {
    Running: "running",
    Failed: "failed",
    Succeeded: "succeeded",
};
exports.NetworkInsightsPathProtocol = {
    Tcp: "tcp",
    Udp: "udp",
};
exports.PrefixListAddressFamily = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.SecurityGroupVpcAssociationState = {
    Associating: "associating",
    Associated: "associated",
    AssociationFailed: "association-failed",
    Disassociating: "disassociating",
    Disassociated: "disassociated",
    DisassociationFailed: "disassociation-failed",
};
exports.SnapshotBlockPublicAccessState = {
    BlockAllSharing: "block-all-sharing",
    BlockNewSharing: "block-new-sharing",
};
exports.SpotFleetEbsBlockDeviceVolumeType = {
    Gp2: "gp2",
    Gp3: "gp3",
    Io1: "io1",
    Io2: "io2",
    Sc1: "sc1",
    St1: "st1",
    Standard: "standard",
};
exports.SpotFleetInstanceRequirementsRequestAcceleratorManufacturersItem = {
    AmazonWebServices: "amazon-web-services",
    Amd: "amd",
    Habana: "habana",
    Nvidia: "nvidia",
    Xilinx: "xilinx",
};
exports.SpotFleetInstanceRequirementsRequestAcceleratorNamesItem = {
    A10g: "a10g",
    A100: "a100",
    H100: "h100",
    Inferentia: "inferentia",
    K520: "k520",
    K80: "k80",
    M60: "m60",
    RadeonProV520: "radeon-pro-v520",
    T4: "t4",
    T4g: "t4g",
    Vu9p: "vu9p",
    V100: "v100",
};
exports.SpotFleetInstanceRequirementsRequestAcceleratorTypesItem = {
    Gpu: "gpu",
    Fpga: "fpga",
    Inference: "inference",
};
exports.SpotFleetInstanceRequirementsRequestBareMetal = {
    Included: "included",
    Required: "required",
    Excluded: "excluded",
};
exports.SpotFleetInstanceRequirementsRequestBurstablePerformance = {
    Included: "included",
    Required: "required",
    Excluded: "excluded",
};
exports.SpotFleetInstanceRequirementsRequestCpuManufacturersItem = {
    Intel: "intel",
    Amd: "amd",
    AmazonWebServices: "amazon-web-services",
    Apple: "apple",
};
exports.SpotFleetInstanceRequirementsRequestInstanceGenerationsItem = {
    Current: "current",
    Previous: "previous",
};
exports.SpotFleetInstanceRequirementsRequestLocalStorage = {
    Included: "included",
    Required: "required",
    Excluded: "excluded",
};
exports.SpotFleetInstanceRequirementsRequestLocalStorageTypesItem = {
    Hdd: "hdd",
    Ssd: "ssd",
};
exports.SpotFleetRequestConfigDataAllocationStrategy = {
    CapacityOptimized: "capacityOptimized",
    CapacityOptimizedPrioritized: "capacityOptimizedPrioritized",
    Diversified: "diversified",
    LowestPrice: "lowestPrice",
    PriceCapacityOptimized: "priceCapacityOptimized",
};
exports.SpotFleetRequestConfigDataExcessCapacityTerminationPolicy = {
    Default: "Default",
    NoTermination: "NoTermination",
};
exports.SpotFleetRequestConfigDataInstanceInterruptionBehavior = {
    Hibernate: "hibernate",
    Stop: "stop",
    Terminate: "terminate",
};
exports.SpotFleetRequestConfigDataTargetCapacityUnitType = {
    Vcpu: "vcpu",
    MemoryMib: "memory-mib",
    Units: "units",
};
exports.SpotFleetRequestConfigDataType = {
    Maintain: "maintain",
    Request: "request",
};
exports.SpotFleetSpotCapacityRebalanceReplacementStrategy = {
    Launch: "launch",
    LaunchBeforeTerminate: "launch-before-terminate",
};
exports.SpotFleetSpotPlacementTenancy = {
    Dedicated: "dedicated",
    Default: "default",
    Host: "host",
};
exports.SpotFleetTagSpecificationResourceType = {
    ClientVpnEndpoint: "client-vpn-endpoint",
    CustomerGateway: "customer-gateway",
    DedicatedHost: "dedicated-host",
    DhcpOptions: "dhcp-options",
    EgressOnlyInternetGateway: "egress-only-internet-gateway",
    ElasticGpu: "elastic-gpu",
    ElasticIp: "elastic-ip",
    ExportImageTask: "export-image-task",
    ExportInstanceTask: "export-instance-task",
    Fleet: "fleet",
    FpgaImage: "fpga-image",
    HostReservation: "host-reservation",
    Image: "image",
    ImportImageTask: "import-image-task",
    ImportSnapshotTask: "import-snapshot-task",
    Instance: "instance",
    InternetGateway: "internet-gateway",
    KeyPair: "key-pair",
    LaunchTemplate: "launch-template",
    LocalGatewayRouteTableVpcAssociation: "local-gateway-route-table-vpc-association",
    Natgateway: "natgateway",
    NetworkAcl: "network-acl",
    NetworkInsightsAnalysis: "network-insights-analysis",
    NetworkInsightsPath: "network-insights-path",
    NetworkInterface: "network-interface",
    PlacementGroup: "placement-group",
    ReservedInstances: "reserved-instances",
    RouteTable: "route-table",
    SecurityGroup: "security-group",
    Snapshot: "snapshot",
    SpotFleetRequest: "spot-fleet-request",
    SpotInstancesRequest: "spot-instances-request",
    Subnet: "subnet",
    TrafficMirrorFilter: "traffic-mirror-filter",
    TrafficMirrorSession: "traffic-mirror-session",
    TrafficMirrorTarget: "traffic-mirror-target",
    TransitGateway: "transit-gateway",
    TransitGatewayAttachment: "transit-gateway-attachment",
    TransitGatewayConnectPeer: "transit-gateway-connect-peer",
    TransitGatewayMulticastDomain: "transit-gateway-multicast-domain",
    TransitGatewayRouteTable: "transit-gateway-route-table",
    Volume: "volume",
    Vpc: "vpc",
    VpcFlowLog: "vpc-flow-log",
    VpcPeeringConnection: "vpc-peering-connection",
    VpnConnection: "vpn-connection",
    VpnGateway: "vpn-gateway",
};
exports.VpcBlockPublicAccessExclusionInternetGatewayExclusionMode = {
    AllowBidirectional: "allow-bidirectional",
    AllowEgress: "allow-egress",
};
exports.VpcBlockPublicAccessOptionsInternetGatewayBlockMode = {
    BlockBidirectional: "block-bidirectional",
    BlockIngress: "block-ingress",
};
exports.VpcEndpointDnsOptionsSpecificationDnsRecordIpType = {
    Ipv4: "ipv4",
    Ipv6: "ipv6",
    Dualstack: "dualstack",
    ServiceDefined: "service-defined",
    NotSpecified: "not-specified",
};
exports.VpcEndpointDnsOptionsSpecificationPrivateDnsOnlyForInboundResolverEndpoint = {
    OnlyInboundResolver: "OnlyInboundResolver",
    AllResolvers: "AllResolvers",
    NotSpecified: "NotSpecified",
};
exports.VpcEndpointIpAddressType = {
    Ipv4: "ipv4",
    Ipv6: "ipv6",
    Dualstack: "dualstack",
    NotSpecified: "not-specified",
};
exports.VpcEndpointType = {
    Interface: "Interface",
    Gateway: "Gateway",
    GatewayLoadBalancer: "GatewayLoadBalancer",
    ServiceNetwork: "ServiceNetwork",
    Resource: "Resource",
};
exports.VpnConnectionCloudwatchLogOptionsSpecificationLogOutputFormat = {
    Json: "json",
    Text: "text",
};
exports.VpnConnectionIkeVersionsRequestListValueValue = {
    Ikev1: "ikev1",
    Ikev2: "ikev2",
};
exports.VpnConnectionPhase1EncryptionAlgorithmsRequestListValueValue = {
    Aes128: "AES128",
    Aes256: "AES256",
    Aes128Gcm16: "AES128-GCM-16",
    Aes256Gcm16: "AES256-GCM-16",
};
exports.VpnConnectionPhase1IntegrityAlgorithmsRequestListValueValue = {
    Sha1: "SHA1",
    Sha2256: "SHA2-256",
    Sha2384: "SHA2-384",
    Sha2512: "SHA2-512",
};
exports.VpnConnectionPhase2EncryptionAlgorithmsRequestListValueValue = {
    Aes128: "AES128",
    Aes256: "AES256",
    Aes128Gcm16: "AES128-GCM-16",
    Aes256Gcm16: "AES256-GCM-16",
};
exports.VpnConnectionPhase2IntegrityAlgorithmsRequestListValueValue = {
    Sha1: "SHA1",
    Sha2256: "SHA2-256",
    Sha2384: "SHA2-384",
    Sha2512: "SHA2-512",
};
exports.VpnConnectionVpnTunnelOptionsSpecificationDpdTimeoutAction = {
    Clear: "clear",
    None: "none",
    Restart: "restart",
};
exports.VpnConnectionVpnTunnelOptionsSpecificationStartupAction = {
    Add: "add",
    Start: "start",
};
