// @pulumi/aws-native/types/enums/eks/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityProviderConfigType = exports.ClusterUpgradePolicySupportType = exports.ClusterLoggingTypeConfigType = exports.ClusterKubernetesNetworkConfigIpFamily = exports.ClusterAccessConfigAuthenticationMode = exports.AddonResolveConflicts = exports.AccessEntryAccessScopeType = void 0;
exports.AccessEntryAccessScopeType = {
    Namespace: "namespace",
    Cluster: "cluster",
};
exports.AddonResolveConflicts = {
    None: "NONE",
    Overwrite: "OVERWRITE",
    Preserve: "PRESERVE",
};
exports.ClusterAccessConfigAuthenticationMode = {
    ConfigMap: "CONFIG_MAP",
    ApiAndConfigMap: "API_AND_CONFIG_MAP",
    Api: "API",
};
exports.ClusterKubernetesNetworkConfigIpFamily = {
    Ipv4: "ipv4",
    Ipv6: "ipv6",
};
exports.ClusterLoggingTypeConfigType = {
    Api: "api",
    Audit: "audit",
    Authenticator: "authenticator",
    ControllerManager: "controllerManager",
    Scheduler: "scheduler",
};
exports.ClusterUpgradePolicySupportType = {
    Standard: "STANDARD",
    Extended: "EXTENDED",
};
exports.IdentityProviderConfigType = {
    Oidc: "oidc",
};
