// @pulumi/aws-native/b2bi/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transformer = exports.Profile = exports.Partnership = exports.getTransformerOutput = exports.getTransformer = exports.getProfileOutput = exports.getProfile = exports.getPartnershipOutput = exports.getPartnership = exports.getCapabilityOutput = exports.getCapability = exports.Capability = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Capability = null;
utilities.lazyLoad(exports, ["Capability"], () => require("./capability"));
exports.getCapability = null;
exports.getCapabilityOutput = null;
utilities.lazyLoad(exports, ["getCapability", "getCapabilityOutput"], () => require("./getCapability"));
exports.getPartnership = null;
exports.getPartnershipOutput = null;
utilities.lazyLoad(exports, ["getPartnership", "getPartnershipOutput"], () => require("./getPartnership"));
exports.getProfile = null;
exports.getProfileOutput = null;
utilities.lazyLoad(exports, ["getProfile", "getProfileOutput"], () => require("./getProfile"));
exports.getTransformer = null;
exports.getTransformerOutput = null;
utilities.lazyLoad(exports, ["getTransformer", "getTransformerOutput"], () => require("./getTransformer"));
exports.Partnership = null;
utilities.lazyLoad(exports, ["Partnership"], () => require("./partnership"));
exports.Profile = null;
utilities.lazyLoad(exports, ["Profile"], () => require("./profile"));
exports.Transformer = null;
utilities.lazyLoad(exports, ["Transformer"], () => require("./transformer"));
// Export enums:
__exportStar(require("../types/enums/b2bi"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:b2bi:Capability":
                return new exports.Capability(name, undefined, { urn });
            case "aws-native:b2bi:Partnership":
                return new exports.Partnership(name, undefined, { urn });
            case "aws-native:b2bi:Profile":
                return new exports.Profile(name, undefined, { urn });
            case "aws-native:b2bi:Transformer":
                return new exports.Transformer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "b2bi", _module);
