// @pulumi/aws-native/codepipeline/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pipeline = exports.getPipelineOutput = exports.getPipeline = exports.getCustomActionTypeOutput = exports.getCustomActionType = exports.CustomActionType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CustomActionType = null;
utilities.lazyLoad(exports, ["CustomActionType"], () => require("./customActionType"));
exports.getCustomActionType = null;
exports.getCustomActionTypeOutput = null;
utilities.lazyLoad(exports, ["getCustomActionType", "getCustomActionTypeOutput"], () => require("./getCustomActionType"));
exports.getPipeline = null;
exports.getPipelineOutput = null;
utilities.lazyLoad(exports, ["getPipeline", "getPipelineOutput"], () => require("./getPipeline"));
exports.Pipeline = null;
utilities.lazyLoad(exports, ["Pipeline"], () => require("./pipeline"));
// Export enums:
__exportStar(require("../types/enums/codepipeline"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:codepipeline:CustomActionType":
                return new exports.CustomActionType(name, undefined, { urn });
            case "aws-native:codepipeline:Pipeline":
                return new exports.Pipeline(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "codepipeline", _module);
