// @pulumi/aws-native/types/enums/appsync/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceApiAssociationStatus = exports.SourceApiAssociationConfigMergeType = exports.ResolverMetricsConfig = exports.DataSourceMetricsConfig = exports.ChannelNamespaceAuthenticationType = exports.ApiEventLogLevel = exports.ApiAuthenticationType = void 0;
exports.ApiAuthenticationType = {
    AmazonCognitoUserPools: "AMAZON_COGNITO_USER_POOLS",
    AwsIam: "AWS_IAM",
    ApiKey: "API_KEY",
    OpenidConnect: "OPENID_CONNECT",
    AwsLambda: "AWS_LAMBDA",
};
exports.ApiEventLogLevel = {
    None: "NONE",
    Error: "ERROR",
    All: "ALL",
    Info: "INFO",
    Debug: "DEBUG",
};
exports.ChannelNamespaceAuthenticationType = {
    AmazonCognitoUserPools: "AMAZON_COGNITO_USER_POOLS",
    AwsIam: "AWS_IAM",
    ApiKey: "API_KEY",
    OpenidConnect: "OPENID_CONNECT",
    AwsLambda: "AWS_LAMBDA",
};
exports.DataSourceMetricsConfig = {
    Disabled: "DISABLED",
    Enabled: "ENABLED",
};
exports.ResolverMetricsConfig = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.SourceApiAssociationConfigMergeType = {
    AutoMerge: "AUTO_MERGE",
    ManualMerge: "MANUAL_MERGE",
};
exports.SourceApiAssociationStatus = {
    MergeScheduled: "MERGE_SCHEDULED",
    MergeFailed: "MERGE_FAILED",
    MergeSuccess: "MERGE_SUCCESS",
    MergeInProgress: "MERGE_IN_PROGRESS",
    AutoMergeScheduleFailed: "AUTO_MERGE_SCHEDULE_FAILED",
    DeletionScheduled: "DELETION_SCHEDULED",
    DeletionInProgress: "DELETION_IN_PROGRESS",
    DeletionFailed: "DELETION_FAILED",
};
