// @pulumi/aws-native/medialive/signalMap.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalMap = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::MediaLive::SignalMap Resource Type
 */
class SignalMap extends pulumi.CustomResource {
    /**
     * Get an existing SignalMap resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SignalMap(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SignalMap.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SignalMap.__pulumiType;
    }
    /**
     * Create a SignalMap resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.discoveryEntryPointArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'discoveryEntryPointArn'");
            }
            resourceInputs["cloudWatchAlarmTemplateGroupIdentifiers"] = args ? args.cloudWatchAlarmTemplateGroupIdentifiers : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["discoveryEntryPointArn"] = args ? args.discoveryEntryPointArn : undefined;
            resourceInputs["eventBridgeRuleTemplateGroupIdentifiers"] = args ? args.eventBridgeRuleTemplateGroupIdentifiers : undefined;
            resourceInputs["forceRediscovery"] = args ? args.forceRediscovery : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["awsId"] = undefined /*out*/;
            resourceInputs["cloudWatchAlarmTemplateGroupIds"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["errorMessage"] = undefined /*out*/;
            resourceInputs["eventBridgeRuleTemplateGroupIds"] = undefined /*out*/;
            resourceInputs["failedMediaResourceMap"] = undefined /*out*/;
            resourceInputs["identifier"] = undefined /*out*/;
            resourceInputs["lastDiscoveredAt"] = undefined /*out*/;
            resourceInputs["lastSuccessfulMonitorDeployment"] = undefined /*out*/;
            resourceInputs["mediaResourceMap"] = undefined /*out*/;
            resourceInputs["modifiedAt"] = undefined /*out*/;
            resourceInputs["monitorChangesPendingDeployment"] = undefined /*out*/;
            resourceInputs["monitorDeployment"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["awsId"] = undefined /*out*/;
            resourceInputs["cloudWatchAlarmTemplateGroupIdentifiers"] = undefined /*out*/;
            resourceInputs["cloudWatchAlarmTemplateGroupIds"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["discoveryEntryPointArn"] = undefined /*out*/;
            resourceInputs["errorMessage"] = undefined /*out*/;
            resourceInputs["eventBridgeRuleTemplateGroupIdentifiers"] = undefined /*out*/;
            resourceInputs["eventBridgeRuleTemplateGroupIds"] = undefined /*out*/;
            resourceInputs["failedMediaResourceMap"] = undefined /*out*/;
            resourceInputs["forceRediscovery"] = undefined /*out*/;
            resourceInputs["identifier"] = undefined /*out*/;
            resourceInputs["lastDiscoveredAt"] = undefined /*out*/;
            resourceInputs["lastSuccessfulMonitorDeployment"] = undefined /*out*/;
            resourceInputs["mediaResourceMap"] = undefined /*out*/;
            resourceInputs["modifiedAt"] = undefined /*out*/;
            resourceInputs["monitorChangesPendingDeployment"] = undefined /*out*/;
            resourceInputs["monitorDeployment"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["tags.*"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(SignalMap.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SignalMap = SignalMap;
/** @internal */
SignalMap.__pulumiType = 'aws-native:medialive:SignalMap';
