// @pulumi/aws-native/sns/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopicPolicy = exports.TopicInlinePolicy = exports.Topic = exports.Subscription = exports.getTopicPolicyOutput = exports.getTopicPolicy = exports.getTopicInlinePolicyOutput = exports.getTopicInlinePolicy = exports.getTopicOutput = exports.getTopic = exports.getSubscriptionOutput = exports.getSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getSubscription = null;
exports.getSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getSubscription", "getSubscriptionOutput"], () => require("./getSubscription"));
exports.getTopic = null;
exports.getTopicOutput = null;
utilities.lazyLoad(exports, ["getTopic", "getTopicOutput"], () => require("./getTopic"));
exports.getTopicInlinePolicy = null;
exports.getTopicInlinePolicyOutput = null;
utilities.lazyLoad(exports, ["getTopicInlinePolicy", "getTopicInlinePolicyOutput"], () => require("./getTopicInlinePolicy"));
exports.getTopicPolicy = null;
exports.getTopicPolicyOutput = null;
utilities.lazyLoad(exports, ["getTopicPolicy", "getTopicPolicyOutput"], () => require("./getTopicPolicy"));
exports.Subscription = null;
utilities.lazyLoad(exports, ["Subscription"], () => require("./subscription"));
exports.Topic = null;
utilities.lazyLoad(exports, ["Topic"], () => require("./topic"));
exports.TopicInlinePolicy = null;
utilities.lazyLoad(exports, ["TopicInlinePolicy"], () => require("./topicInlinePolicy"));
exports.TopicPolicy = null;
utilities.lazyLoad(exports, ["TopicPolicy"], () => require("./topicPolicy"));
// Export enums:
__exportStar(require("../types/enums/sns"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:sns:Subscription":
                return new exports.Subscription(name, undefined, { urn });
            case "aws-native:sns:Topic":
                return new exports.Topic(name, undefined, { urn });
            case "aws-native:sns:TopicInlinePolicy":
                return new exports.TopicInlinePolicy(name, undefined, { urn });
            case "aws-native:sns:TopicPolicy":
                return new exports.TopicPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "sns", _module);
