// @pulumi/aws-native/ssmquicksetup/configurationManager.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationManager = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::SSMQuickSetup::ConfigurationManager Resource Type
 */
class ConfigurationManager extends pulumi.CustomResource {
    /**
     * Get an existing ConfigurationManager resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConfigurationManager(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConfigurationManager.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConfigurationManager.__pulumiType;
    }
    /**
     * Create a ConfigurationManager resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.configurationDefinitions === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configurationDefinitions'");
            }
            resourceInputs["configurationDefinitions"] = args ? args.configurationDefinitions : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["lastModifiedAt"] = undefined /*out*/;
            resourceInputs["managerArn"] = undefined /*out*/;
            resourceInputs["statusSummaries"] = undefined /*out*/;
        }
        else {
            resourceInputs["configurationDefinitions"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["lastModifiedAt"] = undefined /*out*/;
            resourceInputs["managerArn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["statusSummaries"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["configurationDefinitions[*].type", "configurationDefinitions[*].typeVersion"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(ConfigurationManager.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ConfigurationManager = ConfigurationManager;
/** @internal */
ConfigurationManager.__pulumiType = 'aws-native:ssmquicksetup:ConfigurationManager';
