// @pulumi/aws-native/types/enums/cognito/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPoolTier = exports.ManagedLoginBrandingExtensionType = exports.ManagedLoginBrandingColorModeType = exports.ManagedLoginBrandingCategoryType = void 0;
exports.ManagedLoginBrandingCategoryType = {
    FaviconIco: "FAVICON_ICO",
    FaviconSvg: "FAVICON_SVG",
    EmailGraphic: "EMAIL_GRAPHIC",
    SmsGraphic: "SMS_GRAPHIC",
    AuthAppGraphic: "AUTH_APP_GRAPHIC",
    PasswordGraphic: "PASSWORD_GRAPHIC",
    PasskeyGraphic: "PASSKEY_GRAPHIC",
    PageHeaderLogo: "PAGE_HEADER_LOGO",
    PageHeaderBackground: "PAGE_HEADER_BACKGROUND",
    PageFooterLogo: "PAGE_FOOTER_LOGO",
    PageFooterBackground: "PAGE_FOOTER_BACKGROUND",
    PageBackground: "PAGE_BACKGROUND",
    FormBackground: "FORM_BACKGROUND",
    FormLogo: "FORM_LOGO",
    IdpButtonIcon: "IDP_BUTTON_ICON",
};
exports.ManagedLoginBrandingColorModeType = {
    Light: "LIGHT",
    Dark: "DARK",
    Dynamic: "DYNAMIC",
};
exports.ManagedLoginBrandingExtensionType = {
    Ico: "ICO",
    Jpeg: "JPEG",
    Png: "PNG",
    Svg: "SVG",
    Webp: "WEBP",
};
exports.UserPoolTier = {
    Lite: "LITE",
    Essentials: "ESSENTIALS",
    Plus: "PLUS",
};
