// @pulumi/aws-native/synthetics/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Group = exports.getGroupOutput = exports.getGroup = exports.getCanaryOutput = exports.getCanary = exports.Canary = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Canary = null;
utilities.lazyLoad(exports, ["Canary"], () => require("./canary"));
exports.getCanary = null;
exports.getCanaryOutput = null;
utilities.lazyLoad(exports, ["getCanary", "getCanaryOutput"], () => require("./getCanary"));
exports.getGroup = null;
exports.getGroupOutput = null;
utilities.lazyLoad(exports, ["getGroup", "getGroupOutput"], () => require("./getGroup"));
exports.Group = null;
utilities.lazyLoad(exports, ["Group"], () => require("./group"));
// Export enums:
__exportStar(require("../types/enums/synthetics"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:synthetics:Canary":
                return new exports.Canary(name, undefined, { urn });
            case "aws-native:synthetics:Group":
                return new exports.Group(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "synthetics", _module);
