// @pulumi/aws-native/types/enums/datazone/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfileUserType = exports.UserProfileType = exports.UserProfileStatus = exports.ProjectMembershipUserDesignation = exports.GroupProfileStatus = exports.EnvironmentStatus = exports.DomainUserAssignment = exports.DomainStatus = exports.DomainAuthType = exports.DataSourceStatus = exports.DataSourceFilterExpressionType = exports.DataSourceEnableSetting = void 0;
exports.DataSourceEnableSetting = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.DataSourceFilterExpressionType = {
    Include: "INCLUDE",
    Exclude: "EXCLUDE",
};
exports.DataSourceStatus = {
    Creating: "CREATING",
    FailedCreation: "FAILED_CREATION",
    Ready: "READY",
    Updating: "UPDATING",
    FailedUpdate: "FAILED_UPDATE",
    Running: "RUNNING",
    Deleting: "DELETING",
    FailedDeletion: "FAILED_DELETION",
};
exports.DomainAuthType = {
    IamIdc: "IAM_IDC",
    Disabled: "DISABLED",
};
exports.DomainStatus = {
    Creating: "CREATING",
    Available: "AVAILABLE",
    CreationFailed: "CREATION_FAILED",
    Deleting: "DELETING",
    Deleted: "DELETED",
    DeletionFailed: "DELETION_FAILED",
};
exports.DomainUserAssignment = {
    Automatic: "AUTOMATIC",
    Manual: "MANUAL",
};
exports.EnvironmentStatus = {
    Active: "ACTIVE",
    Creating: "CREATING",
    Updating: "UPDATING",
    Deleting: "DELETING",
    CreateFailed: "CREATE_FAILED",
    UpdateFailed: "UPDATE_FAILED",
    DeleteFailed: "DELETE_FAILED",
    ValidationFailed: "VALIDATION_FAILED",
    Suspended: "SUSPENDED",
    Disabled: "DISABLED",
    Expired: "EXPIRED",
    Deleted: "DELETED",
    Inaccessible: "INACCESSIBLE",
};
exports.GroupProfileStatus = {
    Assigned: "ASSIGNED",
    NotAssigned: "NOT_ASSIGNED",
};
exports.ProjectMembershipUserDesignation = {
    ProjectOwner: "PROJECT_OWNER",
    ProjectContributor: "PROJECT_CONTRIBUTOR",
    ProjectCatalogViewer: "PROJECT_CATALOG_VIEWER",
    ProjectCatalogConsumer: "PROJECT_CATALOG_CONSUMER",
    ProjectCatalogSteward: "PROJECT_CATALOG_STEWARD",
};
exports.UserProfileStatus = {
    Assigned: "ASSIGNED",
    NotAssigned: "NOT_ASSIGNED",
    Activated: "ACTIVATED",
    Deactivated: "DEACTIVATED",
};
exports.UserProfileType = {
    Iam: "IAM",
    Sso: "SSO",
};
exports.UserProfileUserType = {
    IamUser: "IAM_USER",
    IamRole: "IAM_ROLE",
    SsoUser: "SSO_USER",
};
