// @pulumi/aws-native/ses/mailManagerTrafficPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailManagerTrafficPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::SES::MailManagerTrafficPolicy Resource Type
 */
class MailManagerTrafficPolicy extends pulumi.CustomResource {
    /**
     * Get an existing MailManagerTrafficPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MailManagerTrafficPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MailManagerTrafficPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MailManagerTrafficPolicy.__pulumiType;
    }
    /**
     * Create a MailManagerTrafficPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.defaultAction === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultAction'");
            }
            if ((!args || args.policyStatements === undefined) && !opts.urn) {
                throw new Error("Missing required property 'policyStatements'");
            }
            resourceInputs["defaultAction"] = args ? args.defaultAction : undefined;
            resourceInputs["maxMessageSizeBytes"] = args ? args.maxMessageSizeBytes : undefined;
            resourceInputs["policyStatements"] = args ? args.policyStatements : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["trafficPolicyName"] = args ? args.trafficPolicyName : undefined;
            resourceInputs["trafficPolicyArn"] = undefined /*out*/;
            resourceInputs["trafficPolicyId"] = undefined /*out*/;
        }
        else {
            resourceInputs["defaultAction"] = undefined /*out*/;
            resourceInputs["maxMessageSizeBytes"] = undefined /*out*/;
            resourceInputs["policyStatements"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["trafficPolicyArn"] = undefined /*out*/;
            resourceInputs["trafficPolicyId"] = undefined /*out*/;
            resourceInputs["trafficPolicyName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MailManagerTrafficPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MailManagerTrafficPolicy = MailManagerTrafficPolicy;
/** @internal */
MailManagerTrafficPolicy.__pulumiType = 'aws-native:ses:MailManagerTrafficPolicy';
