// @pulumi/aws-native/ses/mailManagerArchive.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailManagerArchive = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::SES::MailManagerArchive Resource Type
 */
class MailManagerArchive extends pulumi.CustomResource {
    /**
     * Get an existing MailManagerArchive resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MailManagerArchive(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MailManagerArchive.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MailManagerArchive.__pulumiType;
    }
    /**
     * Create a MailManagerArchive resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["archiveName"] = args ? args.archiveName : undefined;
            resourceInputs["kmsKeyArn"] = args ? args.kmsKeyArn : undefined;
            resourceInputs["retention"] = args ? args.retention : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["archiveArn"] = undefined /*out*/;
            resourceInputs["archiveId"] = undefined /*out*/;
            resourceInputs["archiveState"] = undefined /*out*/;
        }
        else {
            resourceInputs["archiveArn"] = undefined /*out*/;
            resourceInputs["archiveId"] = undefined /*out*/;
            resourceInputs["archiveName"] = undefined /*out*/;
            resourceInputs["archiveState"] = undefined /*out*/;
            resourceInputs["kmsKeyArn"] = undefined /*out*/;
            resourceInputs["retention"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["kmsKeyArn"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(MailManagerArchive.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MailManagerArchive = MailManagerArchive;
/** @internal */
MailManagerArchive.__pulumiType = 'aws-native:ses:MailManagerArchive';
