// @pulumi/aws-native/ses/mailManagerIngressPoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailManagerIngressPoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::SES::MailManagerIngressPoint Resource Type
 */
class MailManagerIngressPoint extends pulumi.CustomResource {
    /**
     * Get an existing MailManagerIngressPoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MailManagerIngressPoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MailManagerIngressPoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MailManagerIngressPoint.__pulumiType;
    }
    /**
     * Create a MailManagerIngressPoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.ruleSetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ruleSetId'");
            }
            if ((!args || args.trafficPolicyId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'trafficPolicyId'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["ingressPointConfiguration"] = args ? args.ingressPointConfiguration : undefined;
            resourceInputs["ingressPointName"] = args ? args.ingressPointName : undefined;
            resourceInputs["ruleSetId"] = args ? args.ruleSetId : undefined;
            resourceInputs["statusToUpdate"] = args ? args.statusToUpdate : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["trafficPolicyId"] = args ? args.trafficPolicyId : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["aRecord"] = undefined /*out*/;
            resourceInputs["ingressPointArn"] = undefined /*out*/;
            resourceInputs["ingressPointId"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        else {
            resourceInputs["aRecord"] = undefined /*out*/;
            resourceInputs["ingressPointArn"] = undefined /*out*/;
            resourceInputs["ingressPointConfiguration"] = undefined /*out*/;
            resourceInputs["ingressPointId"] = undefined /*out*/;
            resourceInputs["ingressPointName"] = undefined /*out*/;
            resourceInputs["ruleSetId"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusToUpdate"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["trafficPolicyId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["type"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(MailManagerIngressPoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MailManagerIngressPoint = MailManagerIngressPoint;
/** @internal */
MailManagerIngressPoint.__pulumiType = 'aws-native:ses:MailManagerIngressPoint';
