// @pulumi/aws-native/sqs/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueInlinePolicy = exports.Queue = exports.getQueueInlinePolicyOutput = exports.getQueueInlinePolicy = exports.getQueueOutput = exports.getQueue = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getQueue = null;
exports.getQueueOutput = null;
utilities.lazyLoad(exports, ["getQueue", "getQueueOutput"], () => require("./getQueue"));
exports.getQueueInlinePolicy = null;
exports.getQueueInlinePolicyOutput = null;
utilities.lazyLoad(exports, ["getQueueInlinePolicy", "getQueueInlinePolicyOutput"], () => require("./getQueueInlinePolicy"));
exports.Queue = null;
utilities.lazyLoad(exports, ["Queue"], () => require("./queue"));
exports.QueueInlinePolicy = null;
utilities.lazyLoad(exports, ["QueueInlinePolicy"], () => require("./queueInlinePolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:sqs:Queue":
                return new exports.Queue(name, undefined, { urn });
            case "aws-native:sqs:QueueInlinePolicy":
                return new exports.QueueInlinePolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "sqs", _module);
