// @pulumi/aws-native/kafkaconnect/workerConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkerConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The configuration of the workers, which are the processes that run the connector logic.
 */
class WorkerConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing WorkerConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkerConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkerConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkerConfiguration.__pulumiType;
    }
    /**
     * Create a WorkerConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.propertiesFileContent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'propertiesFileContent'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["propertiesFileContent"] = args ? args.propertiesFileContent : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["revision"] = undefined /*out*/;
            resourceInputs["workerConfigurationArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["propertiesFileContent"] = undefined /*out*/;
            resourceInputs["revision"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["workerConfigurationArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["description", "name", "propertiesFileContent"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(WorkerConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkerConfiguration = WorkerConfiguration;
/** @internal */
WorkerConfiguration.__pulumiType = 'aws-native:kafkaconnect:WorkerConfiguration';
