// @pulumi/aws-native/comprehend/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getFlywheelOutput = exports.getFlywheel = exports.getDocumentClassifierOutput = exports.getDocumentClassifier = exports.Flywheel = exports.DocumentClassifier = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DocumentClassifier = null;
utilities.lazyLoad(exports, ["DocumentClassifier"], () => require("./documentClassifier"));
exports.Flywheel = null;
utilities.lazyLoad(exports, ["Flywheel"], () => require("./flywheel"));
exports.getDocumentClassifier = null;
exports.getDocumentClassifierOutput = null;
utilities.lazyLoad(exports, ["getDocumentClassifier", "getDocumentClassifierOutput"], () => require("./getDocumentClassifier"));
exports.getFlywheel = null;
exports.getFlywheelOutput = null;
utilities.lazyLoad(exports, ["getFlywheel", "getFlywheelOutput"], () => require("./getFlywheel"));
// Export enums:
__exportStar(require("../types/enums/comprehend"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:comprehend:DocumentClassifier":
                return new exports.DocumentClassifier(name, undefined, { urn });
            case "aws-native:comprehend:Flywheel":
                return new exports.Flywheel(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "comprehend", _module);
