// @pulumi/aws-native/codestarconnections/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncConfiguration = exports.RepositoryLink = exports.getSyncConfigurationOutput = exports.getSyncConfiguration = exports.getRepositoryLinkOutput = exports.getRepositoryLink = exports.getConnectionOutput = exports.getConnection = exports.Connection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.getConnection = null;
exports.getConnectionOutput = null;
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
exports.getRepositoryLink = null;
exports.getRepositoryLinkOutput = null;
utilities.lazyLoad(exports, ["getRepositoryLink", "getRepositoryLinkOutput"], () => require("./getRepositoryLink"));
exports.getSyncConfiguration = null;
exports.getSyncConfigurationOutput = null;
utilities.lazyLoad(exports, ["getSyncConfiguration", "getSyncConfigurationOutput"], () => require("./getSyncConfiguration"));
exports.RepositoryLink = null;
utilities.lazyLoad(exports, ["RepositoryLink"], () => require("./repositoryLink"));
exports.SyncConfiguration = null;
utilities.lazyLoad(exports, ["SyncConfiguration"], () => require("./syncConfiguration"));
// Export enums:
__exportStar(require("../types/enums/codestarconnections"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:codestarconnections:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "aws-native:codestarconnections:RepositoryLink":
                return new exports.RepositoryLink(name, undefined, { urn });
            case "aws-native:codestarconnections:SyncConfiguration":
                return new exports.SyncConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "codestarconnections", _module);
