// @pulumi/aws-native/codestarconnections/syncConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Schema for AWS::CodeStarConnections::SyncConfiguration resource which is used to enables an AWS resource to be synchronized from a source-provider.
 */
class SyncConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing SyncConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SyncConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SyncConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SyncConfiguration.__pulumiType;
    }
    /**
     * Create a SyncConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.branch === undefined) && !opts.urn) {
                throw new Error("Missing required property 'branch'");
            }
            if ((!args || args.configFile === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configFile'");
            }
            if ((!args || args.repositoryLinkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'repositoryLinkId'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            if ((!args || args.syncType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'syncType'");
            }
            resourceInputs["branch"] = args ? args.branch : undefined;
            resourceInputs["configFile"] = args ? args.configFile : undefined;
            resourceInputs["publishDeploymentStatus"] = args ? args.publishDeploymentStatus : undefined;
            resourceInputs["repositoryLinkId"] = args ? args.repositoryLinkId : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["syncType"] = args ? args.syncType : undefined;
            resourceInputs["triggerResourceUpdateOn"] = args ? args.triggerResourceUpdateOn : undefined;
            resourceInputs["ownerId"] = undefined /*out*/;
            resourceInputs["providerType"] = undefined /*out*/;
            resourceInputs["repositoryName"] = undefined /*out*/;
        }
        else {
            resourceInputs["branch"] = undefined /*out*/;
            resourceInputs["configFile"] = undefined /*out*/;
            resourceInputs["ownerId"] = undefined /*out*/;
            resourceInputs["providerType"] = undefined /*out*/;
            resourceInputs["publishDeploymentStatus"] = undefined /*out*/;
            resourceInputs["repositoryLinkId"] = undefined /*out*/;
            resourceInputs["repositoryName"] = undefined /*out*/;
            resourceInputs["resourceName"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["syncType"] = undefined /*out*/;
            resourceInputs["triggerResourceUpdateOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["resourceName", "syncType"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(SyncConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SyncConfiguration = SyncConfiguration;
/** @internal */
SyncConfiguration.__pulumiType = 'aws-native:codestarconnections:SyncConfiguration';
