// @pulumi/aws-native/finspace/environment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Environment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An example resource schema demonstrating some basic constructs and validation rules.
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const finSpaceEnvironment = new aws_native.finspace.Environment("finSpaceEnvironment", {
 *     name: "MyEnvironment",
 *     kmsKeyId: "arn:aws:kms:us-east-1:123456789012:key/44efed01-30d0-4b39-80e7-165d5ed34524",
 *     federationMode: aws_native.finspace.EnvironmentFederationMode.Local,
 * });
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const finSpaceEnvironment = new aws_native.finspace.Environment("finSpaceEnvironment", {
 *     name: "MyEnvironment",
 *     kmsKeyId: "arn:aws:kms:us-east-1:123456789012:key/44efed01-30d0-4b39-80e7-165d5ed34524",
 *     federationMode: aws_native.finspace.EnvironmentFederationMode.Local,
 * });
 *
 * ```
 */
class Environment extends pulumi.CustomResource {
    /**
     * Get an existing Environment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Environment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Environment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Environment.__pulumiType;
    }
    /**
     * Create a Environment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["dataBundles"] = args ? args.dataBundles : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["federationMode"] = args ? args.federationMode : undefined;
            resourceInputs["federationParameters"] = args ? args.federationParameters : undefined;
            resourceInputs["kmsKeyId"] = args ? args.kmsKeyId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["superuserParameters"] = args ? args.superuserParameters : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["awsAccountId"] = undefined /*out*/;
            resourceInputs["dedicatedServiceAccountId"] = undefined /*out*/;
            resourceInputs["environmentArn"] = undefined /*out*/;
            resourceInputs["environmentId"] = undefined /*out*/;
            resourceInputs["environmentUrl"] = undefined /*out*/;
            resourceInputs["sageMakerStudioDomainUrl"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        else {
            resourceInputs["awsAccountId"] = undefined /*out*/;
            resourceInputs["dataBundles"] = undefined /*out*/;
            resourceInputs["dedicatedServiceAccountId"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["environmentArn"] = undefined /*out*/;
            resourceInputs["environmentId"] = undefined /*out*/;
            resourceInputs["environmentUrl"] = undefined /*out*/;
            resourceInputs["federationMode"] = undefined /*out*/;
            resourceInputs["federationParameters"] = undefined /*out*/;
            resourceInputs["kmsKeyId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["sageMakerStudioDomainUrl"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["superuserParameters"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["dataBundles[*]", "federationParameters", "kmsKeyId", "superuserParameters", "tags[*]"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(Environment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Environment = Environment;
/** @internal */
Environment.__pulumiType = 'aws-native:finspace:Environment';
