// @pulumi/aws-native/controltower/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LandingZone = exports.getLandingZoneOutput = exports.getLandingZone = exports.getEnabledControlOutput = exports.getEnabledControl = exports.getEnabledBaselineOutput = exports.getEnabledBaseline = exports.EnabledControl = exports.EnabledBaseline = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.EnabledBaseline = null;
utilities.lazyLoad(exports, ["EnabledBaseline"], () => require("./enabledBaseline"));
exports.EnabledControl = null;
utilities.lazyLoad(exports, ["EnabledControl"], () => require("./enabledControl"));
exports.getEnabledBaseline = null;
exports.getEnabledBaselineOutput = null;
utilities.lazyLoad(exports, ["getEnabledBaseline", "getEnabledBaselineOutput"], () => require("./getEnabledBaseline"));
exports.getEnabledControl = null;
exports.getEnabledControlOutput = null;
utilities.lazyLoad(exports, ["getEnabledControl", "getEnabledControlOutput"], () => require("./getEnabledControl"));
exports.getLandingZone = null;
exports.getLandingZoneOutput = null;
utilities.lazyLoad(exports, ["getLandingZone", "getLandingZoneOutput"], () => require("./getLandingZone"));
exports.LandingZone = null;
utilities.lazyLoad(exports, ["LandingZone"], () => require("./landingZone"));
// Export enums:
__exportStar(require("../types/enums/controltower"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:controltower:EnabledBaseline":
                return new exports.EnabledBaseline(name, undefined, { urn });
            case "aws-native:controltower:EnabledControl":
                return new exports.EnabledControl(name, undefined, { urn });
            case "aws-native:controltower:LandingZone":
                return new exports.LandingZone(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "controltower", _module);
