// @pulumi/aws-native/types/enums/msk/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcConnectionAuthentication = exports.ReplicatorReplicationTopicNameConfigurationType = exports.ReplicatorReplicationStartingPositionType = exports.ReplicatorReplicationInfoTargetCompressionType = exports.ClusterStorageMode = exports.ClusterEnhancedMonitoring = exports.ClusterEncryptionInTransitClientBroker = void 0;
exports.ClusterEncryptionInTransitClientBroker = {
    Tls: "TLS",
    TlsPlaintext: "TLS_PLAINTEXT",
    Plaintext: "PLAINTEXT",
};
exports.ClusterEnhancedMonitoring = {
    Default: "DEFAULT",
    PerBroker: "PER_BROKER",
    PerTopicPerBroker: "PER_TOPIC_PER_BROKER",
    PerTopicPerPartition: "PER_TOPIC_PER_PARTITION",
};
exports.ClusterStorageMode = {
    Local: "LOCAL",
    Tiered: "TIERED",
};
exports.ReplicatorReplicationInfoTargetCompressionType = {
    None: "NONE",
    Gzip: "GZIP",
    Snappy: "SNAPPY",
    Lz4: "LZ4",
    Zstd: "ZSTD",
};
exports.ReplicatorReplicationStartingPositionType = {
    Latest: "LATEST",
    Earliest: "EARLIEST",
};
exports.ReplicatorReplicationTopicNameConfigurationType = {
    PrefixedWithSourceClusterAlias: "PREFIXED_WITH_SOURCE_CLUSTER_ALIAS",
    Identical: "IDENTICAL",
};
exports.VpcConnectionAuthentication = {
    SaslIam: "SASL_IAM",
    SaslScram: "SASL_SCRAM",
    Tls: "TLS",
};
