// @pulumi/aws-native/cognito/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPoolUserToGroupAttachment = exports.UserPoolUser = exports.UserPoolUiCustomizationAttachment = exports.UserPoolRiskConfigurationAttachment = exports.UserPoolResourceServer = exports.UserPoolIdentityProvider = exports.UserPoolGroup = exports.UserPoolDomain = exports.UserPoolClient = exports.UserPool = exports.ManagedLoginBranding = exports.LogDeliveryConfiguration = exports.IdentityPoolRoleAttachment = exports.IdentityPoolPrincipalTag = exports.IdentityPool = exports.getUserPoolUiCustomizationAttachmentOutput = exports.getUserPoolUiCustomizationAttachment = exports.getUserPoolRiskConfigurationAttachmentOutput = exports.getUserPoolRiskConfigurationAttachment = exports.getUserPoolResourceServerOutput = exports.getUserPoolResourceServer = exports.getUserPoolIdentityProviderOutput = exports.getUserPoolIdentityProvider = exports.getUserPoolGroupOutput = exports.getUserPoolGroup = exports.getUserPoolDomainOutput = exports.getUserPoolDomain = exports.getUserPoolClientOutput = exports.getUserPoolClient = exports.getUserPoolOutput = exports.getUserPool = exports.getManagedLoginBrandingOutput = exports.getManagedLoginBranding = exports.getLogDeliveryConfigurationOutput = exports.getLogDeliveryConfiguration = exports.getIdentityPoolRoleAttachmentOutput = exports.getIdentityPoolRoleAttachment = exports.getIdentityPoolPrincipalTagOutput = exports.getIdentityPoolPrincipalTag = exports.getIdentityPoolOutput = exports.getIdentityPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getIdentityPool = null;
exports.getIdentityPoolOutput = null;
utilities.lazyLoad(exports, ["getIdentityPool", "getIdentityPoolOutput"], () => require("./getIdentityPool"));
exports.getIdentityPoolPrincipalTag = null;
exports.getIdentityPoolPrincipalTagOutput = null;
utilities.lazyLoad(exports, ["getIdentityPoolPrincipalTag", "getIdentityPoolPrincipalTagOutput"], () => require("./getIdentityPoolPrincipalTag"));
exports.getIdentityPoolRoleAttachment = null;
exports.getIdentityPoolRoleAttachmentOutput = null;
utilities.lazyLoad(exports, ["getIdentityPoolRoleAttachment", "getIdentityPoolRoleAttachmentOutput"], () => require("./getIdentityPoolRoleAttachment"));
exports.getLogDeliveryConfiguration = null;
exports.getLogDeliveryConfigurationOutput = null;
utilities.lazyLoad(exports, ["getLogDeliveryConfiguration", "getLogDeliveryConfigurationOutput"], () => require("./getLogDeliveryConfiguration"));
exports.getManagedLoginBranding = null;
exports.getManagedLoginBrandingOutput = null;
utilities.lazyLoad(exports, ["getManagedLoginBranding", "getManagedLoginBrandingOutput"], () => require("./getManagedLoginBranding"));
exports.getUserPool = null;
exports.getUserPoolOutput = null;
utilities.lazyLoad(exports, ["getUserPool", "getUserPoolOutput"], () => require("./getUserPool"));
exports.getUserPoolClient = null;
exports.getUserPoolClientOutput = null;
utilities.lazyLoad(exports, ["getUserPoolClient", "getUserPoolClientOutput"], () => require("./getUserPoolClient"));
exports.getUserPoolDomain = null;
exports.getUserPoolDomainOutput = null;
utilities.lazyLoad(exports, ["getUserPoolDomain", "getUserPoolDomainOutput"], () => require("./getUserPoolDomain"));
exports.getUserPoolGroup = null;
exports.getUserPoolGroupOutput = null;
utilities.lazyLoad(exports, ["getUserPoolGroup", "getUserPoolGroupOutput"], () => require("./getUserPoolGroup"));
exports.getUserPoolIdentityProvider = null;
exports.getUserPoolIdentityProviderOutput = null;
utilities.lazyLoad(exports, ["getUserPoolIdentityProvider", "getUserPoolIdentityProviderOutput"], () => require("./getUserPoolIdentityProvider"));
exports.getUserPoolResourceServer = null;
exports.getUserPoolResourceServerOutput = null;
utilities.lazyLoad(exports, ["getUserPoolResourceServer", "getUserPoolResourceServerOutput"], () => require("./getUserPoolResourceServer"));
exports.getUserPoolRiskConfigurationAttachment = null;
exports.getUserPoolRiskConfigurationAttachmentOutput = null;
utilities.lazyLoad(exports, ["getUserPoolRiskConfigurationAttachment", "getUserPoolRiskConfigurationAttachmentOutput"], () => require("./getUserPoolRiskConfigurationAttachment"));
exports.getUserPoolUiCustomizationAttachment = null;
exports.getUserPoolUiCustomizationAttachmentOutput = null;
utilities.lazyLoad(exports, ["getUserPoolUiCustomizationAttachment", "getUserPoolUiCustomizationAttachmentOutput"], () => require("./getUserPoolUiCustomizationAttachment"));
exports.IdentityPool = null;
utilities.lazyLoad(exports, ["IdentityPool"], () => require("./identityPool"));
exports.IdentityPoolPrincipalTag = null;
utilities.lazyLoad(exports, ["IdentityPoolPrincipalTag"], () => require("./identityPoolPrincipalTag"));
exports.IdentityPoolRoleAttachment = null;
utilities.lazyLoad(exports, ["IdentityPoolRoleAttachment"], () => require("./identityPoolRoleAttachment"));
exports.LogDeliveryConfiguration = null;
utilities.lazyLoad(exports, ["LogDeliveryConfiguration"], () => require("./logDeliveryConfiguration"));
exports.ManagedLoginBranding = null;
utilities.lazyLoad(exports, ["ManagedLoginBranding"], () => require("./managedLoginBranding"));
exports.UserPool = null;
utilities.lazyLoad(exports, ["UserPool"], () => require("./userPool"));
exports.UserPoolClient = null;
utilities.lazyLoad(exports, ["UserPoolClient"], () => require("./userPoolClient"));
exports.UserPoolDomain = null;
utilities.lazyLoad(exports, ["UserPoolDomain"], () => require("./userPoolDomain"));
exports.UserPoolGroup = null;
utilities.lazyLoad(exports, ["UserPoolGroup"], () => require("./userPoolGroup"));
exports.UserPoolIdentityProvider = null;
utilities.lazyLoad(exports, ["UserPoolIdentityProvider"], () => require("./userPoolIdentityProvider"));
exports.UserPoolResourceServer = null;
utilities.lazyLoad(exports, ["UserPoolResourceServer"], () => require("./userPoolResourceServer"));
exports.UserPoolRiskConfigurationAttachment = null;
utilities.lazyLoad(exports, ["UserPoolRiskConfigurationAttachment"], () => require("./userPoolRiskConfigurationAttachment"));
exports.UserPoolUiCustomizationAttachment = null;
utilities.lazyLoad(exports, ["UserPoolUiCustomizationAttachment"], () => require("./userPoolUiCustomizationAttachment"));
exports.UserPoolUser = null;
utilities.lazyLoad(exports, ["UserPoolUser"], () => require("./userPoolUser"));
exports.UserPoolUserToGroupAttachment = null;
utilities.lazyLoad(exports, ["UserPoolUserToGroupAttachment"], () => require("./userPoolUserToGroupAttachment"));
// Export enums:
__exportStar(require("../types/enums/cognito"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:cognito:IdentityPool":
                return new exports.IdentityPool(name, undefined, { urn });
            case "aws-native:cognito:IdentityPoolPrincipalTag":
                return new exports.IdentityPoolPrincipalTag(name, undefined, { urn });
            case "aws-native:cognito:IdentityPoolRoleAttachment":
                return new exports.IdentityPoolRoleAttachment(name, undefined, { urn });
            case "aws-native:cognito:LogDeliveryConfiguration":
                return new exports.LogDeliveryConfiguration(name, undefined, { urn });
            case "aws-native:cognito:ManagedLoginBranding":
                return new exports.ManagedLoginBranding(name, undefined, { urn });
            case "aws-native:cognito:UserPool":
                return new exports.UserPool(name, undefined, { urn });
            case "aws-native:cognito:UserPoolClient":
                return new exports.UserPoolClient(name, undefined, { urn });
            case "aws-native:cognito:UserPoolDomain":
                return new exports.UserPoolDomain(name, undefined, { urn });
            case "aws-native:cognito:UserPoolGroup":
                return new exports.UserPoolGroup(name, undefined, { urn });
            case "aws-native:cognito:UserPoolIdentityProvider":
                return new exports.UserPoolIdentityProvider(name, undefined, { urn });
            case "aws-native:cognito:UserPoolResourceServer":
                return new exports.UserPoolResourceServer(name, undefined, { urn });
            case "aws-native:cognito:UserPoolRiskConfigurationAttachment":
                return new exports.UserPoolRiskConfigurationAttachment(name, undefined, { urn });
            case "aws-native:cognito:UserPoolUiCustomizationAttachment":
                return new exports.UserPoolUiCustomizationAttachment(name, undefined, { urn });
            case "aws-native:cognito:UserPoolUser":
                return new exports.UserPoolUser(name, undefined, { urn });
            case "aws-native:cognito:UserPoolUserToGroupAttachment":
                return new exports.UserPoolUserToGroupAttachment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "cognito", _module);
