// @pulumi/aws-native/cognito/userPoolUser.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPoolUser = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Cognito::UserPoolUser
 */
class UserPoolUser extends pulumi.CustomResource {
    /**
     * Get an existing UserPoolUser resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new UserPoolUser(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of UserPoolUser.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserPoolUser.__pulumiType;
    }
    /**
     * Create a UserPoolUser resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.userPoolId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'userPoolId'");
            }
            resourceInputs["clientMetadata"] = args ? args.clientMetadata : undefined;
            resourceInputs["desiredDeliveryMediums"] = args ? args.desiredDeliveryMediums : undefined;
            resourceInputs["forceAliasCreation"] = args ? args.forceAliasCreation : undefined;
            resourceInputs["messageAction"] = args ? args.messageAction : undefined;
            resourceInputs["userAttributes"] = args ? args.userAttributes : undefined;
            resourceInputs["userPoolId"] = args ? args.userPoolId : undefined;
            resourceInputs["username"] = args ? args.username : undefined;
            resourceInputs["validationData"] = args ? args.validationData : undefined;
        }
        else {
            resourceInputs["clientMetadata"] = undefined /*out*/;
            resourceInputs["desiredDeliveryMediums"] = undefined /*out*/;
            resourceInputs["forceAliasCreation"] = undefined /*out*/;
            resourceInputs["messageAction"] = undefined /*out*/;
            resourceInputs["userAttributes"] = undefined /*out*/;
            resourceInputs["userPoolId"] = undefined /*out*/;
            resourceInputs["username"] = undefined /*out*/;
            resourceInputs["validationData"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["clientMetadata.*", "desiredDeliveryMediums[*]", "forceAliasCreation", "messageAction", "userAttributes[*]", "userPoolId", "username", "validationData[*]"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(UserPoolUser.__pulumiType, name, resourceInputs, opts);
    }
}
exports.UserPoolUser = UserPoolUser;
/** @internal */
UserPoolUser.__pulumiType = 'aws-native:cognito:UserPoolUser';
