// @pulumi/aws-native/dms/dataProvider.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataProvider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::DMS::DataProvider
 */
class DataProvider extends pulumi.CustomResource {
    /**
     * Get an existing DataProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DataProvider(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataProvider.__pulumiType;
    }
    /**
     * Create a DataProvider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.engine === undefined) && !opts.urn) {
                throw new Error("Missing required property 'engine'");
            }
            resourceInputs["dataProviderIdentifier"] = args ? args.dataProviderIdentifier : undefined;
            resourceInputs["dataProviderName"] = args ? args.dataProviderName : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["engine"] = args ? args.engine : undefined;
            resourceInputs["exactSettings"] = args ? args.exactSettings : undefined;
            resourceInputs["settings"] = args ? args.settings : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["dataProviderArn"] = undefined /*out*/;
            resourceInputs["dataProviderCreationTime"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataProviderArn"] = undefined /*out*/;
            resourceInputs["dataProviderCreationTime"] = undefined /*out*/;
            resourceInputs["dataProviderIdentifier"] = undefined /*out*/;
            resourceInputs["dataProviderName"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["engine"] = undefined /*out*/;
            resourceInputs["exactSettings"] = undefined /*out*/;
            resourceInputs["settings"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DataProvider.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DataProvider = DataProvider;
/** @internal */
DataProvider.__pulumiType = 'aws-native:dms:DataProvider';
