// @pulumi/aws-native/dms/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationConfig = exports.MigrationProject = exports.InstanceProfile = exports.getReplicationConfigOutput = exports.getReplicationConfig = exports.getMigrationProjectOutput = exports.getMigrationProject = exports.getInstanceProfileOutput = exports.getInstanceProfile = exports.getDataProviderOutput = exports.getDataProvider = exports.getDataMigrationOutput = exports.getDataMigration = exports.DataProvider = exports.DataMigration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataMigration = null;
utilities.lazyLoad(exports, ["DataMigration"], () => require("./dataMigration"));
exports.DataProvider = null;
utilities.lazyLoad(exports, ["DataProvider"], () => require("./dataProvider"));
exports.getDataMigration = null;
exports.getDataMigrationOutput = null;
utilities.lazyLoad(exports, ["getDataMigration", "getDataMigrationOutput"], () => require("./getDataMigration"));
exports.getDataProvider = null;
exports.getDataProviderOutput = null;
utilities.lazyLoad(exports, ["getDataProvider", "getDataProviderOutput"], () => require("./getDataProvider"));
exports.getInstanceProfile = null;
exports.getInstanceProfileOutput = null;
utilities.lazyLoad(exports, ["getInstanceProfile", "getInstanceProfileOutput"], () => require("./getInstanceProfile"));
exports.getMigrationProject = null;
exports.getMigrationProjectOutput = null;
utilities.lazyLoad(exports, ["getMigrationProject", "getMigrationProjectOutput"], () => require("./getMigrationProject"));
exports.getReplicationConfig = null;
exports.getReplicationConfigOutput = null;
utilities.lazyLoad(exports, ["getReplicationConfig", "getReplicationConfigOutput"], () => require("./getReplicationConfig"));
exports.InstanceProfile = null;
utilities.lazyLoad(exports, ["InstanceProfile"], () => require("./instanceProfile"));
exports.MigrationProject = null;
utilities.lazyLoad(exports, ["MigrationProject"], () => require("./migrationProject"));
exports.ReplicationConfig = null;
utilities.lazyLoad(exports, ["ReplicationConfig"], () => require("./replicationConfig"));
// Export enums:
__exportStar(require("../types/enums/dms"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:dms:DataMigration":
                return new exports.DataMigration(name, undefined, { urn });
            case "aws-native:dms:DataProvider":
                return new exports.DataProvider(name, undefined, { urn });
            case "aws-native:dms:InstanceProfile":
                return new exports.InstanceProfile(name, undefined, { urn });
            case "aws-native:dms:MigrationProject":
                return new exports.MigrationProject(name, undefined, { urn });
            case "aws-native:dms:ReplicationConfig":
                return new exports.ReplicationConfig(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "dms", _module);
