// @pulumi/aws-native/dms/replicationConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A replication configuration that you later provide to configure and start a AWS DMS Serverless replication
 */
class ReplicationConfig extends pulumi.CustomResource {
    /**
     * Get an existing ReplicationConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReplicationConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationConfig.__pulumiType;
    }
    /**
     * Create a ReplicationConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.computeConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'computeConfig'");
            }
            if ((!args || args.replicationConfigIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'replicationConfigIdentifier'");
            }
            if ((!args || args.replicationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'replicationType'");
            }
            if ((!args || args.sourceEndpointArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceEndpointArn'");
            }
            if ((!args || args.tableMappings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tableMappings'");
            }
            if ((!args || args.targetEndpointArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetEndpointArn'");
            }
            resourceInputs["computeConfig"] = args ? args.computeConfig : undefined;
            resourceInputs["replicationConfigIdentifier"] = args ? args.replicationConfigIdentifier : undefined;
            resourceInputs["replicationSettings"] = args ? args.replicationSettings : undefined;
            resourceInputs["replicationType"] = args ? args.replicationType : undefined;
            resourceInputs["resourceIdentifier"] = args ? args.resourceIdentifier : undefined;
            resourceInputs["sourceEndpointArn"] = args ? args.sourceEndpointArn : undefined;
            resourceInputs["supplementalSettings"] = args ? args.supplementalSettings : undefined;
            resourceInputs["tableMappings"] = args ? args.tableMappings : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetEndpointArn"] = args ? args.targetEndpointArn : undefined;
            resourceInputs["replicationConfigArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["computeConfig"] = undefined /*out*/;
            resourceInputs["replicationConfigArn"] = undefined /*out*/;
            resourceInputs["replicationConfigIdentifier"] = undefined /*out*/;
            resourceInputs["replicationSettings"] = undefined /*out*/;
            resourceInputs["replicationType"] = undefined /*out*/;
            resourceInputs["resourceIdentifier"] = undefined /*out*/;
            resourceInputs["sourceEndpointArn"] = undefined /*out*/;
            resourceInputs["supplementalSettings"] = undefined /*out*/;
            resourceInputs["tableMappings"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetEndpointArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["resourceIdentifier"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(ReplicationConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ReplicationConfig = ReplicationConfig;
/** @internal */
ReplicationConfig.__pulumiType = 'aws-native:dms:ReplicationConfig';
