// @pulumi/aws-native/types/enums/pipes/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PipeTimeFieldType = exports.PipeTargetInvocationType = exports.PipeState = exports.PipeSelfManagedKafkaStartPosition = exports.PipeS3OutputFormat = exports.PipeRequestedPipeState = exports.PipePropagateTags = exports.PipePlacementStrategyType = exports.PipePlacementConstraintType = exports.PipeOnPartialBatchItemFailureStreams = exports.PipeMskStartPosition = exports.PipeMeasureValueType = exports.PipeLogLevel = exports.PipeLaunchType = exports.PipeKinesisStreamStartPosition = exports.PipeIncludeExecutionDataOption = exports.PipeEpochTimeUnit = exports.PipeEcsResourceRequirementType = exports.PipeEcsEnvironmentFileType = exports.PipeDynamoDbStreamStartPosition = exports.PipeDimensionValueType = exports.PipeBatchResourceRequirementType = exports.PipeBatchJobDependencyType = exports.PipeAssignPublicIp = void 0;
exports.PipeAssignPublicIp = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.PipeBatchJobDependencyType = {
    NToN: "N_TO_N",
    Sequential: "SEQUENTIAL",
};
exports.PipeBatchResourceRequirementType = {
    Gpu: "GPU",
    Memory: "MEMORY",
    Vcpu: "VCPU",
};
exports.PipeDimensionValueType = {
    Varchar: "VARCHAR",
};
exports.PipeDynamoDbStreamStartPosition = {
    TrimHorizon: "TRIM_HORIZON",
    Latest: "LATEST",
};
exports.PipeEcsEnvironmentFileType = {
    S3: "s3",
};
exports.PipeEcsResourceRequirementType = {
    Gpu: "GPU",
    InferenceAccelerator: "InferenceAccelerator",
};
exports.PipeEpochTimeUnit = {
    Milliseconds: "MILLISECONDS",
    Seconds: "SECONDS",
    Microseconds: "MICROSECONDS",
    Nanoseconds: "NANOSECONDS",
};
exports.PipeIncludeExecutionDataOption = {
    All: "ALL",
};
exports.PipeKinesisStreamStartPosition = {
    TrimHorizon: "TRIM_HORIZON",
    Latest: "LATEST",
    AtTimestamp: "AT_TIMESTAMP",
};
exports.PipeLaunchType = {
    Ec2: "EC2",
    Fargate: "FARGATE",
    External: "EXTERNAL",
};
exports.PipeLogLevel = {
    Off: "OFF",
    Error: "ERROR",
    Info: "INFO",
    Trace: "TRACE",
};
exports.PipeMeasureValueType = {
    Double: "DOUBLE",
    Bigint: "BIGINT",
    Varchar: "VARCHAR",
    Boolean: "BOOLEAN",
    Timestamp: "TIMESTAMP",
};
exports.PipeMskStartPosition = {
    TrimHorizon: "TRIM_HORIZON",
    Latest: "LATEST",
};
exports.PipeOnPartialBatchItemFailureStreams = {
    AutomaticBisect: "AUTOMATIC_BISECT",
};
exports.PipePlacementConstraintType = {
    DistinctInstance: "distinctInstance",
    MemberOf: "memberOf",
};
exports.PipePlacementStrategyType = {
    Random: "random",
    Spread: "spread",
    Binpack: "binpack",
};
exports.PipePropagateTags = {
    TaskDefinition: "TASK_DEFINITION",
};
exports.PipeRequestedPipeState = {
    Running: "RUNNING",
    Stopped: "STOPPED",
};
exports.PipeS3OutputFormat = {
    Json: "json",
    Plain: "plain",
    W3c: "w3c",
};
exports.PipeSelfManagedKafkaStartPosition = {
    TrimHorizon: "TRIM_HORIZON",
    Latest: "LATEST",
};
exports.PipeState = {
    Running: "RUNNING",
    Stopped: "STOPPED",
    Creating: "CREATING",
    Updating: "UPDATING",
    Deleting: "DELETING",
    Starting: "STARTING",
    Stopping: "STOPPING",
    CreateFailed: "CREATE_FAILED",
    UpdateFailed: "UPDATE_FAILED",
    StartFailed: "START_FAILED",
    StopFailed: "STOP_FAILED",
    DeleteFailed: "DELETE_FAILED",
    CreateRollbackFailed: "CREATE_ROLLBACK_FAILED",
    DeleteRollbackFailed: "DELETE_ROLLBACK_FAILED",
    UpdateRollbackFailed: "UPDATE_ROLLBACK_FAILED",
};
exports.PipeTargetInvocationType = {
    RequestResponse: "REQUEST_RESPONSE",
    FireAndForget: "FIRE_AND_FORGET",
};
exports.PipeTimeFieldType = {
    Epoch: "EPOCH",
    TimestampFormat: "TIMESTAMP_FORMAT",
};
