// @pulumi/aws-native/securityhub/hub.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Hub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::SecurityHub::Hub resource represents the implementation of the AWS Security Hub service in your account. One hub resource is created for each Region in which you enable Security Hub.
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const exampleHubWithTags = new aws_native.securityhub.Hub("exampleHubWithTags", {
 *     tags: {
 *         key1: "value1",
 *         key2: "value2",
 *     },
 *     enableDefaultStandards: true,
 *     controlFindingGenerator: "SECURITY_CONTROL",
 * });
 * export const hubArn = exampleHubWithTags.id;
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const exampleHubWithTags = new aws_native.securityhub.Hub("exampleHubWithTags", {
 *     tags: {
 *         key1: "value1",
 *         key2: "value2",
 *     },
 *     enableDefaultStandards: true,
 *     controlFindingGenerator: "SECURITY_CONTROL",
 * });
 * export const hubArn = exampleHubWithTags.id;
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const exampleHubWithTags = new aws_native.securityhub.Hub("exampleHubWithTags", {
 *     tags: {
 *         key1: "value1",
 *         key2: "value2",
 *     },
 *     enableDefaultStandards: false,
 *     controlFindingGenerator: "STANDARD_CONTROL",
 * });
 * export const hubArn = exampleHubWithTags.id;
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const exampleHubWithTags = new aws_native.securityhub.Hub("exampleHubWithTags", {
 *     tags: {
 *         key1: "value1",
 *         key2: "value2",
 *     },
 *     enableDefaultStandards: false,
 *     controlFindingGenerator: "STANDARD_CONTROL",
 * });
 * export const hubArn = exampleHubWithTags.id;
 *
 * ```
 */
class Hub extends pulumi.CustomResource {
    /**
     * Get an existing Hub resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Hub(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Hub.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Hub.__pulumiType;
    }
    /**
     * Create a Hub resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["autoEnableControls"] = args ? args.autoEnableControls : undefined;
            resourceInputs["controlFindingGenerator"] = args ? args.controlFindingGenerator : undefined;
            resourceInputs["enableDefaultStandards"] = args ? args.enableDefaultStandards : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["subscribedAt"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["autoEnableControls"] = undefined /*out*/;
            resourceInputs["controlFindingGenerator"] = undefined /*out*/;
            resourceInputs["enableDefaultStandards"] = undefined /*out*/;
            resourceInputs["subscribedAt"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Hub.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Hub = Hub;
/** @internal */
Hub.__pulumiType = 'aws-native:securityhub:Hub';
