// @pulumi/aws-native/securityhub/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Standard = exports.SecurityControl = exports.ProductSubscription = exports.PolicyAssociation = exports.OrganizationConfiguration = exports.Insight = exports.Hub = exports.getStandardOutput = exports.getStandard = exports.getSecurityControlOutput = exports.getSecurityControl = exports.getProductSubscriptionOutput = exports.getProductSubscription = exports.getPolicyAssociationOutput = exports.getPolicyAssociation = exports.getOrganizationConfigurationOutput = exports.getOrganizationConfiguration = exports.getInsightOutput = exports.getInsight = exports.getHubOutput = exports.getHub = exports.getFindingAggregatorOutput = exports.getFindingAggregator = exports.getDelegatedAdminOutput = exports.getDelegatedAdmin = exports.getConfigurationPolicyOutput = exports.getConfigurationPolicy = exports.getAutomationRuleOutput = exports.getAutomationRule = exports.FindingAggregator = exports.DelegatedAdmin = exports.ConfigurationPolicy = exports.AutomationRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AutomationRule = null;
utilities.lazyLoad(exports, ["AutomationRule"], () => require("./automationRule"));
exports.ConfigurationPolicy = null;
utilities.lazyLoad(exports, ["ConfigurationPolicy"], () => require("./configurationPolicy"));
exports.DelegatedAdmin = null;
utilities.lazyLoad(exports, ["DelegatedAdmin"], () => require("./delegatedAdmin"));
exports.FindingAggregator = null;
utilities.lazyLoad(exports, ["FindingAggregator"], () => require("./findingAggregator"));
exports.getAutomationRule = null;
exports.getAutomationRuleOutput = null;
utilities.lazyLoad(exports, ["getAutomationRule", "getAutomationRuleOutput"], () => require("./getAutomationRule"));
exports.getConfigurationPolicy = null;
exports.getConfigurationPolicyOutput = null;
utilities.lazyLoad(exports, ["getConfigurationPolicy", "getConfigurationPolicyOutput"], () => require("./getConfigurationPolicy"));
exports.getDelegatedAdmin = null;
exports.getDelegatedAdminOutput = null;
utilities.lazyLoad(exports, ["getDelegatedAdmin", "getDelegatedAdminOutput"], () => require("./getDelegatedAdmin"));
exports.getFindingAggregator = null;
exports.getFindingAggregatorOutput = null;
utilities.lazyLoad(exports, ["getFindingAggregator", "getFindingAggregatorOutput"], () => require("./getFindingAggregator"));
exports.getHub = null;
exports.getHubOutput = null;
utilities.lazyLoad(exports, ["getHub", "getHubOutput"], () => require("./getHub"));
exports.getInsight = null;
exports.getInsightOutput = null;
utilities.lazyLoad(exports, ["getInsight", "getInsightOutput"], () => require("./getInsight"));
exports.getOrganizationConfiguration = null;
exports.getOrganizationConfigurationOutput = null;
utilities.lazyLoad(exports, ["getOrganizationConfiguration", "getOrganizationConfigurationOutput"], () => require("./getOrganizationConfiguration"));
exports.getPolicyAssociation = null;
exports.getPolicyAssociationOutput = null;
utilities.lazyLoad(exports, ["getPolicyAssociation", "getPolicyAssociationOutput"], () => require("./getPolicyAssociation"));
exports.getProductSubscription = null;
exports.getProductSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getProductSubscription", "getProductSubscriptionOutput"], () => require("./getProductSubscription"));
exports.getSecurityControl = null;
exports.getSecurityControlOutput = null;
utilities.lazyLoad(exports, ["getSecurityControl", "getSecurityControlOutput"], () => require("./getSecurityControl"));
exports.getStandard = null;
exports.getStandardOutput = null;
utilities.lazyLoad(exports, ["getStandard", "getStandardOutput"], () => require("./getStandard"));
exports.Hub = null;
utilities.lazyLoad(exports, ["Hub"], () => require("./hub"));
exports.Insight = null;
utilities.lazyLoad(exports, ["Insight"], () => require("./insight"));
exports.OrganizationConfiguration = null;
utilities.lazyLoad(exports, ["OrganizationConfiguration"], () => require("./organizationConfiguration"));
exports.PolicyAssociation = null;
utilities.lazyLoad(exports, ["PolicyAssociation"], () => require("./policyAssociation"));
exports.ProductSubscription = null;
utilities.lazyLoad(exports, ["ProductSubscription"], () => require("./productSubscription"));
exports.SecurityControl = null;
utilities.lazyLoad(exports, ["SecurityControl"], () => require("./securityControl"));
exports.Standard = null;
utilities.lazyLoad(exports, ["Standard"], () => require("./standard"));
// Export enums:
__exportStar(require("../types/enums/securityhub"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:securityhub:AutomationRule":
                return new exports.AutomationRule(name, undefined, { urn });
            case "aws-native:securityhub:ConfigurationPolicy":
                return new exports.ConfigurationPolicy(name, undefined, { urn });
            case "aws-native:securityhub:DelegatedAdmin":
                return new exports.DelegatedAdmin(name, undefined, { urn });
            case "aws-native:securityhub:FindingAggregator":
                return new exports.FindingAggregator(name, undefined, { urn });
            case "aws-native:securityhub:Hub":
                return new exports.Hub(name, undefined, { urn });
            case "aws-native:securityhub:Insight":
                return new exports.Insight(name, undefined, { urn });
            case "aws-native:securityhub:OrganizationConfiguration":
                return new exports.OrganizationConfiguration(name, undefined, { urn });
            case "aws-native:securityhub:PolicyAssociation":
                return new exports.PolicyAssociation(name, undefined, { urn });
            case "aws-native:securityhub:ProductSubscription":
                return new exports.ProductSubscription(name, undefined, { urn });
            case "aws-native:securityhub:SecurityControl":
                return new exports.SecurityControl(name, undefined, { urn });
            case "aws-native:securityhub:Standard":
                return new exports.Standard(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "securityhub", _module);
