// @pulumi/aws-native/neptunegraph/privateGraphEndpoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateGraphEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::NeptuneGraph::PrivateGraphEndpoint resource creates an Amazon NeptuneGraph PrivateGraphEndpoint.
 */
class PrivateGraphEndpoint extends pulumi.CustomResource {
    /**
     * Get an existing PrivateGraphEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateGraphEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateGraphEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateGraphEndpoint.__pulumiType;
    }
    /**
     * Create a PrivateGraphEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.graphIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'graphIdentifier'");
            }
            if ((!args || args.vpcId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcId'");
            }
            resourceInputs["graphIdentifier"] = args ? args.graphIdentifier : undefined;
            resourceInputs["securityGroupIds"] = args ? args.securityGroupIds : undefined;
            resourceInputs["subnetIds"] = args ? args.subnetIds : undefined;
            resourceInputs["vpcId"] = args ? args.vpcId : undefined;
            resourceInputs["privateGraphEndpointIdentifier"] = undefined /*out*/;
            resourceInputs["vpcEndpointId"] = undefined /*out*/;
        }
        else {
            resourceInputs["graphIdentifier"] = undefined /*out*/;
            resourceInputs["privateGraphEndpointIdentifier"] = undefined /*out*/;
            resourceInputs["securityGroupIds"] = undefined /*out*/;
            resourceInputs["subnetIds"] = undefined /*out*/;
            resourceInputs["vpcEndpointId"] = undefined /*out*/;
            resourceInputs["vpcId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["graphIdentifier", "securityGroupIds[*]", "subnetIds[*]", "vpcId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(PrivateGraphEndpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateGraphEndpoint = PrivateGraphEndpoint;
/** @internal */
PrivateGraphEndpoint.__pulumiType = 'aws-native:neptunegraph:PrivateGraphEndpoint';
