// @pulumi/aws-native/eventschemas/discoverer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Discoverer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EventSchemas::Discoverer
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const myDiscoverer = new aws_native.eventschemas.Discoverer("myDiscoverer", {
 *     sourceArn: "arn:aws:events:us-west-2:012345678910:event-bus/default",
 *     description: "discover all custom schemas",
 * });
 *
 * ```
 */
class Discoverer extends pulumi.CustomResource {
    /**
     * Get an existing Discoverer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Discoverer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Discoverer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Discoverer.__pulumiType;
    }
    /**
     * Create a Discoverer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.sourceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceArn'");
            }
            resourceInputs["crossAccount"] = args ? args.crossAccount : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["sourceArn"] = args ? args.sourceArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["discovererArn"] = undefined /*out*/;
            resourceInputs["discovererId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
        }
        else {
            resourceInputs["crossAccount"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["discovererArn"] = undefined /*out*/;
            resourceInputs["discovererId"] = undefined /*out*/;
            resourceInputs["sourceArn"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["sourceArn"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(Discoverer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Discoverer = Discoverer;
/** @internal */
Discoverer.__pulumiType = 'aws-native:eventschemas:Discoverer';
