// @pulumi/aws-native/wafv2/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAclAssociation = exports.WebAcl = exports.RuleGroup = exports.RegexPatternSet = exports.LoggingConfiguration = exports.IpSet = exports.getWebAclOutput = exports.getWebAcl = exports.getRuleGroupOutput = exports.getRuleGroup = exports.getRegexPatternSetOutput = exports.getRegexPatternSet = exports.getLoggingConfigurationOutput = exports.getLoggingConfiguration = exports.getIpSetOutput = exports.getIpSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getIpSet = null;
exports.getIpSetOutput = null;
utilities.lazyLoad(exports, ["getIpSet", "getIpSetOutput"], () => require("./getIpSet"));
exports.getLoggingConfiguration = null;
exports.getLoggingConfigurationOutput = null;
utilities.lazyLoad(exports, ["getLoggingConfiguration", "getLoggingConfigurationOutput"], () => require("./getLoggingConfiguration"));
exports.getRegexPatternSet = null;
exports.getRegexPatternSetOutput = null;
utilities.lazyLoad(exports, ["getRegexPatternSet", "getRegexPatternSetOutput"], () => require("./getRegexPatternSet"));
exports.getRuleGroup = null;
exports.getRuleGroupOutput = null;
utilities.lazyLoad(exports, ["getRuleGroup", "getRuleGroupOutput"], () => require("./getRuleGroup"));
exports.getWebAcl = null;
exports.getWebAclOutput = null;
utilities.lazyLoad(exports, ["getWebAcl", "getWebAclOutput"], () => require("./getWebAcl"));
exports.IpSet = null;
utilities.lazyLoad(exports, ["IpSet"], () => require("./ipSet"));
exports.LoggingConfiguration = null;
utilities.lazyLoad(exports, ["LoggingConfiguration"], () => require("./loggingConfiguration"));
exports.RegexPatternSet = null;
utilities.lazyLoad(exports, ["RegexPatternSet"], () => require("./regexPatternSet"));
exports.RuleGroup = null;
utilities.lazyLoad(exports, ["RuleGroup"], () => require("./ruleGroup"));
exports.WebAcl = null;
utilities.lazyLoad(exports, ["WebAcl"], () => require("./webAcl"));
exports.WebAclAssociation = null;
utilities.lazyLoad(exports, ["WebAclAssociation"], () => require("./webAclAssociation"));
// Export enums:
__exportStar(require("../types/enums/wafv2"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:wafv2:IpSet":
                return new exports.IpSet(name, undefined, { urn });
            case "aws-native:wafv2:LoggingConfiguration":
                return new exports.LoggingConfiguration(name, undefined, { urn });
            case "aws-native:wafv2:RegexPatternSet":
                return new exports.RegexPatternSet(name, undefined, { urn });
            case "aws-native:wafv2:RuleGroup":
                return new exports.RuleGroup(name, undefined, { urn });
            case "aws-native:wafv2:WebAcl":
                return new exports.WebAcl(name, undefined, { urn });
            case "aws-native:wafv2:WebAclAssociation":
                return new exports.WebAclAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "wafv2", _module);
