// @pulumi/aws-native/types/enums/cleanrooms/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivacyBudgetTemplatePrivacyBudgetType = exports.PrivacyBudgetTemplateAutoRefresh = exports.MembershipResultFormat = exports.MembershipQueryLogStatus = exports.IdNamespaceAssociationInputReferencePropertiesIdNamespaceType = exports.IdMappingTableInputSourceType = exports.ConfiguredTableScalarFunctions = exports.ConfiguredTableJoinRequiredOption = exports.ConfiguredTableJoinOperator = exports.ConfiguredTableAssociationAnalysisRuleType = exports.ConfiguredTableAnalysisRuleType = exports.ConfiguredTableAnalysisMethod = exports.ConfiguredTableAggregationType = exports.ConfiguredTableAggregateFunctionName = exports.ConfiguredTableAdditionalAnalyses = exports.CollaborationQueryLogStatus = exports.CollaborationMemberAbility = exports.CollaborationAnalyticsEngine = exports.AnalysisTemplateFormat = exports.AnalysisTemplateAnalysisParameterType = void 0;
exports.AnalysisTemplateAnalysisParameterType = {
    Smallint: "SMALLINT",
    Integer: "INTEGER",
    Bigint: "BIGINT",
    Decimal: "DECIMAL",
    Real: "REAL",
    DoublePrecision: "DOUBLE_PRECISION",
    Boolean: "BOOLEAN",
    Char: "CHAR",
    Varchar: "VARCHAR",
    Date: "DATE",
    Timestamp: "TIMESTAMP",
    Timestamptz: "TIMESTAMPTZ",
    Time: "TIME",
    Timetz: "TIMETZ",
    Varbyte: "VARBYTE",
    Binary: "BINARY",
    Byte: "BYTE",
    Character: "CHARACTER",
    Double: "DOUBLE",
    Float: "FLOAT",
    Int: "INT",
    Long: "LONG",
    Numeric: "NUMERIC",
    Short: "SHORT",
    String: "STRING",
    TimestampLtz: "TIMESTAMP_LTZ",
    TimestampNtz: "TIMESTAMP_NTZ",
    Tinyint: "TINYINT",
};
exports.AnalysisTemplateFormat = {
    Sql: "SQL",
};
exports.CollaborationAnalyticsEngine = {
    CleanRoomsSql: "CLEAN_ROOMS_SQL",
    Spark: "SPARK",
};
exports.CollaborationMemberAbility = {
    CanQuery: "CAN_QUERY",
    CanReceiveResults: "CAN_RECEIVE_RESULTS",
};
exports.CollaborationQueryLogStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.ConfiguredTableAdditionalAnalyses = {
    Allowed: "ALLOWED",
    Required: "REQUIRED",
    NotAllowed: "NOT_ALLOWED",
};
exports.ConfiguredTableAggregateFunctionName = {
    Sum: "SUM",
    SumDistinct: "SUM_DISTINCT",
    Count: "COUNT",
    CountDistinct: "COUNT_DISTINCT",
    Avg: "AVG",
};
exports.ConfiguredTableAggregationType = {
    CountDistinct: "COUNT_DISTINCT",
};
exports.ConfiguredTableAnalysisMethod = {
    DirectQuery: "DIRECT_QUERY",
};
exports.ConfiguredTableAnalysisRuleType = {
    Aggregation: "AGGREGATION",
    List: "LIST",
    Custom: "CUSTOM",
};
exports.ConfiguredTableAssociationAnalysisRuleType = {
    Aggregation: "AGGREGATION",
    List: "LIST",
    Custom: "CUSTOM",
};
exports.ConfiguredTableJoinOperator = {
    Or: "OR",
    And: "AND",
};
exports.ConfiguredTableJoinRequiredOption = {
    QueryRunner: "QUERY_RUNNER",
};
exports.ConfiguredTableScalarFunctions = {
    Trunc: "TRUNC",
    Abs: "ABS",
    Ceiling: "CEILING",
    Floor: "FLOOR",
    Ln: "LN",
    Log: "LOG",
    Round: "ROUND",
    Sqrt: "SQRT",
    Cast: "CAST",
    Lower: "LOWER",
    Rtrim: "RTRIM",
    Upper: "UPPER",
    Coalesce: "COALESCE",
    Convert: "CONVERT",
    CurrentDate: "CURRENT_DATE",
    Dateadd: "DATEADD",
    Extract: "EXTRACT",
    Getdate: "GETDATE",
    Substring: "SUBSTRING",
    ToChar: "TO_CHAR",
    ToDate: "TO_DATE",
    ToNumber: "TO_NUMBER",
    ToTimestamp: "TO_TIMESTAMP",
    Trim: "TRIM",
};
exports.IdMappingTableInputSourceType = {
    Source: "SOURCE",
    Target: "TARGET",
};
exports.IdNamespaceAssociationInputReferencePropertiesIdNamespaceType = {
    Source: "SOURCE",
    Target: "TARGET",
};
exports.MembershipQueryLogStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.MembershipResultFormat = {
    Csv: "CSV",
    Parquet: "PARQUET",
};
exports.PrivacyBudgetTemplateAutoRefresh = {
    CalendarMonth: "CALENDAR_MONTH",
    None: "NONE",
};
exports.PrivacyBudgetTemplatePrivacyBudgetType = {
    DifferentialPrivacy: "DIFFERENTIAL_PRIVACY",
};
