// @pulumi/aws-native/workspacesweb/userSettings.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSettings = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::WorkSpacesWeb::UserSettings Resource Type
 */
class UserSettings extends pulumi.CustomResource {
    /**
     * Get an existing UserSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new UserSettings(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of UserSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserSettings.__pulumiType;
    }
    /**
     * Create a UserSettings resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.copyAllowed === undefined) && !opts.urn) {
                throw new Error("Missing required property 'copyAllowed'");
            }
            if ((!args || args.downloadAllowed === undefined) && !opts.urn) {
                throw new Error("Missing required property 'downloadAllowed'");
            }
            if ((!args || args.pasteAllowed === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pasteAllowed'");
            }
            if ((!args || args.printAllowed === undefined) && !opts.urn) {
                throw new Error("Missing required property 'printAllowed'");
            }
            if ((!args || args.uploadAllowed === undefined) && !opts.urn) {
                throw new Error("Missing required property 'uploadAllowed'");
            }
            resourceInputs["additionalEncryptionContext"] = args ? args.additionalEncryptionContext : undefined;
            resourceInputs["cookieSynchronizationConfiguration"] = args ? args.cookieSynchronizationConfiguration : undefined;
            resourceInputs["copyAllowed"] = args ? args.copyAllowed : undefined;
            resourceInputs["customerManagedKey"] = args ? args.customerManagedKey : undefined;
            resourceInputs["deepLinkAllowed"] = args ? args.deepLinkAllowed : undefined;
            resourceInputs["disconnectTimeoutInMinutes"] = args ? args.disconnectTimeoutInMinutes : undefined;
            resourceInputs["downloadAllowed"] = args ? args.downloadAllowed : undefined;
            resourceInputs["idleDisconnectTimeoutInMinutes"] = args ? args.idleDisconnectTimeoutInMinutes : undefined;
            resourceInputs["pasteAllowed"] = args ? args.pasteAllowed : undefined;
            resourceInputs["printAllowed"] = args ? args.printAllowed : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["uploadAllowed"] = args ? args.uploadAllowed : undefined;
            resourceInputs["associatedPortalArns"] = undefined /*out*/;
            resourceInputs["userSettingsArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["additionalEncryptionContext"] = undefined /*out*/;
            resourceInputs["associatedPortalArns"] = undefined /*out*/;
            resourceInputs["cookieSynchronizationConfiguration"] = undefined /*out*/;
            resourceInputs["copyAllowed"] = undefined /*out*/;
            resourceInputs["customerManagedKey"] = undefined /*out*/;
            resourceInputs["deepLinkAllowed"] = undefined /*out*/;
            resourceInputs["disconnectTimeoutInMinutes"] = undefined /*out*/;
            resourceInputs["downloadAllowed"] = undefined /*out*/;
            resourceInputs["idleDisconnectTimeoutInMinutes"] = undefined /*out*/;
            resourceInputs["pasteAllowed"] = undefined /*out*/;
            resourceInputs["printAllowed"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["uploadAllowed"] = undefined /*out*/;
            resourceInputs["userSettingsArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(UserSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.UserSettings = UserSettings;
/** @internal */
UserSettings.__pulumiType = 'aws-native:workspacesweb:UserSettings';
