// @pulumi/aws-native/types/enums/pcs/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueStatus = exports.ComputeNodeGroupStatus = exports.ComputeNodeGroupSpotOptionsPropertiesAllocationStrategy = exports.ComputeNodeGroupPurchaseOption = exports.ClusterStatus = exports.ClusterSize = exports.ClusterSchedulerPropertiesType = exports.ClusterEndpointType = void 0;
exports.ClusterEndpointType = {
    Slurmctld: "SLURMCTLD",
    Slurmdbd: "SLURMDBD",
};
exports.ClusterSchedulerPropertiesType = {
    Slurm: "SLURM",
};
exports.ClusterSize = {
    Small: "SMALL",
    Medium: "MEDIUM",
    Large: "LARGE",
};
exports.ClusterStatus = {
    Creating: "CREATING",
    Active: "ACTIVE",
    Updating: "UPDATING",
    Deleting: "DELETING",
    CreateFailed: "CREATE_FAILED",
    DeleteFailed: "DELETE_FAILED",
    UpdateFailed: "UPDATE_FAILED",
};
exports.ComputeNodeGroupPurchaseOption = {
    Ondemand: "ONDEMAND",
    Spot: "SPOT",
};
exports.ComputeNodeGroupSpotOptionsPropertiesAllocationStrategy = {
    LowestPrice: "lowest-price",
    CapacityOptimized: "capacity-optimized",
    PriceCapacityOptimized: "price-capacity-optimized",
};
exports.ComputeNodeGroupStatus = {
    Creating: "CREATING",
    Active: "ACTIVE",
    Updating: "UPDATING",
    Deleting: "DELETING",
    CreateFailed: "CREATE_FAILED",
    DeleteFailed: "DELETE_FAILED",
    UpdateFailed: "UPDATE_FAILED",
};
exports.QueueStatus = {
    Creating: "CREATING",
    Active: "ACTIVE",
    Updating: "UPDATING",
    Deleting: "DELETING",
    CreateFailed: "CREATE_FAILED",
    DeleteFailed: "DELETE_FAILED",
    UpdateFailed: "UPDATE_FAILED",
};
