// @pulumi/aws-native/types/enums/sagemaker/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelCardObjectiveFunctionFunctionPropertiesFunction = exports.ModelCardModelPackageDetailsModelPackageStatus = exports.ModelCardModelPackageDetailsModelApprovalStatus = exports.ModelCardMatrixMetricType = exports.ModelCardLinearGraphMetricType = exports.ModelCardBarChartMetricType = exports.ModelBiasJobDefinitionS3OutputS3UploadMode = exports.ModelBiasJobDefinitionEndpointInputS3InputMode = exports.ModelBiasJobDefinitionEndpointInputS3DataDistributionType = exports.ModelBiasJobDefinitionBatchTransformInputS3InputMode = exports.ModelBiasJobDefinitionBatchTransformInputS3DataDistributionType = exports.MlflowTrackingServerTrackingServerSize = exports.InferenceExperimentType = exports.InferenceExperimentStatus = exports.InferenceExperimentModelInfrastructureConfigInfrastructureType = exports.InferenceExperimentEndpointMetadataEndpointStatus = exports.InferenceExperimentDesiredState = exports.InferenceComponentStatus = exports.ImageVersionVendorGuidance = exports.ImageVersionProcessor = exports.ImageVersionJobType = exports.FeatureGroupUnit = exports.FeatureGroupThroughputMode = exports.FeatureGroupTableFormat = exports.FeatureGroupStorageType = exports.FeatureGroupFeatureDefinitionFeatureType = exports.DomainUserSettingsStudioWebPortal = exports.DomainTagPropagation = exports.DomainSharingSettingsNotebookOutputOption = exports.DomainSettingsExecutionRoleIdentityConfig = exports.DomainResourceSpecInstanceType = exports.DomainRStudioServerProAppSettingsUserGroup = exports.DomainRStudioServerProAppSettingsAccessStatus = exports.DomainMlTools = exports.DomainLifecycleManagement = exports.DomainDockerSettingsEnableDockerAccess = exports.DomainAuthMode = exports.DomainAppType = exports.DomainAppSecurityGroupManagement = exports.DomainAppNetworkAccessType = exports.DataQualityJobDefinitionS3OutputS3UploadMode = exports.DataQualityJobDefinitionEndpointInputS3InputMode = exports.DataQualityJobDefinitionEndpointInputS3DataDistributionType = exports.DataQualityJobDefinitionBatchTransformInputS3InputMode = exports.DataQualityJobDefinitionBatchTransformInputS3DataDistributionType = exports.ClusterStatus = exports.ClusterNodeRecovery = exports.ClusterDeepHealthCheckType = exports.AppType = exports.AppResourceSpecInstanceType = void 0;
exports.UserProfileSharingSettingsNotebookOutputOption = exports.UserProfileResourceSpecInstanceType = exports.UserProfileRStudioServerProAppSettingsUserGroup = exports.UserProfileRStudioServerProAppSettingsAccessStatus = exports.UserProfileMlTools = exports.UserProfileLifecycleManagement = exports.UserProfileAppType = exports.StudioLifecycleConfigAppType = exports.SpaceSharingSettingsSharingType = exports.SpaceResourceSpecInstanceType = exports.SpaceAppType = exports.ProjectStatus = exports.PartnerAppType = exports.PartnerAppAuthType = exports.MonitoringScheduleStatus = exports.MonitoringScheduleS3OutputS3UploadMode = exports.MonitoringScheduleMonitoringType = exports.MonitoringScheduleMonitoringExecutionSummaryMonitoringExecutionStatus = exports.MonitoringScheduleEndpointInputS3InputMode = exports.MonitoringScheduleEndpointInputS3DataDistributionType = exports.MonitoringScheduleBatchTransformInputS3InputMode = exports.MonitoringScheduleBatchTransformInputS3DataDistributionType = exports.ModelQualityJobDefinitionS3OutputS3UploadMode = exports.ModelQualityJobDefinitionProblemType = exports.ModelQualityJobDefinitionEndpointInputS3InputMode = exports.ModelQualityJobDefinitionEndpointInputS3DataDistributionType = exports.ModelQualityJobDefinitionBatchTransformInputS3InputMode = exports.ModelQualityJobDefinitionBatchTransformInputS3DataDistributionType = exports.ModelPackageTransformOutputAssembleWith = exports.ModelPackageTransformJobDefinitionBatchStrategy = exports.ModelPackageTransformInputSplitType = exports.ModelPackageTransformInputCompressionType = exports.ModelPackageStatusItemStatus = exports.ModelPackageStatus = exports.ModelPackageSkipModelValidation = exports.ModelPackageS3ModelDataSourceS3DataType = exports.ModelPackageS3ModelDataSourceCompressionType = exports.ModelPackageS3DataSourceS3DataType = exports.ModelPackageModelCardModelCardStatus = exports.ModelPackageModelApprovalStatus = exports.ModelPackageGroupStatus = exports.ModelExplainabilityJobDefinitionS3OutputS3UploadMode = exports.ModelExplainabilityJobDefinitionEndpointInputS3InputMode = exports.ModelExplainabilityJobDefinitionEndpointInputS3DataDistributionType = exports.ModelExplainabilityJobDefinitionBatchTransformInputS3InputMode = exports.ModelExplainabilityJobDefinitionBatchTransformInputS3DataDistributionType = exports.ModelCardStatus = exports.ModelCardSimpleMetricType = exports.ModelCardRiskRating = exports.ModelCardProcessingStatus = void 0;
exports.UserProfileUserSettingsStudioWebPortal = void 0;
exports.AppResourceSpecInstanceType = {
    System: "system",
    MlT3Micro: "ml.t3.micro",
    MlT3Small: "ml.t3.small",
    MlT3Medium: "ml.t3.medium",
    MlT3Large: "ml.t3.large",
    MlT3Xlarge: "ml.t3.xlarge",
    MlT32xlarge: "ml.t3.2xlarge",
    MlM5Large: "ml.m5.large",
    MlM5Xlarge: "ml.m5.xlarge",
    MlM52xlarge: "ml.m5.2xlarge",
    MlM54xlarge: "ml.m5.4xlarge",
    MlM58xlarge: "ml.m5.8xlarge",
    MlM512xlarge: "ml.m5.12xlarge",
    MlM516xlarge: "ml.m5.16xlarge",
    MlM524xlarge: "ml.m5.24xlarge",
    MlC5Large: "ml.c5.large",
    MlC5Xlarge: "ml.c5.xlarge",
    MlC52xlarge: "ml.c5.2xlarge",
    MlC54xlarge: "ml.c5.4xlarge",
    MlC59xlarge: "ml.c5.9xlarge",
    MlC512xlarge: "ml.c5.12xlarge",
    MlC518xlarge: "ml.c5.18xlarge",
    MlC524xlarge: "ml.c5.24xlarge",
    MlP32xlarge: "ml.p3.2xlarge",
    MlP38xlarge: "ml.p3.8xlarge",
    MlP316xlarge: "ml.p3.16xlarge",
    MlG4dnXlarge: "ml.g4dn.xlarge",
    MlG4dn2xlarge: "ml.g4dn.2xlarge",
    MlG4dn4xlarge: "ml.g4dn.4xlarge",
    MlG4dn8xlarge: "ml.g4dn.8xlarge",
    MlG4dn12xlarge: "ml.g4dn.12xlarge",
    MlG4dn16xlarge: "ml.g4dn.16xlarge",
    MlR5Large: "ml.r5.large",
    MlR5Xlarge: "ml.r5.xlarge",
    MlR52xlarge: "ml.r5.2xlarge",
    MlR54xlarge: "ml.r5.4xlarge",
    MlR58xlarge: "ml.r5.8xlarge",
    MlR512xlarge: "ml.r5.12xlarge",
    MlR516xlarge: "ml.r5.16xlarge",
    MlR524xlarge: "ml.r5.24xlarge",
    MlP3dn24xlarge: "ml.p3dn.24xlarge",
    MlM5dLarge: "ml.m5d.large",
    MlM5dXlarge: "ml.m5d.xlarge",
    MlM5d2xlarge: "ml.m5d.2xlarge",
    MlM5d4xlarge: "ml.m5d.4xlarge",
    MlM5d8xlarge: "ml.m5d.8xlarge",
    MlM5d12xlarge: "ml.m5d.12xlarge",
    MlM5d16xlarge: "ml.m5d.16xlarge",
    MlM5d24xlarge: "ml.m5d.24xlarge",
    MlG5Xlarge: "ml.g5.xlarge",
    MlG52xlarge: "ml.g5.2xlarge",
    MlG54xlarge: "ml.g5.4xlarge",
    MlG58xlarge: "ml.g5.8xlarge",
    MlG512xlarge: "ml.g5.12xlarge",
    MlG516xlarge: "ml.g5.16xlarge",
    MlG524xlarge: "ml.g5.24xlarge",
    MlG548xlarge: "ml.g5.48xlarge",
    MlP4d24xlarge: "ml.p4d.24xlarge",
    MlP4de24xlarge: "ml.p4de.24xlarge",
    MlGeospatialInteractive: "ml.geospatial.interactive",
    MlTrn12xlarge: "ml.trn1.2xlarge",
    MlTrn132xlarge: "ml.trn1.32xlarge",
    MlTrn1n32xlarge: "ml.trn1n.32xlarge",
};
exports.AppType = {
    JupyterServer: "JupyterServer",
    KernelGateway: "KernelGateway",
    RStudioServerPro: "RStudioServerPro",
    RSessionGateway: "RSessionGateway",
    Canvas: "Canvas",
};
exports.ClusterDeepHealthCheckType = {
    InstanceStress: "InstanceStress",
    InstanceConnectivity: "InstanceConnectivity",
};
exports.ClusterNodeRecovery = {
    Automatic: "Automatic",
    None: "None",
};
exports.ClusterStatus = {
    Creating: "Creating",
    Deleting: "Deleting",
    Failed: "Failed",
    InService: "InService",
    RollingBack: "RollingBack",
    SystemUpdating: "SystemUpdating",
    Updating: "Updating",
};
exports.DataQualityJobDefinitionBatchTransformInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.DataQualityJobDefinitionBatchTransformInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.DataQualityJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.DataQualityJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.DataQualityJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.DomainAppNetworkAccessType = {
    PublicInternetOnly: "PublicInternetOnly",
    VpcOnly: "VpcOnly",
};
exports.DomainAppSecurityGroupManagement = {
    Service: "Service",
    Customer: "Customer",
};
exports.DomainAppType = {
    JupyterServer: "JupyterServer",
    TensorBoard: "TensorBoard",
    RStudioServerPro: "RStudioServerPro",
    JupyterLab: "JupyterLab",
    CodeEditor: "CodeEditor",
    DetailedProfiler: "DetailedProfiler",
    Canvas: "Canvas",
};
exports.DomainAuthMode = {
    Sso: "SSO",
    Iam: "IAM",
};
exports.DomainDockerSettingsEnableDockerAccess = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.DomainLifecycleManagement = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.DomainMlTools = {
    DataWrangler: "DataWrangler",
    FeatureStore: "FeatureStore",
    EmrClusters: "EmrClusters",
    AutoMl: "AutoMl",
    Experiments: "Experiments",
    Training: "Training",
    ModelEvaluation: "ModelEvaluation",
    Pipelines: "Pipelines",
    Models: "Models",
    JumpStart: "JumpStart",
    InferenceRecommender: "InferenceRecommender",
    Endpoints: "Endpoints",
    Projects: "Projects",
    InferenceOptimization: "InferenceOptimization",
    HyperPodClusters: "HyperPodClusters",
    Comet: "Comet",
    DeepchecksLlmEvaluation: "DeepchecksLLMEvaluation",
    Fiddler: "Fiddler",
    LakeraGuard: "LakeraGuard",
};
exports.DomainRStudioServerProAppSettingsAccessStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.DomainRStudioServerProAppSettingsUserGroup = {
    RStudioAdmin: "R_STUDIO_ADMIN",
    RStudioUser: "R_STUDIO_USER",
};
exports.DomainResourceSpecInstanceType = {
    System: "system",
    MlT3Micro: "ml.t3.micro",
    MlT3Small: "ml.t3.small",
    MlT3Medium: "ml.t3.medium",
    MlT3Large: "ml.t3.large",
    MlT3Xlarge: "ml.t3.xlarge",
    MlT32xlarge: "ml.t3.2xlarge",
    MlM5Large: "ml.m5.large",
    MlM5Xlarge: "ml.m5.xlarge",
    MlM52xlarge: "ml.m5.2xlarge",
    MlM54xlarge: "ml.m5.4xlarge",
    MlM58xlarge: "ml.m5.8xlarge",
    MlM512xlarge: "ml.m5.12xlarge",
    MlM516xlarge: "ml.m5.16xlarge",
    MlM524xlarge: "ml.m5.24xlarge",
    MlC5Large: "ml.c5.large",
    MlC5Xlarge: "ml.c5.xlarge",
    MlC52xlarge: "ml.c5.2xlarge",
    MlC54xlarge: "ml.c5.4xlarge",
    MlC59xlarge: "ml.c5.9xlarge",
    MlC512xlarge: "ml.c5.12xlarge",
    MlC518xlarge: "ml.c5.18xlarge",
    MlC524xlarge: "ml.c5.24xlarge",
    MlP32xlarge: "ml.p3.2xlarge",
    MlP38xlarge: "ml.p3.8xlarge",
    MlP316xlarge: "ml.p3.16xlarge",
    MlG4dnXlarge: "ml.g4dn.xlarge",
    MlG4dn2xlarge: "ml.g4dn.2xlarge",
    MlG4dn4xlarge: "ml.g4dn.4xlarge",
    MlG4dn8xlarge: "ml.g4dn.8xlarge",
    MlG4dn12xlarge: "ml.g4dn.12xlarge",
    MlG4dn16xlarge: "ml.g4dn.16xlarge",
    MlR5Large: "ml.r5.large",
    MlR5Xlarge: "ml.r5.xlarge",
    MlR52xlarge: "ml.r5.2xlarge",
    MlR54xlarge: "ml.r5.4xlarge",
    MlR58xlarge: "ml.r5.8xlarge",
    MlR512xlarge: "ml.r5.12xlarge",
    MlR516xlarge: "ml.r5.16xlarge",
    MlR524xlarge: "ml.r5.24xlarge",
    MlP3dn24xlarge: "ml.p3dn.24xlarge",
    MlM5dLarge: "ml.m5d.large",
    MlM5dXlarge: "ml.m5d.xlarge",
    MlM5d2xlarge: "ml.m5d.2xlarge",
    MlM5d4xlarge: "ml.m5d.4xlarge",
    MlM5d8xlarge: "ml.m5d.8xlarge",
    MlM5d12xlarge: "ml.m5d.12xlarge",
    MlM5d16xlarge: "ml.m5d.16xlarge",
    MlM5d24xlarge: "ml.m5d.24xlarge",
    MlG5Xlarge: "ml.g5.xlarge",
    MlG52xlarge: "ml.g5.2xlarge",
    MlG54xlarge: "ml.g5.4xlarge",
    MlG58xlarge: "ml.g5.8xlarge",
    MlG512xlarge: "ml.g5.12xlarge",
    MlG516xlarge: "ml.g5.16xlarge",
    MlG524xlarge: "ml.g5.24xlarge",
    MlG548xlarge: "ml.g5.48xlarge",
    MlP4d24xlarge: "ml.p4d.24xlarge",
    MlP4de24xlarge: "ml.p4de.24xlarge",
    MlGeospatialInteractive: "ml.geospatial.interactive",
    MlTrn12xlarge: "ml.trn1.2xlarge",
    MlTrn132xlarge: "ml.trn1.32xlarge",
    MlTrn1n32xlarge: "ml.trn1n.32xlarge",
};
exports.DomainSettingsExecutionRoleIdentityConfig = {
    UserProfileName: "USER_PROFILE_NAME",
    Disabled: "DISABLED",
};
exports.DomainSharingSettingsNotebookOutputOption = {
    Allowed: "Allowed",
    Disabled: "Disabled",
};
exports.DomainTagPropagation = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.DomainUserSettingsStudioWebPortal = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.FeatureGroupFeatureDefinitionFeatureType = {
    Integral: "Integral",
    Fractional: "Fractional",
    String: "String",
};
exports.FeatureGroupStorageType = {
    Standard: "Standard",
    InMemory: "InMemory",
};
exports.FeatureGroupTableFormat = {
    Iceberg: "Iceberg",
    Glue: "Glue",
};
exports.FeatureGroupThroughputMode = {
    OnDemand: "OnDemand",
    Provisioned: "Provisioned",
};
exports.FeatureGroupUnit = {
    Seconds: "Seconds",
    Minutes: "Minutes",
    Hours: "Hours",
    Days: "Days",
    Weeks: "Weeks",
};
exports.ImageVersionJobType = {
    Training: "TRAINING",
    Inference: "INFERENCE",
    NotebookKernel: "NOTEBOOK_KERNEL",
};
exports.ImageVersionProcessor = {
    Cpu: "CPU",
    Gpu: "GPU",
};
exports.ImageVersionVendorGuidance = {
    NotProvided: "NOT_PROVIDED",
    Stable: "STABLE",
    ToBeArchived: "TO_BE_ARCHIVED",
    Archived: "ARCHIVED",
};
exports.InferenceComponentStatus = {
    InService: "InService",
    Creating: "Creating",
    Updating: "Updating",
    Failed: "Failed",
    Deleting: "Deleting",
};
exports.InferenceExperimentDesiredState = {
    Running: "Running",
    Completed: "Completed",
    Cancelled: "Cancelled",
};
exports.InferenceExperimentEndpointMetadataEndpointStatus = {
    Creating: "Creating",
    Updating: "Updating",
    SystemUpdating: "SystemUpdating",
    RollingBack: "RollingBack",
    InService: "InService",
    OutOfService: "OutOfService",
    Deleting: "Deleting",
    Failed: "Failed",
};
exports.InferenceExperimentModelInfrastructureConfigInfrastructureType = {
    RealTimeInference: "RealTimeInference",
};
exports.InferenceExperimentStatus = {
    Creating: "Creating",
    Created: "Created",
    Updating: "Updating",
    Starting: "Starting",
    Stopping: "Stopping",
    Running: "Running",
    Completed: "Completed",
    Cancelled: "Cancelled",
};
exports.InferenceExperimentType = {
    ShadowMode: "ShadowMode",
};
exports.MlflowTrackingServerTrackingServerSize = {
    Small: "Small",
    Medium: "Medium",
    Large: "Large",
};
exports.ModelBiasJobDefinitionBatchTransformInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelBiasJobDefinitionBatchTransformInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelBiasJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelBiasJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelBiasJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.ModelCardBarChartMetricType = {
    BarChart: "bar_chart",
};
exports.ModelCardLinearGraphMetricType = {
    LinearGraph: "linear_graph",
};
exports.ModelCardMatrixMetricType = {
    Matrix: "matrix",
};
exports.ModelCardModelPackageDetailsModelApprovalStatus = {
    Approved: "Approved",
    Rejected: "Rejected",
    PendingManualApproval: "PendingManualApproval",
};
exports.ModelCardModelPackageDetailsModelPackageStatus = {
    Pending: "Pending",
    InProgress: "InProgress",
    Completed: "Completed",
    Failed: "Failed",
    Deleting: "Deleting",
};
exports.ModelCardObjectiveFunctionFunctionPropertiesFunction = {
    Maximize: "Maximize",
    Minimize: "Minimize",
};
exports.ModelCardProcessingStatus = {
    UnsetValue: "UnsetValue",
    DeleteInProgress: "DeleteInProgress",
    DeletePending: "DeletePending",
    ContentDeleted: "ContentDeleted",
    ExportJobsDeleted: "ExportJobsDeleted",
    DeleteCompleted: "DeleteCompleted",
    DeleteFailed: "DeleteFailed",
};
exports.ModelCardRiskRating = {
    High: "High",
    Medium: "Medium",
    Low: "Low",
    Unknown: "Unknown",
};
exports.ModelCardSimpleMetricType = {
    Number: "number",
    String: "string",
    Boolean: "boolean",
};
exports.ModelCardStatus = {
    Draft: "Draft",
    PendingReview: "PendingReview",
    Approved: "Approved",
    Archived: "Archived",
};
exports.ModelExplainabilityJobDefinitionBatchTransformInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelExplainabilityJobDefinitionBatchTransformInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelExplainabilityJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelExplainabilityJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelExplainabilityJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.ModelPackageGroupStatus = {
    Pending: "Pending",
    InProgress: "InProgress",
    Completed: "Completed",
    Failed: "Failed",
    Deleting: "Deleting",
    DeleteFailed: "DeleteFailed",
};
exports.ModelPackageModelApprovalStatus = {
    Approved: "Approved",
    Rejected: "Rejected",
    PendingManualApproval: "PendingManualApproval",
};
exports.ModelPackageModelCardModelCardStatus = {
    Draft: "Draft",
    PendingReview: "PendingReview",
    Approved: "Approved",
    Archived: "Archived",
};
exports.ModelPackageS3DataSourceS3DataType = {
    ManifestFile: "ManifestFile",
    S3Prefix: "S3Prefix",
    AugmentedManifestFile: "AugmentedManifestFile",
};
exports.ModelPackageS3ModelDataSourceCompressionType = {
    None: "None",
    Gzip: "Gzip",
};
exports.ModelPackageS3ModelDataSourceS3DataType = {
    S3Prefix: "S3Prefix",
    S3Object: "S3Object",
};
exports.ModelPackageSkipModelValidation = {
    None: "None",
    All: "All",
};
exports.ModelPackageStatus = {
    Pending: "Pending",
    Deleting: "Deleting",
    InProgress: "InProgress",
    Completed: "Completed",
    Failed: "Failed",
};
exports.ModelPackageStatusItemStatus = {
    NotStarted: "NotStarted",
    Failed: "Failed",
    InProgress: "InProgress",
    Completed: "Completed",
};
exports.ModelPackageTransformInputCompressionType = {
    None: "None",
    Gzip: "Gzip",
};
exports.ModelPackageTransformInputSplitType = {
    None: "None",
    TfRecord: "TFRecord",
    Line: "Line",
    RecordIo: "RecordIO",
};
exports.ModelPackageTransformJobDefinitionBatchStrategy = {
    MultiRecord: "MultiRecord",
    SingleRecord: "SingleRecord",
};
exports.ModelPackageTransformOutputAssembleWith = {
    None: "None",
    Line: "Line",
};
exports.ModelQualityJobDefinitionBatchTransformInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelQualityJobDefinitionBatchTransformInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelQualityJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelQualityJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelQualityJobDefinitionProblemType = {
    BinaryClassification: "BinaryClassification",
    MulticlassClassification: "MulticlassClassification",
    Regression: "Regression",
};
exports.ModelQualityJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.MonitoringScheduleBatchTransformInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.MonitoringScheduleBatchTransformInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.MonitoringScheduleEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.MonitoringScheduleEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.MonitoringScheduleMonitoringExecutionSummaryMonitoringExecutionStatus = {
    Pending: "Pending",
    Completed: "Completed",
    CompletedWithViolations: "CompletedWithViolations",
    InProgress: "InProgress",
    Failed: "Failed",
    Stopping: "Stopping",
    Stopped: "Stopped",
};
exports.MonitoringScheduleMonitoringType = {
    DataQuality: "DataQuality",
    ModelQuality: "ModelQuality",
    ModelBias: "ModelBias",
    ModelExplainability: "ModelExplainability",
};
exports.MonitoringScheduleS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.MonitoringScheduleStatus = {
    Pending: "Pending",
    Failed: "Failed",
    Scheduled: "Scheduled",
    Stopped: "Stopped",
};
exports.PartnerAppAuthType = {
    Iam: "IAM",
};
exports.PartnerAppType = {
    LakeraGuard: "lakera-guard",
    Comet: "comet",
    DeepchecksLlmEvaluation: "deepchecks-llm-evaluation",
    Fiddler: "fiddler",
};
exports.ProjectStatus = {
    Pending: "Pending",
    CreateInProgress: "CreateInProgress",
    CreateCompleted: "CreateCompleted",
    CreateFailed: "CreateFailed",
    DeleteInProgress: "DeleteInProgress",
    DeleteFailed: "DeleteFailed",
    DeleteCompleted: "DeleteCompleted",
};
exports.SpaceAppType = {
    JupyterServer: "JupyterServer",
    KernelGateway: "KernelGateway",
    TensorBoard: "TensorBoard",
    RStudioServerPro: "RStudioServerPro",
    RSessionGateway: "RSessionGateway",
    JupyterLab: "JupyterLab",
    CodeEditor: "CodeEditor",
};
exports.SpaceResourceSpecInstanceType = {
    System: "system",
    MlT3Micro: "ml.t3.micro",
    MlT3Small: "ml.t3.small",
    MlT3Medium: "ml.t3.medium",
    MlT3Large: "ml.t3.large",
    MlT3Xlarge: "ml.t3.xlarge",
    MlT32xlarge: "ml.t3.2xlarge",
    MlM5Large: "ml.m5.large",
    MlM5Xlarge: "ml.m5.xlarge",
    MlM52xlarge: "ml.m5.2xlarge",
    MlM54xlarge: "ml.m5.4xlarge",
    MlM58xlarge: "ml.m5.8xlarge",
    MlM512xlarge: "ml.m5.12xlarge",
    MlM516xlarge: "ml.m5.16xlarge",
    MlM524xlarge: "ml.m5.24xlarge",
    MlC5Large: "ml.c5.large",
    MlC5Xlarge: "ml.c5.xlarge",
    MlC52xlarge: "ml.c5.2xlarge",
    MlC54xlarge: "ml.c5.4xlarge",
    MlC59xlarge: "ml.c5.9xlarge",
    MlC512xlarge: "ml.c5.12xlarge",
    MlC518xlarge: "ml.c5.18xlarge",
    MlC524xlarge: "ml.c5.24xlarge",
    MlP32xlarge: "ml.p3.2xlarge",
    MlP38xlarge: "ml.p3.8xlarge",
    MlP316xlarge: "ml.p3.16xlarge",
    MlG4dnXlarge: "ml.g4dn.xlarge",
    MlG4dn2xlarge: "ml.g4dn.2xlarge",
    MlG4dn4xlarge: "ml.g4dn.4xlarge",
    MlG4dn8xlarge: "ml.g4dn.8xlarge",
    MlG4dn12xlarge: "ml.g4dn.12xlarge",
    MlG4dn16xlarge: "ml.g4dn.16xlarge",
    MlR5Large: "ml.r5.large",
    MlR5Xlarge: "ml.r5.xlarge",
    MlR52xlarge: "ml.r5.2xlarge",
    MlR54xlarge: "ml.r5.4xlarge",
    MlR58xlarge: "ml.r5.8xlarge",
    MlR512xlarge: "ml.r5.12xlarge",
    MlR516xlarge: "ml.r5.16xlarge",
    MlR524xlarge: "ml.r5.24xlarge",
    MlP3dn24xlarge: "ml.p3dn.24xlarge",
    MlM5dLarge: "ml.m5d.large",
    MlM5dXlarge: "ml.m5d.xlarge",
    MlM5d2xlarge: "ml.m5d.2xlarge",
    MlM5d4xlarge: "ml.m5d.4xlarge",
    MlM5d8xlarge: "ml.m5d.8xlarge",
    MlM5d12xlarge: "ml.m5d.12xlarge",
    MlM5d16xlarge: "ml.m5d.16xlarge",
    MlM5d24xlarge: "ml.m5d.24xlarge",
    MlG5Xlarge: "ml.g5.xlarge",
    MlG52xlarge: "ml.g5.2xlarge",
    MlG54xlarge: "ml.g5.4xlarge",
    MlG58xlarge: "ml.g5.8xlarge",
    MlG512xlarge: "ml.g5.12xlarge",
    MlG516xlarge: "ml.g5.16xlarge",
    MlG524xlarge: "ml.g5.24xlarge",
    MlG548xlarge: "ml.g5.48xlarge",
    MlP4d24xlarge: "ml.p4d.24xlarge",
    MlP4de24xlarge: "ml.p4de.24xlarge",
    MlGeospatialInteractive: "ml.geospatial.interactive",
    MlTrn12xlarge: "ml.trn1.2xlarge",
    MlTrn132xlarge: "ml.trn1.32xlarge",
    MlTrn1n32xlarge: "ml.trn1n.32xlarge",
};
exports.SpaceSharingSettingsSharingType = {
    Private: "Private",
    Shared: "Shared",
};
exports.StudioLifecycleConfigAppType = {
    JupyterServer: "JupyterServer",
    KernelGateway: "KernelGateway",
    CodeEditor: "CodeEditor",
    JupyterLab: "JupyterLab",
};
exports.UserProfileAppType = {
    JupyterServer: "JupyterServer",
    TensorBoard: "TensorBoard",
    RStudioServerPro: "RStudioServerPro",
    JupyterLab: "JupyterLab",
    CodeEditor: "CodeEditor",
    DetailedProfiler: "DetailedProfiler",
    Canvas: "Canvas",
};
exports.UserProfileLifecycleManagement = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.UserProfileMlTools = {
    DataWrangler: "DataWrangler",
    FeatureStore: "FeatureStore",
    EmrClusters: "EmrClusters",
    AutoMl: "AutoMl",
    Experiments: "Experiments",
    Training: "Training",
    ModelEvaluation: "ModelEvaluation",
    Pipelines: "Pipelines",
    Models: "Models",
    JumpStart: "JumpStart",
    InferenceRecommender: "InferenceRecommender",
    Endpoints: "Endpoints",
    Projects: "Projects",
    InferenceOptimization: "InferenceOptimization",
    HyperPodClusters: "HyperPodClusters",
    Comet: "Comet",
    DeepchecksLlmEvaluation: "DeepchecksLLMEvaluation",
    Fiddler: "Fiddler",
    LakeraGuard: "LakeraGuard",
};
exports.UserProfileRStudioServerProAppSettingsAccessStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.UserProfileRStudioServerProAppSettingsUserGroup = {
    RStudioAdmin: "R_STUDIO_ADMIN",
    RStudioUser: "R_STUDIO_USER",
};
exports.UserProfileResourceSpecInstanceType = {
    System: "system",
    MlT3Micro: "ml.t3.micro",
    MlT3Small: "ml.t3.small",
    MlT3Medium: "ml.t3.medium",
    MlT3Large: "ml.t3.large",
    MlT3Xlarge: "ml.t3.xlarge",
    MlT32xlarge: "ml.t3.2xlarge",
    MlM5Large: "ml.m5.large",
    MlM5Xlarge: "ml.m5.xlarge",
    MlM52xlarge: "ml.m5.2xlarge",
    MlM54xlarge: "ml.m5.4xlarge",
    MlM58xlarge: "ml.m5.8xlarge",
    MlM512xlarge: "ml.m5.12xlarge",
    MlM516xlarge: "ml.m5.16xlarge",
    MlM524xlarge: "ml.m5.24xlarge",
    MlC5Large: "ml.c5.large",
    MlC5Xlarge: "ml.c5.xlarge",
    MlC52xlarge: "ml.c5.2xlarge",
    MlC54xlarge: "ml.c5.4xlarge",
    MlC59xlarge: "ml.c5.9xlarge",
    MlC512xlarge: "ml.c5.12xlarge",
    MlC518xlarge: "ml.c5.18xlarge",
    MlC524xlarge: "ml.c5.24xlarge",
    MlP32xlarge: "ml.p3.2xlarge",
    MlP38xlarge: "ml.p3.8xlarge",
    MlP316xlarge: "ml.p3.16xlarge",
    MlG4dnXlarge: "ml.g4dn.xlarge",
    MlG4dn2xlarge: "ml.g4dn.2xlarge",
    MlG4dn4xlarge: "ml.g4dn.4xlarge",
    MlG4dn8xlarge: "ml.g4dn.8xlarge",
    MlG4dn12xlarge: "ml.g4dn.12xlarge",
    MlG4dn16xlarge: "ml.g4dn.16xlarge",
    MlR5Large: "ml.r5.large",
    MlR5Xlarge: "ml.r5.xlarge",
    MlR52xlarge: "ml.r5.2xlarge",
    MlR54xlarge: "ml.r5.4xlarge",
    MlR58xlarge: "ml.r5.8xlarge",
    MlR512xlarge: "ml.r5.12xlarge",
    MlR516xlarge: "ml.r5.16xlarge",
    MlR524xlarge: "ml.r5.24xlarge",
    MlP3dn24xlarge: "ml.p3dn.24xlarge",
    MlM5dLarge: "ml.m5d.large",
    MlM5dXlarge: "ml.m5d.xlarge",
    MlM5d2xlarge: "ml.m5d.2xlarge",
    MlM5d4xlarge: "ml.m5d.4xlarge",
    MlM5d8xlarge: "ml.m5d.8xlarge",
    MlM5d12xlarge: "ml.m5d.12xlarge",
    MlM5d16xlarge: "ml.m5d.16xlarge",
    MlM5d24xlarge: "ml.m5d.24xlarge",
    MlG5Xlarge: "ml.g5.xlarge",
    MlG52xlarge: "ml.g5.2xlarge",
    MlG54xlarge: "ml.g5.4xlarge",
    MlG58xlarge: "ml.g5.8xlarge",
    MlG512xlarge: "ml.g5.12xlarge",
    MlG516xlarge: "ml.g5.16xlarge",
    MlG524xlarge: "ml.g5.24xlarge",
    MlG548xlarge: "ml.g5.48xlarge",
    MlP4d24xlarge: "ml.p4d.24xlarge",
    MlP4de24xlarge: "ml.p4de.24xlarge",
    MlGeospatialInteractive: "ml.geospatial.interactive",
    MlTrn12xlarge: "ml.trn1.2xlarge",
    MlTrn132xlarge: "ml.trn1.32xlarge",
    MlTrn1n32xlarge: "ml.trn1n.32xlarge",
};
exports.UserProfileSharingSettingsNotebookOutputOption = {
    Allowed: "Allowed",
    Disabled: "Disabled",
};
exports.UserProfileUserSettingsStudioWebPortal = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
