// @pulumi/aws-native/types/enums/timestream/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TablePartitionKeyType = exports.TablePartitionKeyEnforcementLevel = exports.ScheduledQueryMultiMeasureAttributeMappingMeasureValueType = exports.ScheduledQueryMixedMeasureMappingMeasureValueType = exports.ScheduledQueryEncryptionOption = exports.ScheduledQueryDimensionValueType = exports.InfluxDbInstanceStatus = exports.InfluxDbInstanceDeploymentType = exports.InfluxDbInstanceDbStorageType = exports.InfluxDbInstanceDbInstanceType = void 0;
exports.InfluxDbInstanceDbInstanceType = {
    DbInfluxMedium: "db.influx.medium",
    DbInfluxLarge: "db.influx.large",
    DbInfluxXlarge: "db.influx.xlarge",
    DbInflux2xlarge: "db.influx.2xlarge",
    DbInflux4xlarge: "db.influx.4xlarge",
    DbInflux8xlarge: "db.influx.8xlarge",
    DbInflux12xlarge: "db.influx.12xlarge",
    DbInflux16xlarge: "db.influx.16xlarge",
};
exports.InfluxDbInstanceDbStorageType = {
    InfluxIoIncludedT1: "InfluxIOIncludedT1",
    InfluxIoIncludedT2: "InfluxIOIncludedT2",
    InfluxIoIncludedT3: "InfluxIOIncludedT3",
};
exports.InfluxDbInstanceDeploymentType = {
    SingleAz: "SINGLE_AZ",
    WithMultiazStandby: "WITH_MULTIAZ_STANDBY",
};
exports.InfluxDbInstanceStatus = {
    Creating: "CREATING",
    Available: "AVAILABLE",
    Deleting: "DELETING",
    Modifying: "MODIFYING",
    Updating: "UPDATING",
    Deleted: "DELETED",
    Failed: "FAILED",
};
exports.ScheduledQueryDimensionValueType = {
    Varchar: "VARCHAR",
};
exports.ScheduledQueryEncryptionOption = {
    SseS3: "SSE_S3",
    SseKms: "SSE_KMS",
};
exports.ScheduledQueryMixedMeasureMappingMeasureValueType = {
    Bigint: "BIGINT",
    Boolean: "BOOLEAN",
    Double: "DOUBLE",
    Varchar: "VARCHAR",
    Multi: "MULTI",
};
exports.ScheduledQueryMultiMeasureAttributeMappingMeasureValueType = {
    Bigint: "BIGINT",
    Boolean: "BOOLEAN",
    Double: "DOUBLE",
    Varchar: "VARCHAR",
    Timestamp: "TIMESTAMP",
};
exports.TablePartitionKeyEnforcementLevel = {
    Required: "REQUIRED",
    Optional: "OPTIONAL",
};
exports.TablePartitionKeyType = {
    Dimension: "DIMENSION",
    Measure: "MEASURE",
};
