// @pulumi/aws-native/cleanrooms/idNamespaceAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdNamespaceAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents an association between an ID namespace and a collaboration
 */
class IdNamespaceAssociation extends pulumi.CustomResource {
    /**
     * Get an existing IdNamespaceAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IdNamespaceAssociation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IdNamespaceAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IdNamespaceAssociation.__pulumiType;
    }
    /**
     * Create a IdNamespaceAssociation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.inputReferenceConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inputReferenceConfig'");
            }
            if ((!args || args.membershipIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'membershipIdentifier'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["idMappingConfig"] = args ? args.idMappingConfig : undefined;
            resourceInputs["inputReferenceConfig"] = args ? args.inputReferenceConfig : undefined;
            resourceInputs["membershipIdentifier"] = args ? args.membershipIdentifier : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["collaborationArn"] = undefined /*out*/;
            resourceInputs["collaborationIdentifier"] = undefined /*out*/;
            resourceInputs["idNamespaceAssociationIdentifier"] = undefined /*out*/;
            resourceInputs["inputReferenceProperties"] = undefined /*out*/;
            resourceInputs["membershipArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["collaborationArn"] = undefined /*out*/;
            resourceInputs["collaborationIdentifier"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["idMappingConfig"] = undefined /*out*/;
            resourceInputs["idNamespaceAssociationIdentifier"] = undefined /*out*/;
            resourceInputs["inputReferenceConfig"] = undefined /*out*/;
            resourceInputs["inputReferenceProperties"] = undefined /*out*/;
            resourceInputs["membershipArn"] = undefined /*out*/;
            resourceInputs["membershipIdentifier"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["inputReferenceConfig", "membershipIdentifier"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(IdNamespaceAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IdNamespaceAssociation = IdNamespaceAssociation;
/** @internal */
IdNamespaceAssociation.__pulumiType = 'aws-native:cleanrooms:IdNamespaceAssociation';
