// @pulumi/aws-native/cleanrooms/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivacyBudgetTemplate = exports.Membership = exports.IdNamespaceAssociation = exports.IdMappingTable = exports.getPrivacyBudgetTemplateOutput = exports.getPrivacyBudgetTemplate = exports.getMembershipOutput = exports.getMembership = exports.getIdNamespaceAssociationOutput = exports.getIdNamespaceAssociation = exports.getIdMappingTableOutput = exports.getIdMappingTable = exports.getConfiguredTableAssociationOutput = exports.getConfiguredTableAssociation = exports.getConfiguredTableOutput = exports.getConfiguredTable = exports.getCollaborationOutput = exports.getCollaboration = exports.getAnalysisTemplateOutput = exports.getAnalysisTemplate = exports.ConfiguredTableAssociation = exports.ConfiguredTable = exports.Collaboration = exports.AnalysisTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnalysisTemplate = null;
utilities.lazyLoad(exports, ["AnalysisTemplate"], () => require("./analysisTemplate"));
exports.Collaboration = null;
utilities.lazyLoad(exports, ["Collaboration"], () => require("./collaboration"));
exports.ConfiguredTable = null;
utilities.lazyLoad(exports, ["ConfiguredTable"], () => require("./configuredTable"));
exports.ConfiguredTableAssociation = null;
utilities.lazyLoad(exports, ["ConfiguredTableAssociation"], () => require("./configuredTableAssociation"));
exports.getAnalysisTemplate = null;
exports.getAnalysisTemplateOutput = null;
utilities.lazyLoad(exports, ["getAnalysisTemplate", "getAnalysisTemplateOutput"], () => require("./getAnalysisTemplate"));
exports.getCollaboration = null;
exports.getCollaborationOutput = null;
utilities.lazyLoad(exports, ["getCollaboration", "getCollaborationOutput"], () => require("./getCollaboration"));
exports.getConfiguredTable = null;
exports.getConfiguredTableOutput = null;
utilities.lazyLoad(exports, ["getConfiguredTable", "getConfiguredTableOutput"], () => require("./getConfiguredTable"));
exports.getConfiguredTableAssociation = null;
exports.getConfiguredTableAssociationOutput = null;
utilities.lazyLoad(exports, ["getConfiguredTableAssociation", "getConfiguredTableAssociationOutput"], () => require("./getConfiguredTableAssociation"));
exports.getIdMappingTable = null;
exports.getIdMappingTableOutput = null;
utilities.lazyLoad(exports, ["getIdMappingTable", "getIdMappingTableOutput"], () => require("./getIdMappingTable"));
exports.getIdNamespaceAssociation = null;
exports.getIdNamespaceAssociationOutput = null;
utilities.lazyLoad(exports, ["getIdNamespaceAssociation", "getIdNamespaceAssociationOutput"], () => require("./getIdNamespaceAssociation"));
exports.getMembership = null;
exports.getMembershipOutput = null;
utilities.lazyLoad(exports, ["getMembership", "getMembershipOutput"], () => require("./getMembership"));
exports.getPrivacyBudgetTemplate = null;
exports.getPrivacyBudgetTemplateOutput = null;
utilities.lazyLoad(exports, ["getPrivacyBudgetTemplate", "getPrivacyBudgetTemplateOutput"], () => require("./getPrivacyBudgetTemplate"));
exports.IdMappingTable = null;
utilities.lazyLoad(exports, ["IdMappingTable"], () => require("./idMappingTable"));
exports.IdNamespaceAssociation = null;
utilities.lazyLoad(exports, ["IdNamespaceAssociation"], () => require("./idNamespaceAssociation"));
exports.Membership = null;
utilities.lazyLoad(exports, ["Membership"], () => require("./membership"));
exports.PrivacyBudgetTemplate = null;
utilities.lazyLoad(exports, ["PrivacyBudgetTemplate"], () => require("./privacyBudgetTemplate"));
// Export enums:
__exportStar(require("../types/enums/cleanrooms"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:cleanrooms:AnalysisTemplate":
                return new exports.AnalysisTemplate(name, undefined, { urn });
            case "aws-native:cleanrooms:Collaboration":
                return new exports.Collaboration(name, undefined, { urn });
            case "aws-native:cleanrooms:ConfiguredTable":
                return new exports.ConfiguredTable(name, undefined, { urn });
            case "aws-native:cleanrooms:ConfiguredTableAssociation":
                return new exports.ConfiguredTableAssociation(name, undefined, { urn });
            case "aws-native:cleanrooms:IdMappingTable":
                return new exports.IdMappingTable(name, undefined, { urn });
            case "aws-native:cleanrooms:IdNamespaceAssociation":
                return new exports.IdNamespaceAssociation(name, undefined, { urn });
            case "aws-native:cleanrooms:Membership":
                return new exports.Membership(name, undefined, { urn });
            case "aws-native:cleanrooms:PrivacyBudgetTemplate":
                return new exports.PrivacyBudgetTemplate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "cleanrooms", _module);
