// @pulumi/aws-native/cleanrooms/privacyBudgetTemplate.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivacyBudgetTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a privacy budget within a collaboration
 */
class PrivacyBudgetTemplate extends pulumi.CustomResource {
    /**
     * Get an existing PrivacyBudgetTemplate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivacyBudgetTemplate(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivacyBudgetTemplate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivacyBudgetTemplate.__pulumiType;
    }
    /**
     * Create a PrivacyBudgetTemplate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.autoRefresh === undefined) && !opts.urn) {
                throw new Error("Missing required property 'autoRefresh'");
            }
            if ((!args || args.membershipIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'membershipIdentifier'");
            }
            if ((!args || args.parameters === undefined) && !opts.urn) {
                throw new Error("Missing required property 'parameters'");
            }
            if ((!args || args.privacyBudgetType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privacyBudgetType'");
            }
            resourceInputs["autoRefresh"] = args ? args.autoRefresh : undefined;
            resourceInputs["membershipIdentifier"] = args ? args.membershipIdentifier : undefined;
            resourceInputs["parameters"] = args ? args.parameters : undefined;
            resourceInputs["privacyBudgetType"] = args ? args.privacyBudgetType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["collaborationArn"] = undefined /*out*/;
            resourceInputs["collaborationIdentifier"] = undefined /*out*/;
            resourceInputs["membershipArn"] = undefined /*out*/;
            resourceInputs["privacyBudgetTemplateIdentifier"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["autoRefresh"] = undefined /*out*/;
            resourceInputs["collaborationArn"] = undefined /*out*/;
            resourceInputs["collaborationIdentifier"] = undefined /*out*/;
            resourceInputs["membershipArn"] = undefined /*out*/;
            resourceInputs["membershipIdentifier"] = undefined /*out*/;
            resourceInputs["parameters"] = undefined /*out*/;
            resourceInputs["privacyBudgetTemplateIdentifier"] = undefined /*out*/;
            resourceInputs["privacyBudgetType"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["autoRefresh", "membershipIdentifier", "privacyBudgetType"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(PrivacyBudgetTemplate.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivacyBudgetTemplate = PrivacyBudgetTemplate;
/** @internal */
PrivacyBudgetTemplate.__pulumiType = 'aws-native:cleanrooms:PrivacyBudgetTemplate';
