// @pulumi/aws-native/types/enums/scheduler/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleState = exports.SchedulePropagateTags = exports.SchedulePlacementStrategyType = exports.SchedulePlacementConstraintType = exports.ScheduleLaunchType = exports.ScheduleGroupState = exports.ScheduleFlexibleTimeWindowMode = exports.ScheduleAssignPublicIp = void 0;
exports.ScheduleAssignPublicIp = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.ScheduleFlexibleTimeWindowMode = {
    Off: "OFF",
    Flexible: "FLEXIBLE",
};
exports.ScheduleGroupState = {
    Active: "ACTIVE",
    Deleting: "DELETING",
};
exports.ScheduleLaunchType = {
    Ec2: "EC2",
    Fargate: "FARGATE",
    External: "EXTERNAL",
};
exports.SchedulePlacementConstraintType = {
    DistinctInstance: "distinctInstance",
    MemberOf: "memberOf",
};
exports.SchedulePlacementStrategyType = {
    Random: "random",
    Spread: "spread",
    Binpack: "binpack",
};
exports.SchedulePropagateTags = {
    TaskDefinition: "TASK_DEFINITION",
};
exports.ScheduleState = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
