// @pulumi/aws-native/types/enums/healthlake/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FhirDatastorePreloadDataConfigPreloadDataType = exports.FhirDatastoreKmsEncryptionConfigCmkType = exports.FhirDatastoreIdentityProviderConfigurationAuthorizationStrategy = exports.FhirDatastoreDatastoreTypeVersion = exports.FhirDatastoreDatastoreStatus = void 0;
exports.FhirDatastoreDatastoreStatus = {
    Creating: "CREATING",
    Active: "ACTIVE",
    Deleting: "DELETING",
    Deleted: "DELETED",
};
exports.FhirDatastoreDatastoreTypeVersion = {
    R4: "R4",
};
exports.FhirDatastoreIdentityProviderConfigurationAuthorizationStrategy = {
    SmartOnFhirV1: "SMART_ON_FHIR_V1",
    AwsAuth: "AWS_AUTH",
};
exports.FhirDatastoreKmsEncryptionConfigCmkType = {
    CustomerManagedKmsKey: "CUSTOMER_MANAGED_KMS_KEY",
    AwsOwnedKmsKey: "AWS_OWNED_KMS_KEY",
};
exports.FhirDatastorePreloadDataConfigPreloadDataType = {
    Synthea: "SYNTHEA",
};
