// @pulumi/aws-native/appflow/connector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Connector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::AppFlow::Connector
 */
class Connector extends pulumi.CustomResource {
    /**
     * Get an existing Connector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Connector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Connector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Connector.__pulumiType;
    }
    /**
     * Create a Connector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.connectorProvisioningConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectorProvisioningConfig'");
            }
            if ((!args || args.connectorProvisioningType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectorProvisioningType'");
            }
            resourceInputs["connectorLabel"] = args ? args.connectorLabel : undefined;
            resourceInputs["connectorProvisioningConfig"] = args ? args.connectorProvisioningConfig : undefined;
            resourceInputs["connectorProvisioningType"] = args ? args.connectorProvisioningType : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["connectorArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["connectorArn"] = undefined /*out*/;
            resourceInputs["connectorLabel"] = undefined /*out*/;
            resourceInputs["connectorProvisioningConfig"] = undefined /*out*/;
            resourceInputs["connectorProvisioningType"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["connectorLabel"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(Connector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Connector = Connector;
/** @internal */
Connector.__pulumiType = 'aws-native:appflow:Connector';
