// @pulumi/aws-native/resourcegroups/tagSyncTask.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagSyncTask = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Schema for ResourceGroups::TagSyncTask
 */
class TagSyncTask extends pulumi.CustomResource {
    /**
     * Get an existing TagSyncTask resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TagSyncTask(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TagSyncTask.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TagSyncTask.__pulumiType;
    }
    /**
     * Create a TagSyncTask resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.group === undefined) && !opts.urn) {
                throw new Error("Missing required property 'group'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            if ((!args || args.tagKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tagKey'");
            }
            if ((!args || args.tagValue === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tagValue'");
            }
            resourceInputs["group"] = args ? args.group : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["tagKey"] = args ? args.tagKey : undefined;
            resourceInputs["tagValue"] = args ? args.tagValue : undefined;
            resourceInputs["groupArn"] = undefined /*out*/;
            resourceInputs["groupName"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["taskArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["group"] = undefined /*out*/;
            resourceInputs["groupArn"] = undefined /*out*/;
            resourceInputs["groupName"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagKey"] = undefined /*out*/;
            resourceInputs["tagValue"] = undefined /*out*/;
            resourceInputs["taskArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["group", "roleArn", "tagKey", "tagValue"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(TagSyncTask.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TagSyncTask = TagSyncTask;
/** @internal */
TagSyncTask.__pulumiType = 'aws-native:resourcegroups:TagSyncTask';
