// @pulumi/aws-native/iot/domainConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Create and manage a Domain Configuration
 */
class DomainConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing DomainConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DomainConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DomainConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainConfiguration.__pulumiType;
    }
    /**
     * Create a DomainConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["applicationProtocol"] = args ? args.applicationProtocol : undefined;
            resourceInputs["authenticationType"] = args ? args.authenticationType : undefined;
            resourceInputs["authorizerConfig"] = args ? args.authorizerConfig : undefined;
            resourceInputs["clientCertificateConfig"] = args ? args.clientCertificateConfig : undefined;
            resourceInputs["domainConfigurationName"] = args ? args.domainConfigurationName : undefined;
            resourceInputs["domainConfigurationStatus"] = args ? args.domainConfigurationStatus : undefined;
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["serverCertificateArns"] = args ? args.serverCertificateArns : undefined;
            resourceInputs["serverCertificateConfig"] = args ? args.serverCertificateConfig : undefined;
            resourceInputs["serviceType"] = args ? args.serviceType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tlsConfig"] = args ? args.tlsConfig : undefined;
            resourceInputs["validationCertificateArn"] = args ? args.validationCertificateArn : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["domainType"] = undefined /*out*/;
            resourceInputs["serverCertificates"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationProtocol"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["authenticationType"] = undefined /*out*/;
            resourceInputs["authorizerConfig"] = undefined /*out*/;
            resourceInputs["clientCertificateConfig"] = undefined /*out*/;
            resourceInputs["domainConfigurationName"] = undefined /*out*/;
            resourceInputs["domainConfigurationStatus"] = undefined /*out*/;
            resourceInputs["domainName"] = undefined /*out*/;
            resourceInputs["domainType"] = undefined /*out*/;
            resourceInputs["serverCertificateArns"] = undefined /*out*/;
            resourceInputs["serverCertificateConfig"] = undefined /*out*/;
            resourceInputs["serverCertificates"] = undefined /*out*/;
            resourceInputs["serviceType"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["tlsConfig"] = undefined /*out*/;
            resourceInputs["validationCertificateArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["domainConfigurationName", "domainName", "serverCertificateArns[*]", "serviceType", "validationCertificateArn"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(DomainConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DomainConfiguration = DomainConfiguration;
/** @internal */
DomainConfiguration.__pulumiType = 'aws-native:iot:DomainConfiguration';
