// @pulumi/aws-native/iot/thing.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Thing = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::IoT::Thing
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const config = new pulumi.Config();
 * const nameParameter = config.require("nameParameter");
 * const myAttributeValueA = config.get("myAttributeValueA") || "myStringA123";
 * const myAttributeValueB = config.get("myAttributeValueB") || "myStringB123";
 * const myAttributeValueC = config.get("myAttributeValueC") || "myStringC123";
 * const myThing = new aws_native.iot.Thing("myThing", {
 *     thingName: nameParameter,
 *     attributePayload: {
 *         attributes: {
 *             myAttributeA: myAttributeValueA,
 *             myAttributeB: myAttributeValueB,
 *             myAttributeC: myAttributeValueC,
 *         },
 *     },
 * });
 *
 * ```
 */
class Thing extends pulumi.CustomResource {
    /**
     * Get an existing Thing resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Thing(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Thing.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Thing.__pulumiType;
    }
    /**
     * Create a Thing resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["attributePayload"] = args ? args.attributePayload : undefined;
            resourceInputs["thingName"] = args ? args.thingName : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["awsId"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["attributePayload"] = undefined /*out*/;
            resourceInputs["awsId"] = undefined /*out*/;
            resourceInputs["thingName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["thingName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(Thing.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Thing = Thing;
/** @internal */
Thing.__pulumiType = 'aws-native:iot:Thing';
