// @pulumi/aws-native/bedrock/guardrailVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GuardrailVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Bedrock::GuardrailVersion Resource Type
 */
class GuardrailVersion extends pulumi.CustomResource {
    /**
     * Get an existing GuardrailVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GuardrailVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GuardrailVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GuardrailVersion.__pulumiType;
    }
    /**
     * Create a GuardrailVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.guardrailIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'guardrailIdentifier'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["guardrailIdentifier"] = args ? args.guardrailIdentifier : undefined;
            resourceInputs["guardrailArn"] = undefined /*out*/;
            resourceInputs["guardrailId"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        else {
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["guardrailArn"] = undefined /*out*/;
            resourceInputs["guardrailId"] = undefined /*out*/;
            resourceInputs["guardrailIdentifier"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["description", "guardrailIdentifier"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(GuardrailVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GuardrailVersion = GuardrailVersion;
/** @internal */
GuardrailVersion.__pulumiType = 'aws-native:bedrock:GuardrailVersion';
