// @pulumi/aws-native/fis/targetAccountConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetAccountConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::FIS::TargetAccountConfiguration
 */
class TargetAccountConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing TargetAccountConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TargetAccountConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TargetAccountConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TargetAccountConfiguration.__pulumiType;
    }
    /**
     * Create a TargetAccountConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.experimentTemplateId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'experimentTemplateId'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["experimentTemplateId"] = args ? args.experimentTemplateId : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
        }
        else {
            resourceInputs["accountId"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["experimentTemplateId"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["accountId", "experimentTemplateId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(TargetAccountConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TargetAccountConfiguration = TargetAccountConfiguration;
/** @internal */
TargetAccountConfiguration.__pulumiType = 'aws-native:fis:TargetAccountConfiguration';
