// @pulumi/aws-native/workspaces/workspacesPool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspacesPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::WorkSpaces::WorkspacesPool
 */
class WorkspacesPool extends pulumi.CustomResource {
    /**
     * Get an existing WorkspacesPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkspacesPool(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkspacesPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkspacesPool.__pulumiType;
    }
    /**
     * Create a WorkspacesPool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.bundleId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'bundleId'");
            }
            if ((!args || args.capacity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'capacity'");
            }
            if ((!args || args.directoryId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'directoryId'");
            }
            resourceInputs["applicationSettings"] = args ? args.applicationSettings : undefined;
            resourceInputs["bundleId"] = args ? args.bundleId : undefined;
            resourceInputs["capacity"] = args ? args.capacity : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["directoryId"] = args ? args.directoryId : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["timeoutSettings"] = args ? args.timeoutSettings : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["poolArn"] = undefined /*out*/;
            resourceInputs["poolId"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationSettings"] = undefined /*out*/;
            resourceInputs["bundleId"] = undefined /*out*/;
            resourceInputs["capacity"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["directoryId"] = undefined /*out*/;
            resourceInputs["poolArn"] = undefined /*out*/;
            resourceInputs["poolId"] = undefined /*out*/;
            resourceInputs["poolName"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["timeoutSettings"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["poolName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(WorkspacesPool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkspacesPool = WorkspacesPool;
/** @internal */
WorkspacesPool.__pulumiType = 'aws-native:workspaces:WorkspacesPool';
