// @pulumi/aws-native/mediapackagev2/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OriginEndpointPolicy = exports.OriginEndpoint = exports.getOriginEndpointPolicyOutput = exports.getOriginEndpointPolicy = exports.getOriginEndpointOutput = exports.getOriginEndpoint = exports.getChannelPolicyOutput = exports.getChannelPolicy = exports.getChannelGroupOutput = exports.getChannelGroup = exports.getChannelOutput = exports.getChannel = exports.ChannelPolicy = exports.ChannelGroup = exports.Channel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Channel = null;
utilities.lazyLoad(exports, ["Channel"], () => require("./channel"));
exports.ChannelGroup = null;
utilities.lazyLoad(exports, ["ChannelGroup"], () => require("./channelGroup"));
exports.ChannelPolicy = null;
utilities.lazyLoad(exports, ["ChannelPolicy"], () => require("./channelPolicy"));
exports.getChannel = null;
exports.getChannelOutput = null;
utilities.lazyLoad(exports, ["getChannel", "getChannelOutput"], () => require("./getChannel"));
exports.getChannelGroup = null;
exports.getChannelGroupOutput = null;
utilities.lazyLoad(exports, ["getChannelGroup", "getChannelGroupOutput"], () => require("./getChannelGroup"));
exports.getChannelPolicy = null;
exports.getChannelPolicyOutput = null;
utilities.lazyLoad(exports, ["getChannelPolicy", "getChannelPolicyOutput"], () => require("./getChannelPolicy"));
exports.getOriginEndpoint = null;
exports.getOriginEndpointOutput = null;
utilities.lazyLoad(exports, ["getOriginEndpoint", "getOriginEndpointOutput"], () => require("./getOriginEndpoint"));
exports.getOriginEndpointPolicy = null;
exports.getOriginEndpointPolicyOutput = null;
utilities.lazyLoad(exports, ["getOriginEndpointPolicy", "getOriginEndpointPolicyOutput"], () => require("./getOriginEndpointPolicy"));
exports.OriginEndpoint = null;
utilities.lazyLoad(exports, ["OriginEndpoint"], () => require("./originEndpoint"));
exports.OriginEndpointPolicy = null;
utilities.lazyLoad(exports, ["OriginEndpointPolicy"], () => require("./originEndpointPolicy"));
// Export enums:
__exportStar(require("../types/enums/mediapackagev2"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:mediapackagev2:Channel":
                return new exports.Channel(name, undefined, { urn });
            case "aws-native:mediapackagev2:ChannelGroup":
                return new exports.ChannelGroup(name, undefined, { urn });
            case "aws-native:mediapackagev2:ChannelPolicy":
                return new exports.ChannelPolicy(name, undefined, { urn });
            case "aws-native:mediapackagev2:OriginEndpoint":
                return new exports.OriginEndpoint(name, undefined, { urn });
            case "aws-native:mediapackagev2:OriginEndpointPolicy":
                return new exports.OriginEndpointPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "mediapackagev2", _module);
