// @pulumi/aws-native/mediapackagev2/originEndpointPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OriginEndpointPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * <p>Represents a resource policy that allows or denies access to an origin endpoint.</p>
 */
class OriginEndpointPolicy extends pulumi.CustomResource {
    /**
     * Get an existing OriginEndpointPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OriginEndpointPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OriginEndpointPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OriginEndpointPolicy.__pulumiType;
    }
    /**
     * Create a OriginEndpointPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.channelGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'channelGroupName'");
            }
            if ((!args || args.channelName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'channelName'");
            }
            if ((!args || args.originEndpointName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'originEndpointName'");
            }
            if ((!args || args.policy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'policy'");
            }
            resourceInputs["channelGroupName"] = args ? args.channelGroupName : undefined;
            resourceInputs["channelName"] = args ? args.channelName : undefined;
            resourceInputs["originEndpointName"] = args ? args.originEndpointName : undefined;
            resourceInputs["policy"] = args ? args.policy : undefined;
        }
        else {
            resourceInputs["channelGroupName"] = undefined /*out*/;
            resourceInputs["channelName"] = undefined /*out*/;
            resourceInputs["originEndpointName"] = undefined /*out*/;
            resourceInputs["policy"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["channelGroupName", "channelName", "originEndpointName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(OriginEndpointPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OriginEndpointPolicy = OriginEndpointPolicy;
/** @internal */
OriginEndpointPolicy.__pulumiType = 'aws-native:mediapackagev2:OriginEndpointPolicy';
