// @pulumi/aws-native/qbusiness/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebExperience = exports.Retriever = exports.Plugin = exports.Permission = exports.Index = exports.getWebExperienceOutput = exports.getWebExperience = exports.getRetrieverOutput = exports.getRetriever = exports.getPluginOutput = exports.getPlugin = exports.getIndexOutput = exports.getIndex = exports.getDataSourceOutput = exports.getDataSource = exports.getDataAccessorOutput = exports.getDataAccessor = exports.getApplicationOutput = exports.getApplication = exports.DataSource = exports.DataAccessor = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.DataAccessor = null;
utilities.lazyLoad(exports, ["DataAccessor"], () => require("./dataAccessor"));
exports.DataSource = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
exports.getApplication = null;
exports.getApplicationOutput = null;
utilities.lazyLoad(exports, ["getApplication", "getApplicationOutput"], () => require("./getApplication"));
exports.getDataAccessor = null;
exports.getDataAccessorOutput = null;
utilities.lazyLoad(exports, ["getDataAccessor", "getDataAccessorOutput"], () => require("./getDataAccessor"));
exports.getDataSource = null;
exports.getDataSourceOutput = null;
utilities.lazyLoad(exports, ["getDataSource", "getDataSourceOutput"], () => require("./getDataSource"));
exports.getIndex = null;
exports.getIndexOutput = null;
utilities.lazyLoad(exports, ["getIndex", "getIndexOutput"], () => require("./getIndex"));
exports.getPlugin = null;
exports.getPluginOutput = null;
utilities.lazyLoad(exports, ["getPlugin", "getPluginOutput"], () => require("./getPlugin"));
exports.getRetriever = null;
exports.getRetrieverOutput = null;
utilities.lazyLoad(exports, ["getRetriever", "getRetrieverOutput"], () => require("./getRetriever"));
exports.getWebExperience = null;
exports.getWebExperienceOutput = null;
utilities.lazyLoad(exports, ["getWebExperience", "getWebExperienceOutput"], () => require("./getWebExperience"));
exports.Index = null;
utilities.lazyLoad(exports, ["Index"], () => require("./index_"));
exports.Permission = null;
utilities.lazyLoad(exports, ["Permission"], () => require("./permission"));
exports.Plugin = null;
utilities.lazyLoad(exports, ["Plugin"], () => require("./plugin"));
exports.Retriever = null;
utilities.lazyLoad(exports, ["Retriever"], () => require("./retriever"));
exports.WebExperience = null;
utilities.lazyLoad(exports, ["WebExperience"], () => require("./webExperience"));
// Export enums:
__exportStar(require("../types/enums/qbusiness"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:qbusiness:Application":
                return new exports.Application(name, undefined, { urn });
            case "aws-native:qbusiness:DataAccessor":
                return new exports.DataAccessor(name, undefined, { urn });
            case "aws-native:qbusiness:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "aws-native:qbusiness:Index":
                return new exports.Index(name, undefined, { urn });
            case "aws-native:qbusiness:Permission":
                return new exports.Permission(name, undefined, { urn });
            case "aws-native:qbusiness:Plugin":
                return new exports.Plugin(name, undefined, { urn });
            case "aws-native:qbusiness:Retriever":
                return new exports.Retriever(name, undefined, { urn });
            case "aws-native:qbusiness:WebExperience":
                return new exports.WebExperience(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "qbusiness", _module);
