// @pulumi/aws-native/types/enums/applicationsignals/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelObjectiveSliMetricMetricType = exports.ServiceLevelObjectiveSliComparisonOperator = exports.ServiceLevelObjectiveRequestBasedSliMetricMetricType = exports.ServiceLevelObjectiveRequestBasedSliComparisonOperator = exports.ServiceLevelObjectiveEvaluationType = exports.ServiceLevelObjectiveDurationUnit = void 0;
exports.ServiceLevelObjectiveDurationUnit = {
    Day: "DAY",
    Month: "MONTH",
};
exports.ServiceLevelObjectiveEvaluationType = {
    PeriodBased: "PeriodBased",
    RequestBased: "RequestBased",
};
exports.ServiceLevelObjectiveRequestBasedSliComparisonOperator = {
    GreaterThanOrEqualTo: "GreaterThanOrEqualTo",
    LessThanOrEqualTo: "LessThanOrEqualTo",
    LessThan: "LessThan",
    GreaterThan: "GreaterThan",
};
exports.ServiceLevelObjectiveRequestBasedSliMetricMetricType = {
    Latency: "LATENCY",
    Availability: "AVAILABILITY",
};
exports.ServiceLevelObjectiveSliComparisonOperator = {
    GreaterThanOrEqualTo: "GreaterThanOrEqualTo",
    LessThanOrEqualTo: "LessThanOrEqualTo",
    LessThan: "LessThan",
    GreaterThan: "GreaterThan",
};
exports.ServiceLevelObjectiveSliMetricMetricType = {
    Latency: "LATENCY",
    Availability: "AVAILABILITY",
};
