// @pulumi/aws-native/appsync/channelNamespace.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AppSync ChannelNamespace
 */
class ChannelNamespace extends pulumi.CustomResource {
    /**
     * Get an existing ChannelNamespace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ChannelNamespace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ChannelNamespace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ChannelNamespace.__pulumiType;
    }
    /**
     * Create a ChannelNamespace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            resourceInputs["apiId"] = args ? args.apiId : undefined;
            resourceInputs["codeHandlers"] = args ? args.codeHandlers : undefined;
            resourceInputs["codeS3Location"] = args ? args.codeS3Location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["publishAuthModes"] = args ? args.publishAuthModes : undefined;
            resourceInputs["subscribeAuthModes"] = args ? args.subscribeAuthModes : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["channelNamespaceArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["apiId"] = undefined /*out*/;
            resourceInputs["channelNamespaceArn"] = undefined /*out*/;
            resourceInputs["codeHandlers"] = undefined /*out*/;
            resourceInputs["codeS3Location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["publishAuthModes"] = undefined /*out*/;
            resourceInputs["subscribeAuthModes"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["apiId", "name"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(ChannelNamespace.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ChannelNamespace = ChannelNamespace;
/** @internal */
ChannelNamespace.__pulumiType = 'aws-native:appsync:ChannelNamespace';
