// @pulumi/aws-native/appsync/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceApiAssociation = exports.Resolver = exports.GraphQlApi = exports.getSourceApiAssociationOutput = exports.getSourceApiAssociation = exports.getResolverOutput = exports.getResolver = exports.getGraphQlApiOutput = exports.getGraphQlApi = exports.getFunctionConfigurationOutput = exports.getFunctionConfiguration = exports.getDomainNameApiAssociationOutput = exports.getDomainNameApiAssociation = exports.getDomainNameOutput = exports.getDomainName = exports.getDataSourceOutput = exports.getDataSource = exports.getChannelNamespaceOutput = exports.getChannelNamespace = exports.getApiOutput = exports.getApi = exports.FunctionConfiguration = exports.DomainNameApiAssociation = exports.DomainName = exports.DataSource = exports.ChannelNamespace = exports.Api = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Api = null;
utilities.lazyLoad(exports, ["Api"], () => require("./api"));
exports.ChannelNamespace = null;
utilities.lazyLoad(exports, ["ChannelNamespace"], () => require("./channelNamespace"));
exports.DataSource = null;
utilities.lazyLoad(exports, ["DataSource"], () => require("./dataSource"));
exports.DomainName = null;
utilities.lazyLoad(exports, ["DomainName"], () => require("./domainName"));
exports.DomainNameApiAssociation = null;
utilities.lazyLoad(exports, ["DomainNameApiAssociation"], () => require("./domainNameApiAssociation"));
exports.FunctionConfiguration = null;
utilities.lazyLoad(exports, ["FunctionConfiguration"], () => require("./functionConfiguration"));
exports.getApi = null;
exports.getApiOutput = null;
utilities.lazyLoad(exports, ["getApi", "getApiOutput"], () => require("./getApi"));
exports.getChannelNamespace = null;
exports.getChannelNamespaceOutput = null;
utilities.lazyLoad(exports, ["getChannelNamespace", "getChannelNamespaceOutput"], () => require("./getChannelNamespace"));
exports.getDataSource = null;
exports.getDataSourceOutput = null;
utilities.lazyLoad(exports, ["getDataSource", "getDataSourceOutput"], () => require("./getDataSource"));
exports.getDomainName = null;
exports.getDomainNameOutput = null;
utilities.lazyLoad(exports, ["getDomainName", "getDomainNameOutput"], () => require("./getDomainName"));
exports.getDomainNameApiAssociation = null;
exports.getDomainNameApiAssociationOutput = null;
utilities.lazyLoad(exports, ["getDomainNameApiAssociation", "getDomainNameApiAssociationOutput"], () => require("./getDomainNameApiAssociation"));
exports.getFunctionConfiguration = null;
exports.getFunctionConfigurationOutput = null;
utilities.lazyLoad(exports, ["getFunctionConfiguration", "getFunctionConfigurationOutput"], () => require("./getFunctionConfiguration"));
exports.getGraphQlApi = null;
exports.getGraphQlApiOutput = null;
utilities.lazyLoad(exports, ["getGraphQlApi", "getGraphQlApiOutput"], () => require("./getGraphQlApi"));
exports.getResolver = null;
exports.getResolverOutput = null;
utilities.lazyLoad(exports, ["getResolver", "getResolverOutput"], () => require("./getResolver"));
exports.getSourceApiAssociation = null;
exports.getSourceApiAssociationOutput = null;
utilities.lazyLoad(exports, ["getSourceApiAssociation", "getSourceApiAssociationOutput"], () => require("./getSourceApiAssociation"));
exports.GraphQlApi = null;
utilities.lazyLoad(exports, ["GraphQlApi"], () => require("./graphQlApi"));
exports.Resolver = null;
utilities.lazyLoad(exports, ["Resolver"], () => require("./resolver"));
exports.SourceApiAssociation = null;
utilities.lazyLoad(exports, ["SourceApiAssociation"], () => require("./sourceApiAssociation"));
// Export enums:
__exportStar(require("../types/enums/appsync"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:appsync:Api":
                return new exports.Api(name, undefined, { urn });
            case "aws-native:appsync:ChannelNamespace":
                return new exports.ChannelNamespace(name, undefined, { urn });
            case "aws-native:appsync:DataSource":
                return new exports.DataSource(name, undefined, { urn });
            case "aws-native:appsync:DomainName":
                return new exports.DomainName(name, undefined, { urn });
            case "aws-native:appsync:DomainNameApiAssociation":
                return new exports.DomainNameApiAssociation(name, undefined, { urn });
            case "aws-native:appsync:FunctionConfiguration":
                return new exports.FunctionConfiguration(name, undefined, { urn });
            case "aws-native:appsync:GraphQlApi":
                return new exports.GraphQlApi(name, undefined, { urn });
            case "aws-native:appsync:Resolver":
                return new exports.Resolver(name, undefined, { urn });
            case "aws-native:appsync:SourceApiAssociation":
                return new exports.SourceApiAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "appsync", _module);
