// @pulumi/aws-native/sagemaker/mlflowTrackingServer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MlflowTrackingServer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SageMaker::MlflowTrackingServer
 */
class MlflowTrackingServer extends pulumi.CustomResource {
    /**
     * Get an existing MlflowTrackingServer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MlflowTrackingServer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MlflowTrackingServer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MlflowTrackingServer.__pulumiType;
    }
    /**
     * Create a MlflowTrackingServer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.artifactStoreUri === undefined) && !opts.urn) {
                throw new Error("Missing required property 'artifactStoreUri'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["artifactStoreUri"] = args ? args.artifactStoreUri : undefined;
            resourceInputs["automaticModelRegistration"] = args ? args.automaticModelRegistration : undefined;
            resourceInputs["mlflowVersion"] = args ? args.mlflowVersion : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["trackingServerName"] = args ? args.trackingServerName : undefined;
            resourceInputs["trackingServerSize"] = args ? args.trackingServerSize : undefined;
            resourceInputs["weeklyMaintenanceWindowStart"] = args ? args.weeklyMaintenanceWindowStart : undefined;
            resourceInputs["trackingServerArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["artifactStoreUri"] = undefined /*out*/;
            resourceInputs["automaticModelRegistration"] = undefined /*out*/;
            resourceInputs["mlflowVersion"] = undefined /*out*/;
            resourceInputs["roleArn"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["trackingServerArn"] = undefined /*out*/;
            resourceInputs["trackingServerName"] = undefined /*out*/;
            resourceInputs["trackingServerSize"] = undefined /*out*/;
            resourceInputs["weeklyMaintenanceWindowStart"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["trackingServerName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(MlflowTrackingServer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MlflowTrackingServer = MlflowTrackingServer;
/** @internal */
MlflowTrackingServer.__pulumiType = 'aws-native:sagemaker:MlflowTrackingServer';
