// @pulumi/aws-native/sagemaker/studioLifecycleConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioLifecycleConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SageMaker::StudioLifecycleConfig
 */
class StudioLifecycleConfig extends pulumi.CustomResource {
    /**
     * Get an existing StudioLifecycleConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StudioLifecycleConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StudioLifecycleConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StudioLifecycleConfig.__pulumiType;
    }
    /**
     * Create a StudioLifecycleConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.studioLifecycleConfigAppType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'studioLifecycleConfigAppType'");
            }
            if ((!args || args.studioLifecycleConfigContent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'studioLifecycleConfigContent'");
            }
            resourceInputs["studioLifecycleConfigAppType"] = args ? args.studioLifecycleConfigAppType : undefined;
            resourceInputs["studioLifecycleConfigContent"] = args ? args.studioLifecycleConfigContent : undefined;
            resourceInputs["studioLifecycleConfigName"] = args ? args.studioLifecycleConfigName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["studioLifecycleConfigArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["studioLifecycleConfigAppType"] = undefined /*out*/;
            resourceInputs["studioLifecycleConfigArn"] = undefined /*out*/;
            resourceInputs["studioLifecycleConfigContent"] = undefined /*out*/;
            resourceInputs["studioLifecycleConfigName"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["studioLifecycleConfigAppType", "studioLifecycleConfigContent", "studioLifecycleConfigName", "tags[*]"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(StudioLifecycleConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StudioLifecycleConfig = StudioLifecycleConfig;
/** @internal */
StudioLifecycleConfig.__pulumiType = 'aws-native:sagemaker:StudioLifecycleConfig';
