// @pulumi/aws-native/ec2/ipamResourceDiscoveryAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IpamResourceDiscoveryAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Schema of AWS::EC2::IPAMResourceDiscoveryAssociation Type
 */
class IpamResourceDiscoveryAssociation extends pulumi.CustomResource {
    /**
     * Get an existing IpamResourceDiscoveryAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IpamResourceDiscoveryAssociation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IpamResourceDiscoveryAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IpamResourceDiscoveryAssociation.__pulumiType;
    }
    /**
     * Create a IpamResourceDiscoveryAssociation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.ipamId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ipamId'");
            }
            if ((!args || args.ipamResourceDiscoveryId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ipamResourceDiscoveryId'");
            }
            resourceInputs["ipamId"] = args ? args.ipamId : undefined;
            resourceInputs["ipamResourceDiscoveryId"] = args ? args.ipamResourceDiscoveryId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["ipamArn"] = undefined /*out*/;
            resourceInputs["ipamRegion"] = undefined /*out*/;
            resourceInputs["ipamResourceDiscoveryAssociationArn"] = undefined /*out*/;
            resourceInputs["ipamResourceDiscoveryAssociationId"] = undefined /*out*/;
            resourceInputs["isDefault"] = undefined /*out*/;
            resourceInputs["ownerId"] = undefined /*out*/;
            resourceInputs["resourceDiscoveryStatus"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
        }
        else {
            resourceInputs["ipamArn"] = undefined /*out*/;
            resourceInputs["ipamId"] = undefined /*out*/;
            resourceInputs["ipamRegion"] = undefined /*out*/;
            resourceInputs["ipamResourceDiscoveryAssociationArn"] = undefined /*out*/;
            resourceInputs["ipamResourceDiscoveryAssociationId"] = undefined /*out*/;
            resourceInputs["ipamResourceDiscoveryId"] = undefined /*out*/;
            resourceInputs["isDefault"] = undefined /*out*/;
            resourceInputs["ownerId"] = undefined /*out*/;
            resourceInputs["resourceDiscoveryStatus"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["ipamId", "ipamResourceDiscoveryId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(IpamResourceDiscoveryAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IpamResourceDiscoveryAssociation = IpamResourceDiscoveryAssociation;
/** @internal */
IpamResourceDiscoveryAssociation.__pulumiType = 'aws-native:ec2:IpamResourceDiscoveryAssociation';
