// @pulumi/aws-native/ec2/localGatewayRouteTable.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalGatewayRouteTable = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes a route table for a local gateway.
 */
class LocalGatewayRouteTable extends pulumi.CustomResource {
    /**
     * Get an existing LocalGatewayRouteTable resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LocalGatewayRouteTable(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocalGatewayRouteTable.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocalGatewayRouteTable.__pulumiType;
    }
    /**
     * Create a LocalGatewayRouteTable resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.localGatewayId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'localGatewayId'");
            }
            resourceInputs["localGatewayId"] = args ? args.localGatewayId : undefined;
            resourceInputs["mode"] = args ? args.mode : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["localGatewayRouteTableArn"] = undefined /*out*/;
            resourceInputs["localGatewayRouteTableId"] = undefined /*out*/;
            resourceInputs["outpostArn"] = undefined /*out*/;
            resourceInputs["ownerId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
        }
        else {
            resourceInputs["localGatewayId"] = undefined /*out*/;
            resourceInputs["localGatewayRouteTableArn"] = undefined /*out*/;
            resourceInputs["localGatewayRouteTableId"] = undefined /*out*/;
            resourceInputs["mode"] = undefined /*out*/;
            resourceInputs["outpostArn"] = undefined /*out*/;
            resourceInputs["ownerId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["localGatewayId", "mode"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(LocalGatewayRouteTable.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LocalGatewayRouteTable = LocalGatewayRouteTable;
/** @internal */
LocalGatewayRouteTable.__pulumiType = 'aws-native:ec2:LocalGatewayRouteTable';
