// @pulumi/aws-native/wisdom/aiGuardrailVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiGuardrailVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Wisdom::AIGuardrailVersion Resource Type
 */
class AiGuardrailVersion extends pulumi.CustomResource {
    /**
     * Get an existing AiGuardrailVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AiGuardrailVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AiGuardrailVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AiGuardrailVersion.__pulumiType;
    }
    /**
     * Create a AiGuardrailVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.aiGuardrailId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'aiGuardrailId'");
            }
            if ((!args || args.assistantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'assistantId'");
            }
            resourceInputs["aiGuardrailId"] = args ? args.aiGuardrailId : undefined;
            resourceInputs["assistantId"] = args ? args.assistantId : undefined;
            resourceInputs["modifiedTimeSeconds"] = args ? args.modifiedTimeSeconds : undefined;
            resourceInputs["aiGuardrailArn"] = undefined /*out*/;
            resourceInputs["aiGuardrailVersionId"] = undefined /*out*/;
            resourceInputs["assistantArn"] = undefined /*out*/;
            resourceInputs["versionNumber"] = undefined /*out*/;
        }
        else {
            resourceInputs["aiGuardrailArn"] = undefined /*out*/;
            resourceInputs["aiGuardrailId"] = undefined /*out*/;
            resourceInputs["aiGuardrailVersionId"] = undefined /*out*/;
            resourceInputs["assistantArn"] = undefined /*out*/;
            resourceInputs["assistantId"] = undefined /*out*/;
            resourceInputs["modifiedTimeSeconds"] = undefined /*out*/;
            resourceInputs["versionNumber"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["aiGuardrailId", "assistantId", "modifiedTimeSeconds"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(AiGuardrailVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AiGuardrailVersion = AiGuardrailVersion;
/** @internal */
AiGuardrailVersion.__pulumiType = 'aws-native:wisdom:AiGuardrailVersion';
