// @pulumi/aws-native/wisdom/aiPrompt.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiPrompt = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Wisdom::AIPrompt Resource Type
 */
class AiPrompt extends pulumi.CustomResource {
    /**
     * Get an existing AiPrompt resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AiPrompt(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AiPrompt.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AiPrompt.__pulumiType;
    }
    /**
     * Create a AiPrompt resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiFormat === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiFormat'");
            }
            if ((!args || args.modelId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelId'");
            }
            if ((!args || args.templateConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'templateConfiguration'");
            }
            if ((!args || args.templateType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'templateType'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["apiFormat"] = args ? args.apiFormat : undefined;
            resourceInputs["assistantId"] = args ? args.assistantId : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["modelId"] = args ? args.modelId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["templateConfiguration"] = args ? args.templateConfiguration : undefined;
            resourceInputs["templateType"] = args ? args.templateType : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["aiPromptArn"] = undefined /*out*/;
            resourceInputs["aiPromptId"] = undefined /*out*/;
            resourceInputs["assistantArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["aiPromptArn"] = undefined /*out*/;
            resourceInputs["aiPromptId"] = undefined /*out*/;
            resourceInputs["apiFormat"] = undefined /*out*/;
            resourceInputs["assistantArn"] = undefined /*out*/;
            resourceInputs["assistantId"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["modelId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["templateConfiguration"] = undefined /*out*/;
            resourceInputs["templateType"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["apiFormat", "assistantId", "modelId", "name", "tags.*", "templateType", "type"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(AiPrompt.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AiPrompt = AiPrompt;
/** @internal */
AiPrompt.__pulumiType = 'aws-native:wisdom:AiPrompt';
