// @pulumi/aws-native/wisdom/messageTemplateVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageTemplateVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A version for the specified customer-managed message template within the specified knowledge base.
 */
class MessageTemplateVersion extends pulumi.CustomResource {
    /**
     * Get an existing MessageTemplateVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MessageTemplateVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MessageTemplateVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MessageTemplateVersion.__pulumiType;
    }
    /**
     * Create a MessageTemplateVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.messageTemplateArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'messageTemplateArn'");
            }
            resourceInputs["messageTemplateArn"] = args ? args.messageTemplateArn : undefined;
            resourceInputs["messageTemplateContentSha256"] = args ? args.messageTemplateContentSha256 : undefined;
            resourceInputs["messageTemplateVersionArn"] = undefined /*out*/;
            resourceInputs["messageTemplateVersionNumber"] = undefined /*out*/;
        }
        else {
            resourceInputs["messageTemplateArn"] = undefined /*out*/;
            resourceInputs["messageTemplateContentSha256"] = undefined /*out*/;
            resourceInputs["messageTemplateVersionArn"] = undefined /*out*/;
            resourceInputs["messageTemplateVersionNumber"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["messageTemplateArn"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(MessageTemplateVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MessageTemplateVersion = MessageTemplateVersion;
/** @internal */
MessageTemplateVersion.__pulumiType = 'aws-native:wisdom:MessageTemplateVersion';
