// @pulumi/aws-native/types/enums/ecr/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryImageTagMutability = exports.RepositoryEncryptionType = exports.RepositoryCreationTemplateImageTagMutability = exports.RepositoryCreationTemplateEncryptionType = exports.RepositoryCreationTemplateAppliedForItem = exports.ReplicationConfigurationFilterType = void 0;
exports.ReplicationConfigurationFilterType = {
    PrefixMatch: "PREFIX_MATCH",
};
exports.RepositoryCreationTemplateAppliedForItem = {
    Replication: "REPLICATION",
    PullThroughCache: "PULL_THROUGH_CACHE",
};
exports.RepositoryCreationTemplateEncryptionType = {
    Aes256: "AES256",
    Kms: "KMS",
    KmsDsse: "KMS_DSSE",
};
exports.RepositoryCreationTemplateImageTagMutability = {
    Mutable: "MUTABLE",
    Immutable: "IMMUTABLE",
};
exports.RepositoryEncryptionType = {
    Aes256: "AES256",
    Kms: "KMS",
    KmsDsse: "KMS_DSSE",
};
exports.RepositoryImageTagMutability = {
    Mutable: "MUTABLE",
    Immutable: "IMMUTABLE",
};
