// @pulumi/aws-native/types/enums/amplifyuibuilder/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormStorageAccessLevel = exports.FormLabelDecorator = exports.FormFixedPosition = exports.FormDataSourceType = exports.FormButtonsPosition = exports.FormActionType = exports.ComponentSortDirection = void 0;
exports.ComponentSortDirection = {
    Asc: "ASC",
    Desc: "DESC",
};
exports.FormActionType = {
    Create: "create",
    Update: "update",
};
exports.FormButtonsPosition = {
    Top: "top",
    Bottom: "bottom",
    TopAndBottom: "top_and_bottom",
};
exports.FormDataSourceType = {
    DataStore: "DataStore",
    Custom: "Custom",
};
exports.FormFixedPosition = {
    First: "first",
};
exports.FormLabelDecorator = {
    Required: "required",
    Optional: "optional",
    None: "none",
};
exports.FormStorageAccessLevel = {
    Public: "public",
    Protected: "protected",
    Private: "private",
};
