// @pulumi/aws-native/rds/dbProxyTargetGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbProxyTargetGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::RDS::DBProxyTargetGroup
 */
class DbProxyTargetGroup extends pulumi.CustomResource {
    /**
     * Get an existing DbProxyTargetGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DbProxyTargetGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DbProxyTargetGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DbProxyTargetGroup.__pulumiType;
    }
    /**
     * Create a DbProxyTargetGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dbProxyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dbProxyName'");
            }
            if ((!args || args.targetGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetGroupName'");
            }
            resourceInputs["connectionPoolConfigurationInfo"] = args ? args.connectionPoolConfigurationInfo : undefined;
            resourceInputs["dbClusterIdentifiers"] = args ? args.dbClusterIdentifiers : undefined;
            resourceInputs["dbInstanceIdentifiers"] = args ? args.dbInstanceIdentifiers : undefined;
            resourceInputs["dbProxyName"] = args ? args.dbProxyName : undefined;
            resourceInputs["targetGroupName"] = args ? args.targetGroupName : undefined;
            resourceInputs["targetGroupArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["connectionPoolConfigurationInfo"] = undefined /*out*/;
            resourceInputs["dbClusterIdentifiers"] = undefined /*out*/;
            resourceInputs["dbInstanceIdentifiers"] = undefined /*out*/;
            resourceInputs["dbProxyName"] = undefined /*out*/;
            resourceInputs["targetGroupArn"] = undefined /*out*/;
            resourceInputs["targetGroupName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["dbProxyName", "targetGroupName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(DbProxyTargetGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DbProxyTargetGroup = DbProxyTargetGroup;
/** @internal */
DbProxyTargetGroup.__pulumiType = 'aws-native:rds:DbProxyTargetGroup';
