// @pulumi/aws-native/rds/dbShardGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbShardGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::RDS::DBShardGroup resource creates an Amazon Aurora Limitless DB Shard Group.
 */
class DbShardGroup extends pulumi.CustomResource {
    /**
     * Get an existing DbShardGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DbShardGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DbShardGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DbShardGroup.__pulumiType;
    }
    /**
     * Create a DbShardGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dbClusterIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dbClusterIdentifier'");
            }
            if ((!args || args.maxAcu === undefined) && !opts.urn) {
                throw new Error("Missing required property 'maxAcu'");
            }
            resourceInputs["computeRedundancy"] = args ? args.computeRedundancy : undefined;
            resourceInputs["dbClusterIdentifier"] = args ? args.dbClusterIdentifier : undefined;
            resourceInputs["dbShardGroupIdentifier"] = args ? args.dbShardGroupIdentifier : undefined;
            resourceInputs["maxAcu"] = args ? args.maxAcu : undefined;
            resourceInputs["minAcu"] = args ? args.minAcu : undefined;
            resourceInputs["publiclyAccessible"] = args ? args.publiclyAccessible : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["dbShardGroupResourceId"] = undefined /*out*/;
            resourceInputs["endpoint"] = undefined /*out*/;
        }
        else {
            resourceInputs["computeRedundancy"] = undefined /*out*/;
            resourceInputs["dbClusterIdentifier"] = undefined /*out*/;
            resourceInputs["dbShardGroupIdentifier"] = undefined /*out*/;
            resourceInputs["dbShardGroupResourceId"] = undefined /*out*/;
            resourceInputs["endpoint"] = undefined /*out*/;
            resourceInputs["maxAcu"] = undefined /*out*/;
            resourceInputs["minAcu"] = undefined /*out*/;
            resourceInputs["publiclyAccessible"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["dbClusterIdentifier", "dbShardGroupIdentifier", "publiclyAccessible"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(DbShardGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DbShardGroup = DbShardGroup;
/** @internal */
DbShardGroup.__pulumiType = 'aws-native:rds:DbShardGroup';
