// @pulumi/aws-native/s3/accessGrant.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessGrant = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::S3::AccessGrant resource is an Amazon S3 resource type representing permissions to a specific S3 bucket or prefix hosted in an S3 Access Grants instance.
 */
class AccessGrant extends pulumi.CustomResource {
    /**
     * Get an existing AccessGrant resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AccessGrant(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AccessGrant.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccessGrant.__pulumiType;
    }
    /**
     * Create a AccessGrant resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accessGrantsLocationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accessGrantsLocationId'");
            }
            if ((!args || args.grantee === undefined) && !opts.urn) {
                throw new Error("Missing required property 'grantee'");
            }
            if ((!args || args.permission === undefined) && !opts.urn) {
                throw new Error("Missing required property 'permission'");
            }
            resourceInputs["accessGrantsLocationConfiguration"] = args ? args.accessGrantsLocationConfiguration : undefined;
            resourceInputs["accessGrantsLocationId"] = args ? args.accessGrantsLocationId : undefined;
            resourceInputs["applicationArn"] = args ? args.applicationArn : undefined;
            resourceInputs["grantee"] = args ? args.grantee : undefined;
            resourceInputs["permission"] = args ? args.permission : undefined;
            resourceInputs["s3PrefixType"] = args ? args.s3PrefixType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["accessGrantArn"] = undefined /*out*/;
            resourceInputs["accessGrantId"] = undefined /*out*/;
            resourceInputs["grantScope"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessGrantArn"] = undefined /*out*/;
            resourceInputs["accessGrantId"] = undefined /*out*/;
            resourceInputs["accessGrantsLocationConfiguration"] = undefined /*out*/;
            resourceInputs["accessGrantsLocationId"] = undefined /*out*/;
            resourceInputs["applicationArn"] = undefined /*out*/;
            resourceInputs["grantScope"] = undefined /*out*/;
            resourceInputs["grantee"] = undefined /*out*/;
            resourceInputs["permission"] = undefined /*out*/;
            resourceInputs["s3PrefixType"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["s3PrefixType", "tags[*]"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(AccessGrant.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AccessGrant = AccessGrant;
/** @internal */
AccessGrant.__pulumiType = 'aws-native:s3:AccessGrant';
