// @pulumi/aws-native/datasync/locationEfs.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationEfs = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::DataSync::LocationEFS.
 */
class LocationEfs extends pulumi.CustomResource {
    /**
     * Get an existing LocationEfs resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LocationEfs(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocationEfs.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocationEfs.__pulumiType;
    }
    /**
     * Create a LocationEfs resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.ec2Config === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ec2Config'");
            }
            resourceInputs["accessPointArn"] = args ? args.accessPointArn : undefined;
            resourceInputs["ec2Config"] = args ? args.ec2Config : undefined;
            resourceInputs["efsFilesystemArn"] = args ? args.efsFilesystemArn : undefined;
            resourceInputs["fileSystemAccessRoleArn"] = args ? args.fileSystemAccessRoleArn : undefined;
            resourceInputs["inTransitEncryption"] = args ? args.inTransitEncryption : undefined;
            resourceInputs["subdirectory"] = args ? args.subdirectory : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["locationArn"] = undefined /*out*/;
            resourceInputs["locationUri"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessPointArn"] = undefined /*out*/;
            resourceInputs["ec2Config"] = undefined /*out*/;
            resourceInputs["efsFilesystemArn"] = undefined /*out*/;
            resourceInputs["fileSystemAccessRoleArn"] = undefined /*out*/;
            resourceInputs["inTransitEncryption"] = undefined /*out*/;
            resourceInputs["locationArn"] = undefined /*out*/;
            resourceInputs["locationUri"] = undefined /*out*/;
            resourceInputs["subdirectory"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["accessPointArn", "ec2Config", "efsFilesystemArn", "fileSystemAccessRoleArn", "inTransitEncryption", "subdirectory"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(LocationEfs.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LocationEfs = LocationEfs;
/** @internal */
LocationEfs.__pulumiType = 'aws-native:datasync:LocationEfs';
