// @pulumi/aws-native/datasync/locationAzureBlob.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationAzureBlob = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::DataSync::LocationAzureBlob.
 */
class LocationAzureBlob extends pulumi.CustomResource {
    /**
     * Get an existing LocationAzureBlob resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LocationAzureBlob(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocationAzureBlob.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocationAzureBlob.__pulumiType;
    }
    /**
     * Create a LocationAzureBlob resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.agentArns === undefined) && !opts.urn) {
                throw new Error("Missing required property 'agentArns'");
            }
            if ((!args || args.azureBlobAuthenticationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureBlobAuthenticationType'");
            }
            resourceInputs["agentArns"] = args ? args.agentArns : undefined;
            resourceInputs["azureAccessTier"] = args ? args.azureAccessTier : undefined;
            resourceInputs["azureBlobAuthenticationType"] = args ? args.azureBlobAuthenticationType : undefined;
            resourceInputs["azureBlobContainerUrl"] = args ? args.azureBlobContainerUrl : undefined;
            resourceInputs["azureBlobSasConfiguration"] = args ? args.azureBlobSasConfiguration : undefined;
            resourceInputs["azureBlobType"] = args ? args.azureBlobType : undefined;
            resourceInputs["subdirectory"] = args ? args.subdirectory : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["locationArn"] = undefined /*out*/;
            resourceInputs["locationUri"] = undefined /*out*/;
        }
        else {
            resourceInputs["agentArns"] = undefined /*out*/;
            resourceInputs["azureAccessTier"] = undefined /*out*/;
            resourceInputs["azureBlobAuthenticationType"] = undefined /*out*/;
            resourceInputs["azureBlobContainerUrl"] = undefined /*out*/;
            resourceInputs["azureBlobSasConfiguration"] = undefined /*out*/;
            resourceInputs["azureBlobType"] = undefined /*out*/;
            resourceInputs["locationArn"] = undefined /*out*/;
            resourceInputs["locationUri"] = undefined /*out*/;
            resourceInputs["subdirectory"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["azureBlobContainerUrl"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(LocationAzureBlob.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LocationAzureBlob = LocationAzureBlob;
/** @internal */
LocationAzureBlob.__pulumiType = 'aws-native:datasync:LocationAzureBlob';
