// @pulumi/aws-native/datasync/storageSystem.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageSystem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::DataSync::StorageSystem.
 */
class StorageSystem extends pulumi.CustomResource {
    /**
     * Get an existing StorageSystem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StorageSystem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StorageSystem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageSystem.__pulumiType;
    }
    /**
     * Create a StorageSystem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.agentArns === undefined) && !opts.urn) {
                throw new Error("Missing required property 'agentArns'");
            }
            if ((!args || args.serverConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverConfiguration'");
            }
            if ((!args || args.systemType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'systemType'");
            }
            resourceInputs["agentArns"] = args ? args.agentArns : undefined;
            resourceInputs["cloudWatchLogGroupArn"] = args ? args.cloudWatchLogGroupArn : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["serverConfiguration"] = args ? args.serverConfiguration : undefined;
            resourceInputs["serverCredentials"] = args ? args.serverCredentials : undefined;
            resourceInputs["systemType"] = args ? args.systemType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["connectivityStatus"] = undefined /*out*/;
            resourceInputs["secretsManagerArn"] = undefined /*out*/;
            resourceInputs["storageSystemArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["agentArns"] = undefined /*out*/;
            resourceInputs["cloudWatchLogGroupArn"] = undefined /*out*/;
            resourceInputs["connectivityStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["secretsManagerArn"] = undefined /*out*/;
            resourceInputs["serverConfiguration"] = undefined /*out*/;
            resourceInputs["serverCredentials"] = undefined /*out*/;
            resourceInputs["storageSystemArn"] = undefined /*out*/;
            resourceInputs["systemType"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StorageSystem.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StorageSystem = StorageSystem;
/** @internal */
StorageSystem.__pulumiType = 'aws-native:datasync:StorageSystem';
