// @pulumi/aws-native/deadline/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageProfile = exports.QueueFleetAssociation = exports.QueueEnvironment = exports.Queue = exports.Monitor = exports.MeteredProduct = exports.LicenseEndpoint = exports.getStorageProfileOutput = exports.getStorageProfile = exports.getQueueEnvironmentOutput = exports.getQueueEnvironment = exports.getQueueOutput = exports.getQueue = exports.getMonitorOutput = exports.getMonitor = exports.getMeteredProductOutput = exports.getMeteredProduct = exports.getLicenseEndpointOutput = exports.getLicenseEndpoint = exports.getFleetOutput = exports.getFleet = exports.getFarmOutput = exports.getFarm = exports.Fleet = exports.Farm = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Farm = null;
utilities.lazyLoad(exports, ["Farm"], () => require("./farm"));
exports.Fleet = null;
utilities.lazyLoad(exports, ["Fleet"], () => require("./fleet"));
exports.getFarm = null;
exports.getFarmOutput = null;
utilities.lazyLoad(exports, ["getFarm", "getFarmOutput"], () => require("./getFarm"));
exports.getFleet = null;
exports.getFleetOutput = null;
utilities.lazyLoad(exports, ["getFleet", "getFleetOutput"], () => require("./getFleet"));
exports.getLicenseEndpoint = null;
exports.getLicenseEndpointOutput = null;
utilities.lazyLoad(exports, ["getLicenseEndpoint", "getLicenseEndpointOutput"], () => require("./getLicenseEndpoint"));
exports.getMeteredProduct = null;
exports.getMeteredProductOutput = null;
utilities.lazyLoad(exports, ["getMeteredProduct", "getMeteredProductOutput"], () => require("./getMeteredProduct"));
exports.getMonitor = null;
exports.getMonitorOutput = null;
utilities.lazyLoad(exports, ["getMonitor", "getMonitorOutput"], () => require("./getMonitor"));
exports.getQueue = null;
exports.getQueueOutput = null;
utilities.lazyLoad(exports, ["getQueue", "getQueueOutput"], () => require("./getQueue"));
exports.getQueueEnvironment = null;
exports.getQueueEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getQueueEnvironment", "getQueueEnvironmentOutput"], () => require("./getQueueEnvironment"));
exports.getStorageProfile = null;
exports.getStorageProfileOutput = null;
utilities.lazyLoad(exports, ["getStorageProfile", "getStorageProfileOutput"], () => require("./getStorageProfile"));
exports.LicenseEndpoint = null;
utilities.lazyLoad(exports, ["LicenseEndpoint"], () => require("./licenseEndpoint"));
exports.MeteredProduct = null;
utilities.lazyLoad(exports, ["MeteredProduct"], () => require("./meteredProduct"));
exports.Monitor = null;
utilities.lazyLoad(exports, ["Monitor"], () => require("./monitor"));
exports.Queue = null;
utilities.lazyLoad(exports, ["Queue"], () => require("./queue"));
exports.QueueEnvironment = null;
utilities.lazyLoad(exports, ["QueueEnvironment"], () => require("./queueEnvironment"));
exports.QueueFleetAssociation = null;
utilities.lazyLoad(exports, ["QueueFleetAssociation"], () => require("./queueFleetAssociation"));
exports.StorageProfile = null;
utilities.lazyLoad(exports, ["StorageProfile"], () => require("./storageProfile"));
// Export enums:
__exportStar(require("../types/enums/deadline"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:deadline:Farm":
                return new exports.Farm(name, undefined, { urn });
            case "aws-native:deadline:Fleet":
                return new exports.Fleet(name, undefined, { urn });
            case "aws-native:deadline:LicenseEndpoint":
                return new exports.LicenseEndpoint(name, undefined, { urn });
            case "aws-native:deadline:MeteredProduct":
                return new exports.MeteredProduct(name, undefined, { urn });
            case "aws-native:deadline:Monitor":
                return new exports.Monitor(name, undefined, { urn });
            case "aws-native:deadline:Queue":
                return new exports.Queue(name, undefined, { urn });
            case "aws-native:deadline:QueueEnvironment":
                return new exports.QueueEnvironment(name, undefined, { urn });
            case "aws-native:deadline:QueueFleetAssociation":
                return new exports.QueueFleetAssociation(name, undefined, { urn });
            case "aws-native:deadline:StorageProfile":
                return new exports.StorageProfile(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "deadline", _module);
