// @pulumi/aws-native/deadline/meteredProduct.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeteredProduct = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Deadline::MeteredProduct Resource Type
 */
class MeteredProduct extends pulumi.CustomResource {
    /**
     * Get an existing MeteredProduct resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MeteredProduct(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MeteredProduct.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MeteredProduct.__pulumiType;
    }
    /**
     * Create a MeteredProduct resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["licenseEndpointId"] = args ? args.licenseEndpointId : undefined;
            resourceInputs["productId"] = args ? args.productId : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["family"] = undefined /*out*/;
            resourceInputs["port"] = undefined /*out*/;
            resourceInputs["vendor"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["family"] = undefined /*out*/;
            resourceInputs["licenseEndpointId"] = undefined /*out*/;
            resourceInputs["port"] = undefined /*out*/;
            resourceInputs["productId"] = undefined /*out*/;
            resourceInputs["vendor"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["licenseEndpointId", "productId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(MeteredProduct.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MeteredProduct = MeteredProduct;
/** @internal */
MeteredProduct.__pulumiType = 'aws-native:deadline:MeteredProduct';
