// @pulumi/aws-native/deadline/queueEnvironment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueEnvironment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Deadline::QueueEnvironment Resource Type
 */
class QueueEnvironment extends pulumi.CustomResource {
    /**
     * Get an existing QueueEnvironment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new QueueEnvironment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of QueueEnvironment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === QueueEnvironment.__pulumiType;
    }
    /**
     * Create a QueueEnvironment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.farmId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'farmId'");
            }
            if ((!args || args.priority === undefined) && !opts.urn) {
                throw new Error("Missing required property 'priority'");
            }
            if ((!args || args.queueId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queueId'");
            }
            if ((!args || args.template === undefined) && !opts.urn) {
                throw new Error("Missing required property 'template'");
            }
            if ((!args || args.templateType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'templateType'");
            }
            resourceInputs["farmId"] = args ? args.farmId : undefined;
            resourceInputs["priority"] = args ? args.priority : undefined;
            resourceInputs["queueId"] = args ? args.queueId : undefined;
            resourceInputs["template"] = args ? args.template : undefined;
            resourceInputs["templateType"] = args ? args.templateType : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["queueEnvironmentId"] = undefined /*out*/;
        }
        else {
            resourceInputs["farmId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["queueEnvironmentId"] = undefined /*out*/;
            resourceInputs["queueId"] = undefined /*out*/;
            resourceInputs["template"] = undefined /*out*/;
            resourceInputs["templateType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["farmId", "queueId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(QueueEnvironment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.QueueEnvironment = QueueEnvironment;
/** @internal */
QueueEnvironment.__pulumiType = 'aws-native:deadline:QueueEnvironment';
