// @pulumi/aws-native/deadline/storageProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Deadline::StorageProfile Resource Type
 */
class StorageProfile extends pulumi.CustomResource {
    /**
     * Get an existing StorageProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StorageProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StorageProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageProfile.__pulumiType;
    }
    /**
     * Create a StorageProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.farmId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'farmId'");
            }
            if ((!args || args.osFamily === undefined) && !opts.urn) {
                throw new Error("Missing required property 'osFamily'");
            }
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["farmId"] = args ? args.farmId : undefined;
            resourceInputs["fileSystemLocations"] = args ? args.fileSystemLocations : undefined;
            resourceInputs["osFamily"] = args ? args.osFamily : undefined;
            resourceInputs["storageProfileId"] = undefined /*out*/;
        }
        else {
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["farmId"] = undefined /*out*/;
            resourceInputs["fileSystemLocations"] = undefined /*out*/;
            resourceInputs["osFamily"] = undefined /*out*/;
            resourceInputs["storageProfileId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["farmId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(StorageProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StorageProfile = StorageProfile;
/** @internal */
StorageProfile.__pulumiType = 'aws-native:deadline:StorageProfile';
