// @pulumi/aws-native/types/enums/sso/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstanceStatus = exports.AssignmentTargetType = exports.AssignmentPrincipalType = exports.ApplicationStatus = exports.ApplicationSignInOptionsOrigin = exports.ApplicationPortalOptionsConfigurationVisibility = exports.ApplicationAssignmentPrincipalType = void 0;
exports.ApplicationAssignmentPrincipalType = {
    User: "USER",
    Group: "GROUP",
};
exports.ApplicationPortalOptionsConfigurationVisibility = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.ApplicationSignInOptionsOrigin = {
    IdentityCenter: "IDENTITY_CENTER",
    Application: "APPLICATION",
};
exports.ApplicationStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.AssignmentPrincipalType = {
    User: "USER",
    Group: "GROUP",
};
exports.AssignmentTargetType = {
    AwsAccount: "AWS_ACCOUNT",
};
exports.InstanceStatus = {
    CreateInProgress: "CREATE_IN_PROGRESS",
    DeleteInProgress: "DELETE_IN_PROGRESS",
    Active: "ACTIVE",
};
