// @pulumi/aws-native/chatbot/customAction.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Chatbot::CustomAction Resource Type
 */
class CustomAction extends pulumi.CustomResource {
    /**
     * Get an existing CustomAction resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CustomAction(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomAction.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomAction.__pulumiType;
    }
    /**
     * Create a CustomAction resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.definition === undefined) && !opts.urn) {
                throw new Error("Missing required property 'definition'");
            }
            resourceInputs["actionName"] = args ? args.actionName : undefined;
            resourceInputs["aliasName"] = args ? args.aliasName : undefined;
            resourceInputs["attachments"] = args ? args.attachments : undefined;
            resourceInputs["definition"] = args ? args.definition : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["customActionArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["actionName"] = undefined /*out*/;
            resourceInputs["aliasName"] = undefined /*out*/;
            resourceInputs["attachments"] = undefined /*out*/;
            resourceInputs["customActionArn"] = undefined /*out*/;
            resourceInputs["definition"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["actionName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(CustomAction.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CustomAction = CustomAction;
/** @internal */
CustomAction.__pulumiType = 'aws-native:chatbot:CustomAction';
