// @pulumi/aws-native/chatbot/microsoftTeamsChannelConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MicrosoftTeamsChannelConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::Chatbot::MicrosoftTeamsChannelConfiguration.
 */
class MicrosoftTeamsChannelConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing MicrosoftTeamsChannelConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MicrosoftTeamsChannelConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MicrosoftTeamsChannelConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MicrosoftTeamsChannelConfiguration.__pulumiType;
    }
    /**
     * Create a MicrosoftTeamsChannelConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.iamRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iamRoleArn'");
            }
            if ((!args || args.teamId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'teamId'");
            }
            if ((!args || args.teamsChannelId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'teamsChannelId'");
            }
            if ((!args || args.teamsTenantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'teamsTenantId'");
            }
            resourceInputs["configurationName"] = args ? args.configurationName : undefined;
            resourceInputs["customizationResourceArns"] = args ? args.customizationResourceArns : undefined;
            resourceInputs["guardrailPolicies"] = args ? args.guardrailPolicies : undefined;
            resourceInputs["iamRoleArn"] = args ? args.iamRoleArn : undefined;
            resourceInputs["loggingLevel"] = args ? args.loggingLevel : undefined;
            resourceInputs["snsTopicArns"] = args ? args.snsTopicArns : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["teamId"] = args ? args.teamId : undefined;
            resourceInputs["teamsChannelId"] = args ? args.teamsChannelId : undefined;
            resourceInputs["teamsTenantId"] = args ? args.teamsTenantId : undefined;
            resourceInputs["userRoleRequired"] = args ? args.userRoleRequired : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["configurationName"] = undefined /*out*/;
            resourceInputs["customizationResourceArns"] = undefined /*out*/;
            resourceInputs["guardrailPolicies"] = undefined /*out*/;
            resourceInputs["iamRoleArn"] = undefined /*out*/;
            resourceInputs["loggingLevel"] = undefined /*out*/;
            resourceInputs["snsTopicArns"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["teamId"] = undefined /*out*/;
            resourceInputs["teamsChannelId"] = undefined /*out*/;
            resourceInputs["teamsTenantId"] = undefined /*out*/;
            resourceInputs["userRoleRequired"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["configurationName", "teamId", "teamsTenantId"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(MicrosoftTeamsChannelConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MicrosoftTeamsChannelConfiguration = MicrosoftTeamsChannelConfiguration;
/** @internal */
MicrosoftTeamsChannelConfiguration.__pulumiType = 'aws-native:chatbot:MicrosoftTeamsChannelConfiguration';
