// @pulumi/aws-native/ivs/channel.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Channel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::IVS::Channel
 *
 * ## Example Usage
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const channel = new aws_native.ivs.Channel("channel", {
 *     name: "MyChannel",
 *     tags: [{
 *         key: "MyKey",
 *         value: "MyValue",
 *     }],
 *     insecureIngest: true,
 * });
 * const streamKey = new aws_native.ivs.StreamKey("streamKey", {
 *     channelArn: channel.id,
 *     tags: [{
 *         key: "MyKey",
 *         value: "MyValue",
 *     }],
 * });
 * export const channelArn = channel.id;
 * export const channelIngestEndpoint = channel.ingestEndpoint;
 * export const channelPlaybackUrl = channel.playbackUrl;
 * export const streamKeyArn = streamKey.id;
 *
 * ```
 * ### Example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws_native from "@pulumi/aws-native";
 *
 * const channel = new aws_native.ivs.Channel("channel", {
 *     name: "MyChannel",
 *     tags: [{
 *         key: "MyKey",
 *         value: "MyValue",
 *     }],
 *     insecureIngest: true,
 * });
 * const streamKey = new aws_native.ivs.StreamKey("streamKey", {
 *     channelArn: channel.id,
 *     tags: [{
 *         key: "MyKey",
 *         value: "MyValue",
 *     }],
 * });
 * export const channelArn = channel.id;
 * export const channelIngestEndpoint = channel.ingestEndpoint;
 * export const channelPlaybackUrl = channel.playbackUrl;
 * export const streamKeyArn = streamKey.id;
 *
 * ```
 */
class Channel extends pulumi.CustomResource {
    /**
     * Get an existing Channel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Channel(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Channel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Channel.__pulumiType;
    }
    /**
     * Create a Channel resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["authorized"] = args ? args.authorized : undefined;
            resourceInputs["insecureIngest"] = args ? args.insecureIngest : undefined;
            resourceInputs["latencyMode"] = args ? args.latencyMode : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["preset"] = args ? args.preset : undefined;
            resourceInputs["recordingConfigurationArn"] = args ? args.recordingConfigurationArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["ingestEndpoint"] = undefined /*out*/;
            resourceInputs["playbackUrl"] = undefined /*out*/;
        }
        else {
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["authorized"] = undefined /*out*/;
            resourceInputs["ingestEndpoint"] = undefined /*out*/;
            resourceInputs["insecureIngest"] = undefined /*out*/;
            resourceInputs["latencyMode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["playbackUrl"] = undefined /*out*/;
            resourceInputs["preset"] = undefined /*out*/;
            resourceInputs["recordingConfigurationArn"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Channel.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Channel = Channel;
/** @internal */
Channel.__pulumiType = 'aws-native:ivs:Channel';
