// @pulumi/aws-native/types/enums/codepipeline/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PipelineType = exports.PipelineTriggerDeclarationProviderType = exports.PipelineFailureConditionsRetryConfigurationPropertiesRetryMode = exports.PipelineFailureConditionsResult = exports.PipelineExecutionMode = exports.PipelineBlockerDeclarationType = exports.PipelineArtifactStoreType = exports.PipelineActionTypeIdCategory = void 0;
exports.PipelineActionTypeIdCategory = {
    Source: "Source",
    Build: "Build",
    Test: "Test",
    Deploy: "Deploy",
    Invoke: "Invoke",
    Approval: "Approval",
    Compute: "Compute",
};
exports.PipelineArtifactStoreType = {
    S3: "S3",
};
exports.PipelineBlockerDeclarationType = {
    Schedule: "Schedule",
};
exports.PipelineExecutionMode = {
    Queued: "QUEUED",
    Superseded: "SUPERSEDED",
    Parallel: "PARALLEL",
};
exports.PipelineFailureConditionsResult = {
    Rollback: "ROLLBACK",
    Retry: "RETRY",
};
exports.PipelineFailureConditionsRetryConfigurationPropertiesRetryMode = {
    AllActions: "ALL_ACTIONS",
    FailedActions: "FAILED_ACTIONS",
};
exports.PipelineTriggerDeclarationProviderType = {
    CodeStarSourceConnection: "CodeStarSourceConnection",
};
exports.PipelineType = {
    V1: "V1",
    V2: "V2",
};
