// @pulumi/aws-native/types/enums/cloudtrail/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrailEventSelectorReadWriteType = exports.DashboardType = exports.DashboardStatus = exports.DashboardRefreshScheduleStatus = exports.DashboardRefreshScheduleFrequencyPropertiesUnit = exports.ChannelDestinationType = void 0;
exports.ChannelDestinationType = {
    EventDataStore: "EVENT_DATA_STORE",
};
exports.DashboardRefreshScheduleFrequencyPropertiesUnit = {
    Hours: "HOURS",
    Days: "DAYS",
};
exports.DashboardRefreshScheduleStatus = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.DashboardStatus = {
    Creating: "CREATING",
    Created: "CREATED",
    Updating: "UPDATING",
    Updated: "UPDATED",
    Deleting: "DELETING",
};
exports.DashboardType = {
    Managed: "MANAGED",
    Custom: "CUSTOM",
};
exports.TrailEventSelectorReadWriteType = {
    All: "All",
    ReadOnly: "ReadOnly",
    WriteOnly: "WriteOnly",
};
