// @pulumi/aws-native/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.certificatemanager = exports.ce = exports.cassandra = exports.budgets = exports.bedrock = exports.batch = exports.backupgateway = exports.backup = exports.b2bi = exports.autoscaling = exports.auditmanager = exports.athena = exports.arczonalshift = exports.aps = exports.appsync = exports.appstream = exports.apprunner = exports.applicationsignals = exports.applicationinsights = exports.applicationautoscaling = exports.appintegrations = exports.appflow = exports.appconfig = exports.apigatewayv2 = exports.apigateway = exports.amplifyuibuilder = exports.amplify = exports.amazonmq = exports.acmpca = exports.accessanalyzer = exports.Provider = exports.importValueOutput = exports.importValue = exports.getUrlSuffixOutput = exports.getUrlSuffix = exports.getSsmParameterStringOutput = exports.getSsmParameterString = exports.getSsmParameterListOutput = exports.getSsmParameterList = exports.getRegionOutput = exports.getRegion = exports.getPartitionOutput = exports.getPartition = exports.getAzsOutput = exports.getAzs = exports.getAccountIdOutput = exports.getAccountId = exports.ExtensionResource = exports.cidrOutput = exports.cidr = void 0;
exports.events = exports.entityresolution = exports.emrserverless = exports.emrcontainers = exports.emr = exports.elasticloadbalancingv2 = exports.elasticbeanstalk = exports.elasticache = exports.eks = exports.efs = exports.ecs = exports.ecr = exports.ec2 = exports.dynamodb = exports.docdbelastic = exports.dms = exports.directoryservice = exports.devopsguru = exports.devicefarm = exports.detective = exports.deadline = exports.datazone = exports.datasync = exports.datapipeline = exports.databrew = exports.customerprofiles = exports.controltower = exports.connectcampaignsv2 = exports.connectcampaigns = exports.connect = exports.configuration = exports.config = exports.comprehend = exports.cognito = exports.codestarnotifications = exports.codestarconnections = exports.codepipeline = exports.codegurureviewer = exports.codeguruprofiler = exports.codedeploy = exports.codeconnections = exports.codebuild = exports.codeartifact = exports.cloudwatch = exports.cloudtrail = exports.cloudfront = exports.cloudformation = exports.cleanroomsml = exports.cleanrooms = exports.chatbot = void 0;
exports.logs = exports.location = exports.lightsail = exports.licensemanager = exports.lex = exports.launchwizard = exports.lambda = exports.lakeformation = exports.kms = exports.kinesisvideo = exports.kinesisfirehose = exports.kinesisanalyticsv2 = exports.kinesis = exports.kendraranking = exports.kendra = exports.kafkaconnect = exports.ivschat = exports.ivs = exports.iotwireless = exports.iottwinmaker = exports.iotsitewise = exports.iotfleethub = exports.iotevents = exports.iotcoredeviceadvisor = exports.iotanalytics = exports.iot = exports.invoicing = exports.internetmonitor = exports.inspectorv2 = exports.inspector = exports.imagebuilder = exports.identitystore = exports.iam = exports.healthlake = exports.healthimaging = exports.guardduty = exports.groundstation = exports.greengrassv2 = exports.grafana = exports.glue = exports.globalaccelerator = exports.gamelift = exports.fsx = exports.frauddetector = exports.forecast = exports.fms = exports.fis = exports.finspace = exports.evidently = exports.eventschemas = void 0;
exports.route53profiles = exports.route53 = exports.rolesanywhere = exports.robomaker = exports.resourcegroups = exports.resourceexplorer2 = exports.resiliencehub = exports.rekognition = exports.refactorspaces = exports.redshiftserverless = exports.redshift = exports.rds = exports.rbin = exports.ram = exports.quicksight = exports.qldb = exports.qbusiness = exports.proton = exports.pipes = exports.pinpoint = exports.personalize = exports.pcs = exports.pcaconnectorscep = exports.pcaconnectorad = exports.paymentcryptography = exports.panorama = exports.osis = exports.organizations = exports.opsworkscm = exports.opensearchservice = exports.opensearchserverless = exports.omics = exports.oam = exports.nimblestudio = exports.networkmanager = exports.networkfirewall = exports.neptunegraph = exports.neptune = exports.mwaa = exports.msk = exports.memorydb = exports.mediatailor = exports.mediapackagev2 = exports.mediapackage = exports.medialive = exports.mediaconnect = exports.macie = exports.m2 = exports.lookoutvision = exports.lookoutmetrics = void 0;
exports.xray = exports.workspacesweb = exports.workspacesthinclient = exports.workspaces = exports.wisdom = exports.wafv2 = exports.vpclattice = exports.voiceid = exports.verifiedpermissions = exports.types = exports.transfer = exports.timestream = exports.systemsmanagersap = exports.synthetics = exports.supportapp = exports.stepfunctions = exports.sso = exports.ssmquicksetup = exports.ssmincidents = exports.ssmcontacts = exports.ssm = exports.sqs = exports.sns = exports.simspaceweaver = exports.signer = exports.shield = exports.ses = exports.servicecatalogappregistry = exports.servicecatalog = exports.securitylake = exports.securityhub = exports.secretsmanager = exports.scheduler = exports.sagemaker = exports.s3tables = exports.s3outposts = exports.s3objectlambda = exports.s3express = exports.s3 = exports.rum = exports.route53resolver = exports.route53recoveryreadiness = exports.route53recoverycontrol = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
exports.cidr = null;
exports.cidrOutput = null;
utilities.lazyLoad(exports, ["cidr", "cidrOutput"], () => require("./cidr"));
exports.ExtensionResource = null;
utilities.lazyLoad(exports, ["ExtensionResource"], () => require("./extensionResource"));
exports.getAccountId = null;
exports.getAccountIdOutput = null;
utilities.lazyLoad(exports, ["getAccountId", "getAccountIdOutput"], () => require("./getAccountId"));
exports.getAzs = null;
exports.getAzsOutput = null;
utilities.lazyLoad(exports, ["getAzs", "getAzsOutput"], () => require("./getAzs"));
exports.getPartition = null;
exports.getPartitionOutput = null;
utilities.lazyLoad(exports, ["getPartition", "getPartitionOutput"], () => require("./getPartition"));
exports.getRegion = null;
exports.getRegionOutput = null;
utilities.lazyLoad(exports, ["getRegion", "getRegionOutput"], () => require("./getRegion"));
exports.getSsmParameterList = null;
exports.getSsmParameterListOutput = null;
utilities.lazyLoad(exports, ["getSsmParameterList", "getSsmParameterListOutput"], () => require("./getSsmParameterList"));
exports.getSsmParameterString = null;
exports.getSsmParameterStringOutput = null;
utilities.lazyLoad(exports, ["getSsmParameterString", "getSsmParameterStringOutput"], () => require("./getSsmParameterString"));
exports.getUrlSuffix = null;
exports.getUrlSuffixOutput = null;
utilities.lazyLoad(exports, ["getUrlSuffix", "getUrlSuffixOutput"], () => require("./getUrlSuffix"));
exports.importValue = null;
exports.importValueOutput = null;
utilities.lazyLoad(exports, ["importValue", "importValueOutput"], () => require("./importValue"));
exports.Provider = null;
utilities.lazyLoad(exports, ["Provider"], () => require("./provider"));
// Export enums:
__exportStar(require("./types/enums"), exports);
// Export sub-modules:
const accessanalyzer = require("./accessanalyzer");
exports.accessanalyzer = accessanalyzer;
const acmpca = require("./acmpca");
exports.acmpca = acmpca;
const amazonmq = require("./amazonmq");
exports.amazonmq = amazonmq;
const amplify = require("./amplify");
exports.amplify = amplify;
const amplifyuibuilder = require("./amplifyuibuilder");
exports.amplifyuibuilder = amplifyuibuilder;
const apigateway = require("./apigateway");
exports.apigateway = apigateway;
const apigatewayv2 = require("./apigatewayv2");
exports.apigatewayv2 = apigatewayv2;
const appconfig = require("./appconfig");
exports.appconfig = appconfig;
const appflow = require("./appflow");
exports.appflow = appflow;
const appintegrations = require("./appintegrations");
exports.appintegrations = appintegrations;
const applicationautoscaling = require("./applicationautoscaling");
exports.applicationautoscaling = applicationautoscaling;
const applicationinsights = require("./applicationinsights");
exports.applicationinsights = applicationinsights;
const applicationsignals = require("./applicationsignals");
exports.applicationsignals = applicationsignals;
const apprunner = require("./apprunner");
exports.apprunner = apprunner;
const appstream = require("./appstream");
exports.appstream = appstream;
const appsync = require("./appsync");
exports.appsync = appsync;
const aps = require("./aps");
exports.aps = aps;
const arczonalshift = require("./arczonalshift");
exports.arczonalshift = arczonalshift;
const athena = require("./athena");
exports.athena = athena;
const auditmanager = require("./auditmanager");
exports.auditmanager = auditmanager;
const autoscaling = require("./autoscaling");
exports.autoscaling = autoscaling;
const b2bi = require("./b2bi");
exports.b2bi = b2bi;
const backup = require("./backup");
exports.backup = backup;
const backupgateway = require("./backupgateway");
exports.backupgateway = backupgateway;
const batch = require("./batch");
exports.batch = batch;
const bedrock = require("./bedrock");
exports.bedrock = bedrock;
const budgets = require("./budgets");
exports.budgets = budgets;
const cassandra = require("./cassandra");
exports.cassandra = cassandra;
const ce = require("./ce");
exports.ce = ce;
const certificatemanager = require("./certificatemanager");
exports.certificatemanager = certificatemanager;
const chatbot = require("./chatbot");
exports.chatbot = chatbot;
const cleanrooms = require("./cleanrooms");
exports.cleanrooms = cleanrooms;
const cleanroomsml = require("./cleanroomsml");
exports.cleanroomsml = cleanroomsml;
const cloudformation = require("./cloudformation");
exports.cloudformation = cloudformation;
const cloudfront = require("./cloudfront");
exports.cloudfront = cloudfront;
const cloudtrail = require("./cloudtrail");
exports.cloudtrail = cloudtrail;
const cloudwatch = require("./cloudwatch");
exports.cloudwatch = cloudwatch;
const codeartifact = require("./codeartifact");
exports.codeartifact = codeartifact;
const codebuild = require("./codebuild");
exports.codebuild = codebuild;
const codeconnections = require("./codeconnections");
exports.codeconnections = codeconnections;
const codedeploy = require("./codedeploy");
exports.codedeploy = codedeploy;
const codeguruprofiler = require("./codeguruprofiler");
exports.codeguruprofiler = codeguruprofiler;
const codegurureviewer = require("./codegurureviewer");
exports.codegurureviewer = codegurureviewer;
const codepipeline = require("./codepipeline");
exports.codepipeline = codepipeline;
const codestarconnections = require("./codestarconnections");
exports.codestarconnections = codestarconnections;
const codestarnotifications = require("./codestarnotifications");
exports.codestarnotifications = codestarnotifications;
const cognito = require("./cognito");
exports.cognito = cognito;
const comprehend = require("./comprehend");
exports.comprehend = comprehend;
const config = require("./config");
exports.config = config;
const configuration = require("./configuration");
exports.configuration = configuration;
const connect = require("./connect");
exports.connect = connect;
const connectcampaigns = require("./connectcampaigns");
exports.connectcampaigns = connectcampaigns;
const connectcampaignsv2 = require("./connectcampaignsv2");
exports.connectcampaignsv2 = connectcampaignsv2;
const controltower = require("./controltower");
exports.controltower = controltower;
const customerprofiles = require("./customerprofiles");
exports.customerprofiles = customerprofiles;
const databrew = require("./databrew");
exports.databrew = databrew;
const datapipeline = require("./datapipeline");
exports.datapipeline = datapipeline;
const datasync = require("./datasync");
exports.datasync = datasync;
const datazone = require("./datazone");
exports.datazone = datazone;
const deadline = require("./deadline");
exports.deadline = deadline;
const detective = require("./detective");
exports.detective = detective;
const devicefarm = require("./devicefarm");
exports.devicefarm = devicefarm;
const devopsguru = require("./devopsguru");
exports.devopsguru = devopsguru;
const directoryservice = require("./directoryservice");
exports.directoryservice = directoryservice;
const dms = require("./dms");
exports.dms = dms;
const docdbelastic = require("./docdbelastic");
exports.docdbelastic = docdbelastic;
const dynamodb = require("./dynamodb");
exports.dynamodb = dynamodb;
const ec2 = require("./ec2");
exports.ec2 = ec2;
const ecr = require("./ecr");
exports.ecr = ecr;
const ecs = require("./ecs");
exports.ecs = ecs;
const efs = require("./efs");
exports.efs = efs;
const eks = require("./eks");
exports.eks = eks;
const elasticache = require("./elasticache");
exports.elasticache = elasticache;
const elasticbeanstalk = require("./elasticbeanstalk");
exports.elasticbeanstalk = elasticbeanstalk;
const elasticloadbalancingv2 = require("./elasticloadbalancingv2");
exports.elasticloadbalancingv2 = elasticloadbalancingv2;
const emr = require("./emr");
exports.emr = emr;
const emrcontainers = require("./emrcontainers");
exports.emrcontainers = emrcontainers;
const emrserverless = require("./emrserverless");
exports.emrserverless = emrserverless;
const entityresolution = require("./entityresolution");
exports.entityresolution = entityresolution;
const events = require("./events");
exports.events = events;
const eventschemas = require("./eventschemas");
exports.eventschemas = eventschemas;
const evidently = require("./evidently");
exports.evidently = evidently;
const finspace = require("./finspace");
exports.finspace = finspace;
const fis = require("./fis");
exports.fis = fis;
const fms = require("./fms");
exports.fms = fms;
const forecast = require("./forecast");
exports.forecast = forecast;
const frauddetector = require("./frauddetector");
exports.frauddetector = frauddetector;
const fsx = require("./fsx");
exports.fsx = fsx;
const gamelift = require("./gamelift");
exports.gamelift = gamelift;
const globalaccelerator = require("./globalaccelerator");
exports.globalaccelerator = globalaccelerator;
const glue = require("./glue");
exports.glue = glue;
const grafana = require("./grafana");
exports.grafana = grafana;
const greengrassv2 = require("./greengrassv2");
exports.greengrassv2 = greengrassv2;
const groundstation = require("./groundstation");
exports.groundstation = groundstation;
const guardduty = require("./guardduty");
exports.guardduty = guardduty;
const healthimaging = require("./healthimaging");
exports.healthimaging = healthimaging;
const healthlake = require("./healthlake");
exports.healthlake = healthlake;
const iam = require("./iam");
exports.iam = iam;
const identitystore = require("./identitystore");
exports.identitystore = identitystore;
const imagebuilder = require("./imagebuilder");
exports.imagebuilder = imagebuilder;
const inspector = require("./inspector");
exports.inspector = inspector;
const inspectorv2 = require("./inspectorv2");
exports.inspectorv2 = inspectorv2;
const internetmonitor = require("./internetmonitor");
exports.internetmonitor = internetmonitor;
const invoicing = require("./invoicing");
exports.invoicing = invoicing;
const iot = require("./iot");
exports.iot = iot;
const iotanalytics = require("./iotanalytics");
exports.iotanalytics = iotanalytics;
const iotcoredeviceadvisor = require("./iotcoredeviceadvisor");
exports.iotcoredeviceadvisor = iotcoredeviceadvisor;
const iotevents = require("./iotevents");
exports.iotevents = iotevents;
const iotfleethub = require("./iotfleethub");
exports.iotfleethub = iotfleethub;
const iotsitewise = require("./iotsitewise");
exports.iotsitewise = iotsitewise;
const iottwinmaker = require("./iottwinmaker");
exports.iottwinmaker = iottwinmaker;
const iotwireless = require("./iotwireless");
exports.iotwireless = iotwireless;
const ivs = require("./ivs");
exports.ivs = ivs;
const ivschat = require("./ivschat");
exports.ivschat = ivschat;
const kafkaconnect = require("./kafkaconnect");
exports.kafkaconnect = kafkaconnect;
const kendra = require("./kendra");
exports.kendra = kendra;
const kendraranking = require("./kendraranking");
exports.kendraranking = kendraranking;
const kinesis = require("./kinesis");
exports.kinesis = kinesis;
const kinesisanalyticsv2 = require("./kinesisanalyticsv2");
exports.kinesisanalyticsv2 = kinesisanalyticsv2;
const kinesisfirehose = require("./kinesisfirehose");
exports.kinesisfirehose = kinesisfirehose;
const kinesisvideo = require("./kinesisvideo");
exports.kinesisvideo = kinesisvideo;
const kms = require("./kms");
exports.kms = kms;
const lakeformation = require("./lakeformation");
exports.lakeformation = lakeformation;
const lambda = require("./lambda");
exports.lambda = lambda;
const launchwizard = require("./launchwizard");
exports.launchwizard = launchwizard;
const lex = require("./lex");
exports.lex = lex;
const licensemanager = require("./licensemanager");
exports.licensemanager = licensemanager;
const lightsail = require("./lightsail");
exports.lightsail = lightsail;
const location = require("./location");
exports.location = location;
const logs = require("./logs");
exports.logs = logs;
const lookoutmetrics = require("./lookoutmetrics");
exports.lookoutmetrics = lookoutmetrics;
const lookoutvision = require("./lookoutvision");
exports.lookoutvision = lookoutvision;
const m2 = require("./m2");
exports.m2 = m2;
const macie = require("./macie");
exports.macie = macie;
const mediaconnect = require("./mediaconnect");
exports.mediaconnect = mediaconnect;
const medialive = require("./medialive");
exports.medialive = medialive;
const mediapackage = require("./mediapackage");
exports.mediapackage = mediapackage;
const mediapackagev2 = require("./mediapackagev2");
exports.mediapackagev2 = mediapackagev2;
const mediatailor = require("./mediatailor");
exports.mediatailor = mediatailor;
const memorydb = require("./memorydb");
exports.memorydb = memorydb;
const msk = require("./msk");
exports.msk = msk;
const mwaa = require("./mwaa");
exports.mwaa = mwaa;
const neptune = require("./neptune");
exports.neptune = neptune;
const neptunegraph = require("./neptunegraph");
exports.neptunegraph = neptunegraph;
const networkfirewall = require("./networkfirewall");
exports.networkfirewall = networkfirewall;
const networkmanager = require("./networkmanager");
exports.networkmanager = networkmanager;
const nimblestudio = require("./nimblestudio");
exports.nimblestudio = nimblestudio;
const oam = require("./oam");
exports.oam = oam;
const omics = require("./omics");
exports.omics = omics;
const opensearchserverless = require("./opensearchserverless");
exports.opensearchserverless = opensearchserverless;
const opensearchservice = require("./opensearchservice");
exports.opensearchservice = opensearchservice;
const opsworkscm = require("./opsworkscm");
exports.opsworkscm = opsworkscm;
const organizations = require("./organizations");
exports.organizations = organizations;
const osis = require("./osis");
exports.osis = osis;
const panorama = require("./panorama");
exports.panorama = panorama;
const paymentcryptography = require("./paymentcryptography");
exports.paymentcryptography = paymentcryptography;
const pcaconnectorad = require("./pcaconnectorad");
exports.pcaconnectorad = pcaconnectorad;
const pcaconnectorscep = require("./pcaconnectorscep");
exports.pcaconnectorscep = pcaconnectorscep;
const pcs = require("./pcs");
exports.pcs = pcs;
const personalize = require("./personalize");
exports.personalize = personalize;
const pinpoint = require("./pinpoint");
exports.pinpoint = pinpoint;
const pipes = require("./pipes");
exports.pipes = pipes;
const proton = require("./proton");
exports.proton = proton;
const qbusiness = require("./qbusiness");
exports.qbusiness = qbusiness;
const qldb = require("./qldb");
exports.qldb = qldb;
const quicksight = require("./quicksight");
exports.quicksight = quicksight;
const ram = require("./ram");
exports.ram = ram;
const rbin = require("./rbin");
exports.rbin = rbin;
const rds = require("./rds");
exports.rds = rds;
const redshift = require("./redshift");
exports.redshift = redshift;
const redshiftserverless = require("./redshiftserverless");
exports.redshiftserverless = redshiftserverless;
const refactorspaces = require("./refactorspaces");
exports.refactorspaces = refactorspaces;
const rekognition = require("./rekognition");
exports.rekognition = rekognition;
const resiliencehub = require("./resiliencehub");
exports.resiliencehub = resiliencehub;
const resourceexplorer2 = require("./resourceexplorer2");
exports.resourceexplorer2 = resourceexplorer2;
const resourcegroups = require("./resourcegroups");
exports.resourcegroups = resourcegroups;
const robomaker = require("./robomaker");
exports.robomaker = robomaker;
const rolesanywhere = require("./rolesanywhere");
exports.rolesanywhere = rolesanywhere;
const route53 = require("./route53");
exports.route53 = route53;
const route53profiles = require("./route53profiles");
exports.route53profiles = route53profiles;
const route53recoverycontrol = require("./route53recoverycontrol");
exports.route53recoverycontrol = route53recoverycontrol;
const route53recoveryreadiness = require("./route53recoveryreadiness");
exports.route53recoveryreadiness = route53recoveryreadiness;
const route53resolver = require("./route53resolver");
exports.route53resolver = route53resolver;
const rum = require("./rum");
exports.rum = rum;
const s3 = require("./s3");
exports.s3 = s3;
const s3express = require("./s3express");
exports.s3express = s3express;
const s3objectlambda = require("./s3objectlambda");
exports.s3objectlambda = s3objectlambda;
const s3outposts = require("./s3outposts");
exports.s3outposts = s3outposts;
const s3tables = require("./s3tables");
exports.s3tables = s3tables;
const sagemaker = require("./sagemaker");
exports.sagemaker = sagemaker;
const scheduler = require("./scheduler");
exports.scheduler = scheduler;
const secretsmanager = require("./secretsmanager");
exports.secretsmanager = secretsmanager;
const securityhub = require("./securityhub");
exports.securityhub = securityhub;
const securitylake = require("./securitylake");
exports.securitylake = securitylake;
const servicecatalog = require("./servicecatalog");
exports.servicecatalog = servicecatalog;
const servicecatalogappregistry = require("./servicecatalogappregistry");
exports.servicecatalogappregistry = servicecatalogappregistry;
const ses = require("./ses");
exports.ses = ses;
const shield = require("./shield");
exports.shield = shield;
const signer = require("./signer");
exports.signer = signer;
const simspaceweaver = require("./simspaceweaver");
exports.simspaceweaver = simspaceweaver;
const sns = require("./sns");
exports.sns = sns;
const sqs = require("./sqs");
exports.sqs = sqs;
const ssm = require("./ssm");
exports.ssm = ssm;
const ssmcontacts = require("./ssmcontacts");
exports.ssmcontacts = ssmcontacts;
const ssmincidents = require("./ssmincidents");
exports.ssmincidents = ssmincidents;
const ssmquicksetup = require("./ssmquicksetup");
exports.ssmquicksetup = ssmquicksetup;
const sso = require("./sso");
exports.sso = sso;
const stepfunctions = require("./stepfunctions");
exports.stepfunctions = stepfunctions;
const supportapp = require("./supportapp");
exports.supportapp = supportapp;
const synthetics = require("./synthetics");
exports.synthetics = synthetics;
const systemsmanagersap = require("./systemsmanagersap");
exports.systemsmanagersap = systemsmanagersap;
const timestream = require("./timestream");
exports.timestream = timestream;
const transfer = require("./transfer");
exports.transfer = transfer;
const types = require("./types");
exports.types = types;
const verifiedpermissions = require("./verifiedpermissions");
exports.verifiedpermissions = verifiedpermissions;
const voiceid = require("./voiceid");
exports.voiceid = voiceid;
const vpclattice = require("./vpclattice");
exports.vpclattice = vpclattice;
const wafv2 = require("./wafv2");
exports.wafv2 = wafv2;
const wisdom = require("./wisdom");
exports.wisdom = wisdom;
const workspaces = require("./workspaces");
exports.workspaces = workspaces;
const workspacesthinclient = require("./workspacesthinclient");
exports.workspacesthinclient = workspacesthinclient;
const workspacesweb = require("./workspacesweb");
exports.workspacesweb = workspacesweb;
const xray = require("./xray");
exports.xray = xray;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:index:ExtensionResource":
                return new exports.ExtensionResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "index", _module);
pulumi.runtime.registerResourcePackage("aws-native", {
    version: utilities.getVersion(),
    constructProvider: (name, type, urn) => {
        if (type !== "pulumi:providers:aws-native") {
            throw new Error(`unknown provider type ${type}`);
        }
        return new exports.Provider(name, undefined, { urn });
    },
});
