// @pulumi/aws-native/types/enums/rds/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalClusterEngine = exports.DbProxyTargetGroupTargetGroupName = exports.DbProxyEngineFamily = exports.DbProxyEndpointTargetRole = exports.DbProxyAuthFormatIamAuth = exports.DbProxyAuthFormatClientPasswordAuthType = exports.DbProxyAuthFormatAuthScheme = exports.DbInstanceProcessorFeatureName = exports.CustomDbEngineVersionStatus = void 0;
exports.CustomDbEngineVersionStatus = {
    Available: "available",
    Inactive: "inactive",
    InactiveExceptRestore: "inactive-except-restore",
};
exports.DbInstanceProcessorFeatureName = {
    CoreCount: "coreCount",
    ThreadsPerCore: "threadsPerCore",
};
exports.DbProxyAuthFormatAuthScheme = {
    Secrets: "SECRETS",
};
exports.DbProxyAuthFormatClientPasswordAuthType = {
    MysqlNativePassword: "MYSQL_NATIVE_PASSWORD",
    PostgresScramSha256: "POSTGRES_SCRAM_SHA_256",
    PostgresMd5: "POSTGRES_MD5",
    SqlServerAuthentication: "SQL_SERVER_AUTHENTICATION",
};
exports.DbProxyAuthFormatIamAuth = {
    Disabled: "DISABLED",
    Required: "REQUIRED",
    Enabled: "ENABLED",
};
exports.DbProxyEndpointTargetRole = {
    ReadWrite: "READ_WRITE",
    ReadOnly: "READ_ONLY",
};
exports.DbProxyEngineFamily = {
    Mysql: "MYSQL",
    Postgresql: "POSTGRESQL",
    Sqlserver: "SQLSERVER",
};
exports.DbProxyTargetGroupTargetGroupName = {
    Default: "default",
};
exports.GlobalClusterEngine = {
    Aurora: "aurora",
    AuroraMysql: "aurora-mysql",
    AuroraPostgresql: "aurora-postgresql",
};
