// @pulumi/aws-native/logs/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transformer = exports.SubscriptionFilter = exports.ResourcePolicy = exports.QueryDefinition = exports.MetricFilter = exports.LogStream = exports.LogGroup = exports.LogAnomalyDetector = exports.Integration = exports.getTransformerOutput = exports.getTransformer = exports.getSubscriptionFilterOutput = exports.getSubscriptionFilter = exports.getResourcePolicyOutput = exports.getResourcePolicy = exports.getQueryDefinitionOutput = exports.getQueryDefinition = exports.getMetricFilterOutput = exports.getMetricFilter = exports.getLogGroupOutput = exports.getLogGroup = exports.getLogAnomalyDetectorOutput = exports.getLogAnomalyDetector = exports.getIntegrationOutput = exports.getIntegration = exports.getDestinationOutput = exports.getDestination = exports.getDeliverySourceOutput = exports.getDeliverySource = exports.getDeliveryDestinationOutput = exports.getDeliveryDestination = exports.getDeliveryOutput = exports.getDelivery = exports.getAccountPolicyOutput = exports.getAccountPolicy = exports.Destination = exports.DeliverySource = exports.DeliveryDestination = exports.Delivery = exports.AccountPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccountPolicy = null;
utilities.lazyLoad(exports, ["AccountPolicy"], () => require("./accountPolicy"));
exports.Delivery = null;
utilities.lazyLoad(exports, ["Delivery"], () => require("./delivery"));
exports.DeliveryDestination = null;
utilities.lazyLoad(exports, ["DeliveryDestination"], () => require("./deliveryDestination"));
exports.DeliverySource = null;
utilities.lazyLoad(exports, ["DeliverySource"], () => require("./deliverySource"));
exports.Destination = null;
utilities.lazyLoad(exports, ["Destination"], () => require("./destination"));
exports.getAccountPolicy = null;
exports.getAccountPolicyOutput = null;
utilities.lazyLoad(exports, ["getAccountPolicy", "getAccountPolicyOutput"], () => require("./getAccountPolicy"));
exports.getDelivery = null;
exports.getDeliveryOutput = null;
utilities.lazyLoad(exports, ["getDelivery", "getDeliveryOutput"], () => require("./getDelivery"));
exports.getDeliveryDestination = null;
exports.getDeliveryDestinationOutput = null;
utilities.lazyLoad(exports, ["getDeliveryDestination", "getDeliveryDestinationOutput"], () => require("./getDeliveryDestination"));
exports.getDeliverySource = null;
exports.getDeliverySourceOutput = null;
utilities.lazyLoad(exports, ["getDeliverySource", "getDeliverySourceOutput"], () => require("./getDeliverySource"));
exports.getDestination = null;
exports.getDestinationOutput = null;
utilities.lazyLoad(exports, ["getDestination", "getDestinationOutput"], () => require("./getDestination"));
exports.getIntegration = null;
exports.getIntegrationOutput = null;
utilities.lazyLoad(exports, ["getIntegration", "getIntegrationOutput"], () => require("./getIntegration"));
exports.getLogAnomalyDetector = null;
exports.getLogAnomalyDetectorOutput = null;
utilities.lazyLoad(exports, ["getLogAnomalyDetector", "getLogAnomalyDetectorOutput"], () => require("./getLogAnomalyDetector"));
exports.getLogGroup = null;
exports.getLogGroupOutput = null;
utilities.lazyLoad(exports, ["getLogGroup", "getLogGroupOutput"], () => require("./getLogGroup"));
exports.getMetricFilter = null;
exports.getMetricFilterOutput = null;
utilities.lazyLoad(exports, ["getMetricFilter", "getMetricFilterOutput"], () => require("./getMetricFilter"));
exports.getQueryDefinition = null;
exports.getQueryDefinitionOutput = null;
utilities.lazyLoad(exports, ["getQueryDefinition", "getQueryDefinitionOutput"], () => require("./getQueryDefinition"));
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.getSubscriptionFilter = null;
exports.getSubscriptionFilterOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionFilter", "getSubscriptionFilterOutput"], () => require("./getSubscriptionFilter"));
exports.getTransformer = null;
exports.getTransformerOutput = null;
utilities.lazyLoad(exports, ["getTransformer", "getTransformerOutput"], () => require("./getTransformer"));
exports.Integration = null;
utilities.lazyLoad(exports, ["Integration"], () => require("./integration"));
exports.LogAnomalyDetector = null;
utilities.lazyLoad(exports, ["LogAnomalyDetector"], () => require("./logAnomalyDetector"));
exports.LogGroup = null;
utilities.lazyLoad(exports, ["LogGroup"], () => require("./logGroup"));
exports.LogStream = null;
utilities.lazyLoad(exports, ["LogStream"], () => require("./logStream"));
exports.MetricFilter = null;
utilities.lazyLoad(exports, ["MetricFilter"], () => require("./metricFilter"));
exports.QueryDefinition = null;
utilities.lazyLoad(exports, ["QueryDefinition"], () => require("./queryDefinition"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.SubscriptionFilter = null;
utilities.lazyLoad(exports, ["SubscriptionFilter"], () => require("./subscriptionFilter"));
exports.Transformer = null;
utilities.lazyLoad(exports, ["Transformer"], () => require("./transformer"));
// Export enums:
__exportStar(require("../types/enums/logs"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:logs:AccountPolicy":
                return new exports.AccountPolicy(name, undefined, { urn });
            case "aws-native:logs:Delivery":
                return new exports.Delivery(name, undefined, { urn });
            case "aws-native:logs:DeliveryDestination":
                return new exports.DeliveryDestination(name, undefined, { urn });
            case "aws-native:logs:DeliverySource":
                return new exports.DeliverySource(name, undefined, { urn });
            case "aws-native:logs:Destination":
                return new exports.Destination(name, undefined, { urn });
            case "aws-native:logs:Integration":
                return new exports.Integration(name, undefined, { urn });
            case "aws-native:logs:LogAnomalyDetector":
                return new exports.LogAnomalyDetector(name, undefined, { urn });
            case "aws-native:logs:LogGroup":
                return new exports.LogGroup(name, undefined, { urn });
            case "aws-native:logs:LogStream":
                return new exports.LogStream(name, undefined, { urn });
            case "aws-native:logs:MetricFilter":
                return new exports.MetricFilter(name, undefined, { urn });
            case "aws-native:logs:QueryDefinition":
                return new exports.QueryDefinition(name, undefined, { urn });
            case "aws-native:logs:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws-native:logs:SubscriptionFilter":
                return new exports.SubscriptionFilter(name, undefined, { urn });
            case "aws-native:logs:Transformer":
                return new exports.Transformer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "logs", _module);
