// @pulumi/aws-native/logs/logAnomalyDetector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogAnomalyDetector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::Logs::LogAnomalyDetector resource specifies a CloudWatch Logs LogAnomalyDetector.
 */
class LogAnomalyDetector extends pulumi.CustomResource {
    /**
     * Get an existing LogAnomalyDetector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LogAnomalyDetector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LogAnomalyDetector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LogAnomalyDetector.__pulumiType;
    }
    /**
     * Create a LogAnomalyDetector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["anomalyVisibilityTime"] = args ? args.anomalyVisibilityTime : undefined;
            resourceInputs["detectorName"] = args ? args.detectorName : undefined;
            resourceInputs["evaluationFrequency"] = args ? args.evaluationFrequency : undefined;
            resourceInputs["filterPattern"] = args ? args.filterPattern : undefined;
            resourceInputs["kmsKeyId"] = args ? args.kmsKeyId : undefined;
            resourceInputs["logGroupArnList"] = args ? args.logGroupArnList : undefined;
            resourceInputs["anomalyDetectorArn"] = undefined /*out*/;
            resourceInputs["anomalyDetectorStatus"] = undefined /*out*/;
            resourceInputs["creationTimeStamp"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeStamp"] = undefined /*out*/;
        }
        else {
            resourceInputs["accountId"] = undefined /*out*/;
            resourceInputs["anomalyDetectorArn"] = undefined /*out*/;
            resourceInputs["anomalyDetectorStatus"] = undefined /*out*/;
            resourceInputs["anomalyVisibilityTime"] = undefined /*out*/;
            resourceInputs["creationTimeStamp"] = undefined /*out*/;
            resourceInputs["detectorName"] = undefined /*out*/;
            resourceInputs["evaluationFrequency"] = undefined /*out*/;
            resourceInputs["filterPattern"] = undefined /*out*/;
            resourceInputs["kmsKeyId"] = undefined /*out*/;
            resourceInputs["lastModifiedTimeStamp"] = undefined /*out*/;
            resourceInputs["logGroupArnList"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LogAnomalyDetector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LogAnomalyDetector = LogAnomalyDetector;
/** @internal */
LogAnomalyDetector.__pulumiType = 'aws-native:logs:LogAnomalyDetector';
