// @pulumi/aws-native/types/enums/bedrock/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromptVersionPromptTemplateType = exports.PromptTemplateType = exports.PromptConversationRole = exports.KnowledgeBaseType = exports.KnowledgeBaseStorageType = exports.KnowledgeBaseStatus = exports.GuardrailTopicType = exports.GuardrailStatus = exports.GuardrailSensitiveInformationAction = exports.GuardrailPiiEntityType = exports.GuardrailManagedWordsType = exports.GuardrailFilterStrength = exports.GuardrailContextualGroundingFilterType = exports.GuardrailContentFilterType = exports.FlowVersionPromptTemplateType = exports.FlowVersionFlowStatus = exports.FlowVersionFlowNodeType = exports.FlowVersionFlowNodeIoDataType = exports.FlowVersionFlowConnectionType = exports.FlowStatus = exports.FlowPromptTemplateType = exports.FlowNodeType = exports.FlowNodeIoDataType = exports.FlowConnectionType = exports.DataSourceWebScopeType = exports.DataSourceType = exports.DataSourceTransformationStepToApply = exports.DataSourceStatus = exports.DataSourceSharePointSourceConfigurationHostType = exports.DataSourceSharePointSourceConfigurationAuthType = exports.DataSourceSalesforceSourceConfigurationAuthType = exports.DataSourceParsingStrategy = exports.DataSourceDataDeletionPolicy = exports.DataSourceCrawlFilterConfigurationType = exports.DataSourceConfluenceSourceConfigurationHostType = exports.DataSourceConfluenceSourceConfigurationAuthType = exports.DataSourceChunkingStrategy = exports.ApplicationInferenceProfileInferenceProfileType = exports.ApplicationInferenceProfileInferenceProfileStatus = exports.AgentType = exports.AgentStatus = exports.AgentRequireConfirmation = exports.AgentPromptType = exports.AgentPromptState = exports.AgentKnowledgeBaseState = exports.AgentCustomControlMethod = exports.AgentCreationMode = exports.AgentAliasStatus = exports.AgentActionGroupState = exports.AgentActionGroupSignature = void 0;
exports.AgentActionGroupSignature = {
    AmazonUserInput: "AMAZON.UserInput",
    AmazonCodeInterpreter: "AMAZON.CodeInterpreter",
};
exports.AgentActionGroupState = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.AgentAliasStatus = {
    Creating: "CREATING",
    Prepared: "PREPARED",
    Failed: "FAILED",
    Updating: "UPDATING",
    Deleting: "DELETING",
};
exports.AgentCreationMode = {
    Default: "DEFAULT",
    Overridden: "OVERRIDDEN",
};
exports.AgentCustomControlMethod = {
    ReturnControl: "RETURN_CONTROL",
};
exports.AgentKnowledgeBaseState = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.AgentPromptState = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.AgentPromptType = {
    PreProcessing: "PRE_PROCESSING",
    Orchestration: "ORCHESTRATION",
    PostProcessing: "POST_PROCESSING",
    KnowledgeBaseResponseGeneration: "KNOWLEDGE_BASE_RESPONSE_GENERATION",
};
exports.AgentRequireConfirmation = {
    Enabled: "ENABLED",
    Disabled: "DISABLED",
};
exports.AgentStatus = {
    Creating: "CREATING",
    Preparing: "PREPARING",
    Prepared: "PREPARED",
    NotPrepared: "NOT_PREPARED",
    Deleting: "DELETING",
    Failed: "FAILED",
    Versioning: "VERSIONING",
    Updating: "UPDATING",
};
exports.AgentType = {
    String: "string",
    Number: "number",
    Integer: "integer",
    Boolean: "boolean",
    Array: "array",
};
exports.ApplicationInferenceProfileInferenceProfileStatus = {
    Active: "ACTIVE",
};
exports.ApplicationInferenceProfileInferenceProfileType = {
    Application: "APPLICATION",
    SystemDefined: "SYSTEM_DEFINED",
};
exports.DataSourceChunkingStrategy = {
    FixedSize: "FIXED_SIZE",
    None: "NONE",
    Hierarchical: "HIERARCHICAL",
    Semantic: "SEMANTIC",
};
exports.DataSourceConfluenceSourceConfigurationAuthType = {
    Basic: "BASIC",
    Oauth2ClientCredentials: "OAUTH2_CLIENT_CREDENTIALS",
};
exports.DataSourceConfluenceSourceConfigurationHostType = {
    Saas: "SAAS",
};
exports.DataSourceCrawlFilterConfigurationType = {
    Pattern: "PATTERN",
};
exports.DataSourceDataDeletionPolicy = {
    Retain: "RETAIN",
    Delete: "DELETE",
};
exports.DataSourceParsingStrategy = {
    BedrockFoundationModel: "BEDROCK_FOUNDATION_MODEL",
};
exports.DataSourceSalesforceSourceConfigurationAuthType = {
    Oauth2ClientCredentials: "OAUTH2_CLIENT_CREDENTIALS",
};
exports.DataSourceSharePointSourceConfigurationAuthType = {
    Oauth2ClientCredentials: "OAUTH2_CLIENT_CREDENTIALS",
};
exports.DataSourceSharePointSourceConfigurationHostType = {
    Online: "ONLINE",
};
exports.DataSourceStatus = {
    Available: "AVAILABLE",
    Deleting: "DELETING",
    DeleteUnsuccessful: "DELETE_UNSUCCESSFUL",
};
exports.DataSourceTransformationStepToApply = {
    PostChunking: "POST_CHUNKING",
};
exports.DataSourceType = {
    S3: "S3",
    Confluence: "CONFLUENCE",
    Salesforce: "SALESFORCE",
    Sharepoint: "SHAREPOINT",
    Web: "WEB",
};
exports.DataSourceWebScopeType = {
    HostOnly: "HOST_ONLY",
    Subdomains: "SUBDOMAINS",
};
exports.FlowConnectionType = {
    Data: "Data",
    Conditional: "Conditional",
};
exports.FlowNodeIoDataType = {
    String: "String",
    Number: "Number",
    Boolean: "Boolean",
    Object: "Object",
    Array: "Array",
};
exports.FlowNodeType = {
    InputType: "Input",
    OutputType: "Output",
    KnowledgeBase: "KnowledgeBase",
    Condition: "Condition",
    Lex: "Lex",
    Prompt: "Prompt",
    LambdaFunction: "LambdaFunction",
    Agent: "Agent",
    Storage: "Storage",
    Retrieval: "Retrieval",
    Iterator: "Iterator",
    Collector: "Collector",
};
exports.FlowPromptTemplateType = {
    Text: "TEXT",
};
exports.FlowStatus = {
    Failed: "Failed",
    Prepared: "Prepared",
    Preparing: "Preparing",
    NotPrepared: "NotPrepared",
};
exports.FlowVersionFlowConnectionType = {
    Data: "Data",
    Conditional: "Conditional",
};
exports.FlowVersionFlowNodeIoDataType = {
    String: "String",
    Number: "Number",
    Boolean: "Boolean",
    Object: "Object",
    Array: "Array",
};
exports.FlowVersionFlowNodeType = {
    InputType: "Input",
    OutputType: "Output",
    KnowledgeBase: "KnowledgeBase",
    Condition: "Condition",
    Lex: "Lex",
    Prompt: "Prompt",
    LambdaFunction: "LambdaFunction",
    Agent: "Agent",
    Iterator: "Iterator",
    Collector: "Collector",
    Storage: "Storage",
    Retrieval: "Retrieval",
};
exports.FlowVersionFlowStatus = {
    Failed: "Failed",
    Prepared: "Prepared",
    Preparing: "Preparing",
    NotPrepared: "NotPrepared",
};
exports.FlowVersionPromptTemplateType = {
    Text: "TEXT",
};
exports.GuardrailContentFilterType = {
    Sexual: "SEXUAL",
    Violence: "VIOLENCE",
    Hate: "HATE",
    Insults: "INSULTS",
    Misconduct: "MISCONDUCT",
    PromptAttack: "PROMPT_ATTACK",
};
exports.GuardrailContextualGroundingFilterType = {
    Grounding: "GROUNDING",
    Relevance: "RELEVANCE",
};
exports.GuardrailFilterStrength = {
    None: "NONE",
    Low: "LOW",
    Medium: "MEDIUM",
    High: "HIGH",
};
exports.GuardrailManagedWordsType = {
    Profanity: "PROFANITY",
};
exports.GuardrailPiiEntityType = {
    Address: "ADDRESS",
    Age: "AGE",
    AwsAccessKey: "AWS_ACCESS_KEY",
    AwsSecretKey: "AWS_SECRET_KEY",
    CaHealthNumber: "CA_HEALTH_NUMBER",
    CaSocialInsuranceNumber: "CA_SOCIAL_INSURANCE_NUMBER",
    CreditDebitCardCvv: "CREDIT_DEBIT_CARD_CVV",
    CreditDebitCardExpiry: "CREDIT_DEBIT_CARD_EXPIRY",
    CreditDebitCardNumber: "CREDIT_DEBIT_CARD_NUMBER",
    DriverId: "DRIVER_ID",
    Email: "EMAIL",
    InternationalBankAccountNumber: "INTERNATIONAL_BANK_ACCOUNT_NUMBER",
    IpAddress: "IP_ADDRESS",
    LicensePlate: "LICENSE_PLATE",
    MacAddress: "MAC_ADDRESS",
    Name: "NAME",
    Password: "PASSWORD",
    Phone: "PHONE",
    Pin: "PIN",
    SwiftCode: "SWIFT_CODE",
    UkNationalHealthServiceNumber: "UK_NATIONAL_HEALTH_SERVICE_NUMBER",
    UkNationalInsuranceNumber: "UK_NATIONAL_INSURANCE_NUMBER",
    UkUniqueTaxpayerReferenceNumber: "UK_UNIQUE_TAXPAYER_REFERENCE_NUMBER",
    Url: "URL",
    Username: "USERNAME",
    UsBankAccountNumber: "US_BANK_ACCOUNT_NUMBER",
    UsBankRoutingNumber: "US_BANK_ROUTING_NUMBER",
    UsIndividualTaxIdentificationNumber: "US_INDIVIDUAL_TAX_IDENTIFICATION_NUMBER",
    UsPassportNumber: "US_PASSPORT_NUMBER",
    UsSocialSecurityNumber: "US_SOCIAL_SECURITY_NUMBER",
    VehicleIdentificationNumber: "VEHICLE_IDENTIFICATION_NUMBER",
};
exports.GuardrailSensitiveInformationAction = {
    Block: "BLOCK",
    Anonymize: "ANONYMIZE",
};
exports.GuardrailStatus = {
    Creating: "CREATING",
    Updating: "UPDATING",
    Versioning: "VERSIONING",
    Ready: "READY",
    Failed: "FAILED",
    Deleting: "DELETING",
};
exports.GuardrailTopicType = {
    Deny: "DENY",
};
exports.KnowledgeBaseStatus = {
    Creating: "CREATING",
    Active: "ACTIVE",
    Deleting: "DELETING",
    Updating: "UPDATING",
    Failed: "FAILED",
    DeleteUnsuccessful: "DELETE_UNSUCCESSFUL",
};
exports.KnowledgeBaseStorageType = {
    OpensearchServerless: "OPENSEARCH_SERVERLESS",
    Pinecone: "PINECONE",
    Rds: "RDS",
    MongoDbAtlas: "MONGO_DB_ATLAS",
};
exports.KnowledgeBaseType = {
    Vector: "VECTOR",
};
exports.PromptConversationRole = {
    User: "user",
    Assistant: "assistant",
};
exports.PromptTemplateType = {
    Text: "TEXT",
    Chat: "CHAT",
};
exports.PromptVersionPromptTemplateType = {
    Text: "TEXT",
};
