// @pulumi/aws-native/ssm/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourcePolicy = exports.ResourceDataSync = exports.PatchBaseline = exports.Parameter = exports.getResourcePolicyOutput = exports.getResourcePolicy = exports.getResourceDataSyncOutput = exports.getResourceDataSync = exports.getPatchBaselineOutput = exports.getPatchBaseline = exports.getParameterOutput = exports.getParameter = exports.getDocumentOutput = exports.getDocument = exports.getAssociationOutput = exports.getAssociation = exports.Document = exports.Association = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Association = null;
utilities.lazyLoad(exports, ["Association"], () => require("./association"));
exports.Document = null;
utilities.lazyLoad(exports, ["Document"], () => require("./document"));
exports.getAssociation = null;
exports.getAssociationOutput = null;
utilities.lazyLoad(exports, ["getAssociation", "getAssociationOutput"], () => require("./getAssociation"));
exports.getDocument = null;
exports.getDocumentOutput = null;
utilities.lazyLoad(exports, ["getDocument", "getDocumentOutput"], () => require("./getDocument"));
exports.getParameter = null;
exports.getParameterOutput = null;
utilities.lazyLoad(exports, ["getParameter", "getParameterOutput"], () => require("./getParameter"));
exports.getPatchBaseline = null;
exports.getPatchBaselineOutput = null;
utilities.lazyLoad(exports, ["getPatchBaseline", "getPatchBaselineOutput"], () => require("./getPatchBaseline"));
exports.getResourceDataSync = null;
exports.getResourceDataSyncOutput = null;
utilities.lazyLoad(exports, ["getResourceDataSync", "getResourceDataSyncOutput"], () => require("./getResourceDataSync"));
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.Parameter = null;
utilities.lazyLoad(exports, ["Parameter"], () => require("./parameter"));
exports.PatchBaseline = null;
utilities.lazyLoad(exports, ["PatchBaseline"], () => require("./patchBaseline"));
exports.ResourceDataSync = null;
utilities.lazyLoad(exports, ["ResourceDataSync"], () => require("./resourceDataSync"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
// Export enums:
__exportStar(require("../types/enums/ssm"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws-native:ssm:Association":
                return new exports.Association(name, undefined, { urn });
            case "aws-native:ssm:Document":
                return new exports.Document(name, undefined, { urn });
            case "aws-native:ssm:Parameter":
                return new exports.Parameter(name, undefined, { urn });
            case "aws-native:ssm:PatchBaseline":
                return new exports.PatchBaseline(name, undefined, { urn });
            case "aws-native:ssm:ResourceDataSync":
                return new exports.ResourceDataSync(name, undefined, { urn });
            case "aws-native:ssm:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws-native", "ssm", _module);
