// @pulumi/aws-native/types/enums/iottwinmaker/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityStatusState = exports.EntityStatusErrorPropertiesCode = exports.EntityStatusError1PropertiesCode = exports.EntityPropertyGroupGroupType = exports.EntityDataTypeType = exports.ComponentTypeStatusState = exports.ComponentTypeStatusErrorPropertiesCode = exports.ComponentTypeStatusError1PropertiesCode = exports.ComponentTypePropertyGroupGroupType = exports.ComponentTypeFunctionScope = exports.ComponentTypeDataTypeType = void 0;
exports.ComponentTypeDataTypeType = {
    Relationship: "RELATIONSHIP",
    String: "STRING",
    Long: "LONG",
    Boolean: "BOOLEAN",
    Integer: "INTEGER",
    Double: "DOUBLE",
    List: "LIST",
    Map: "MAP",
};
exports.ComponentTypeFunctionScope = {
    Entity: "ENTITY",
    Workspace: "WORKSPACE",
};
exports.ComponentTypePropertyGroupGroupType = {
    Tabular: "TABULAR",
};
exports.ComponentTypeStatusError1PropertiesCode = {
    ValidationError: "VALIDATION_ERROR",
    InternalFailure: "INTERNAL_FAILURE",
};
exports.ComponentTypeStatusErrorPropertiesCode = {
    ValidationError: "VALIDATION_ERROR",
    InternalFailure: "INTERNAL_FAILURE",
};
exports.ComponentTypeStatusState = {
    Creating: "CREATING",
    Updating: "UPDATING",
    Deleting: "DELETING",
    Active: "ACTIVE",
    Error: "ERROR",
};
exports.EntityDataTypeType = {
    Relationship: "RELATIONSHIP",
    String: "STRING",
    Long: "LONG",
    Boolean: "BOOLEAN",
    Integer: "INTEGER",
    Double: "DOUBLE",
    List: "LIST",
    Map: "MAP",
};
exports.EntityPropertyGroupGroupType = {
    Tabular: "TABULAR",
};
exports.EntityStatusError1PropertiesCode = {
    ValidationError: "VALIDATION_ERROR",
    InternalFailure: "INTERNAL_FAILURE",
};
exports.EntityStatusErrorPropertiesCode = {
    ValidationError: "VALIDATION_ERROR",
    InternalFailure: "INTERNAL_FAILURE",
};
exports.EntityStatusState = {
    Creating: "CREATING",
    Updating: "UPDATING",
    Deleting: "DELETING",
    Active: "ACTIVE",
    Error: "ERROR",
};
