// @pulumi/aws-native/backup/logicallyAirGappedBackupVault.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogicallyAirGappedBackupVault = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Backup::LogicallyAirGappedBackupVault
 */
class LogicallyAirGappedBackupVault extends pulumi.CustomResource {
    /**
     * Get an existing LogicallyAirGappedBackupVault resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LogicallyAirGappedBackupVault(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LogicallyAirGappedBackupVault.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LogicallyAirGappedBackupVault.__pulumiType;
    }
    /**
     * Create a LogicallyAirGappedBackupVault resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.maxRetentionDays === undefined) && !opts.urn) {
                throw new Error("Missing required property 'maxRetentionDays'");
            }
            if ((!args || args.minRetentionDays === undefined) && !opts.urn) {
                throw new Error("Missing required property 'minRetentionDays'");
            }
            resourceInputs["accessPolicy"] = args ? args.accessPolicy : undefined;
            resourceInputs["backupVaultName"] = args ? args.backupVaultName : undefined;
            resourceInputs["backupVaultTags"] = args ? args.backupVaultTags : undefined;
            resourceInputs["maxRetentionDays"] = args ? args.maxRetentionDays : undefined;
            resourceInputs["minRetentionDays"] = args ? args.minRetentionDays : undefined;
            resourceInputs["notifications"] = args ? args.notifications : undefined;
            resourceInputs["vaultState"] = args ? args.vaultState : undefined;
            resourceInputs["vaultType"] = args ? args.vaultType : undefined;
            resourceInputs["backupVaultArn"] = undefined /*out*/;
            resourceInputs["encryptionKeyArn"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessPolicy"] = undefined /*out*/;
            resourceInputs["backupVaultArn"] = undefined /*out*/;
            resourceInputs["backupVaultName"] = undefined /*out*/;
            resourceInputs["backupVaultTags"] = undefined /*out*/;
            resourceInputs["encryptionKeyArn"] = undefined /*out*/;
            resourceInputs["maxRetentionDays"] = undefined /*out*/;
            resourceInputs["minRetentionDays"] = undefined /*out*/;
            resourceInputs["notifications"] = undefined /*out*/;
            resourceInputs["vaultState"] = undefined /*out*/;
            resourceInputs["vaultType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["backupVaultName", "maxRetentionDays", "minRetentionDays"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(LogicallyAirGappedBackupVault.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LogicallyAirGappedBackupVault = LogicallyAirGappedBackupVault;
/** @internal */
LogicallyAirGappedBackupVault.__pulumiType = 'aws-native:backup:LogicallyAirGappedBackupVault';
