// @pulumi/aws-native/backup/restoreTestingSelection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestoreTestingSelection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Backup::RestoreTestingSelection
 */
class RestoreTestingSelection extends pulumi.CustomResource {
    /**
     * Get an existing RestoreTestingSelection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RestoreTestingSelection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RestoreTestingSelection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RestoreTestingSelection.__pulumiType;
    }
    /**
     * Create a RestoreTestingSelection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.iamRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iamRoleArn'");
            }
            if ((!args || args.protectedResourceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protectedResourceType'");
            }
            if ((!args || args.restoreTestingPlanName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'restoreTestingPlanName'");
            }
            resourceInputs["iamRoleArn"] = args ? args.iamRoleArn : undefined;
            resourceInputs["protectedResourceArns"] = args ? args.protectedResourceArns : undefined;
            resourceInputs["protectedResourceConditions"] = args ? args.protectedResourceConditions : undefined;
            resourceInputs["protectedResourceType"] = args ? args.protectedResourceType : undefined;
            resourceInputs["restoreMetadataOverrides"] = args ? args.restoreMetadataOverrides : undefined;
            resourceInputs["restoreTestingPlanName"] = args ? args.restoreTestingPlanName : undefined;
            resourceInputs["restoreTestingSelectionName"] = args ? args.restoreTestingSelectionName : undefined;
            resourceInputs["validationWindowHours"] = args ? args.validationWindowHours : undefined;
        }
        else {
            resourceInputs["iamRoleArn"] = undefined /*out*/;
            resourceInputs["protectedResourceArns"] = undefined /*out*/;
            resourceInputs["protectedResourceConditions"] = undefined /*out*/;
            resourceInputs["protectedResourceType"] = undefined /*out*/;
            resourceInputs["restoreMetadataOverrides"] = undefined /*out*/;
            resourceInputs["restoreTestingPlanName"] = undefined /*out*/;
            resourceInputs["restoreTestingSelectionName"] = undefined /*out*/;
            resourceInputs["validationWindowHours"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const replaceOnChanges = { replaceOnChanges: ["protectedResourceType", "restoreTestingPlanName", "restoreTestingSelectionName"] };
        opts = pulumi.mergeOptions(opts, replaceOnChanges);
        super(RestoreTestingSelection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RestoreTestingSelection = RestoreTestingSelection;
/** @internal */
RestoreTestingSelection.__pulumiType = 'aws-native:backup:RestoreTestingSelection';
