// @pulumi/aws-native/types/enums/cassandra/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableMode = exports.TableEncryptionType = exports.TableClusteringKeyColumnOrderBy = exports.KeyspaceReplicationSpecificationReplicationStrategy = exports.KeyspaceRegionListItem = void 0;
exports.KeyspaceRegionListItem = {
    ApNortheast1: "ap-northeast-1",
    ApNortheast2: "ap-northeast-2",
    ApSouth1: "ap-south-1",
    ApSoutheast1: "ap-southeast-1",
    ApSoutheast2: "ap-southeast-2",
    CaCentral1: "ca-central-1",
    EuCentral1: "eu-central-1",
    EuNorth1: "eu-north-1",
    EuWest1: "eu-west-1",
    EuWest2: "eu-west-2",
    EuWest3: "eu-west-3",
    SaEast1: "sa-east-1",
    UsEast1: "us-east-1",
    UsEast2: "us-east-2",
    UsWest1: "us-west-1",
    UsWest2: "us-west-2",
};
exports.KeyspaceReplicationSpecificationReplicationStrategy = {
    SingleRegion: "SINGLE_REGION",
    MultiRegion: "MULTI_REGION",
};
exports.TableClusteringKeyColumnOrderBy = {
    Asc: "ASC",
    Desc: "DESC",
};
exports.TableEncryptionType = {
    AwsOwnedKmsKey: "AWS_OWNED_KMS_KEY",
    CustomerManagedKmsKey: "CUSTOMER_MANAGED_KMS_KEY",
};
exports.TableMode = {
    Provisioned: "PROVISIONED",
    OnDemand: "ON_DEMAND",
};
